// @weavy/uikit-web/dist/es/react.js
var gf = typeof globalThis < "u" ? globalThis : typeof window < "u" ? window : typeof global < "u" ? global : typeof self < "u" ? self : {};
function wf(S) {
  return S && S.__esModule && Object.prototype.hasOwnProperty.call(S, "default") ? S.default : S;
}
function Ef(S) {
  if (S.__esModule)
    return S;
  var O = S.default;
  if (typeof O == "function") {
    var I = function h() {
      return this instanceof h ? Reflect.construct(O, arguments, this.constructor) : O.apply(this, arguments);
    };
    I.prototype = O.prototype;
  } else
    I = {};
  return Object.defineProperty(I, "__esModule", { value: !0 }), Object.keys(S).forEach(function(h) {
    var ge = Object.getOwnPropertyDescriptor(S, h);
    Object.defineProperty(I, h, ge.get ? ge : {
      enumerable: !0,
      get: function() {
        return S[h];
      }
    });
  }), I;
}
var Ii = { exports: {} }, U = {};
var ji, ks;
function Os() {
  if (ks)
    return ji;
  ks = 1;
  var S = Object.getOwnPropertySymbols, O = Object.prototype.hasOwnProperty, I = Object.prototype.propertyIsEnumerable;
  function h(X) {
    if (X == null)
      throw new TypeError("Object.assign cannot be called with null or undefined");
    return Object(X);
  }
  function ge() {
    try {
      if (!Object.assign)
        return !1;
      var X = new String("abc");
      if (X[5] = "de", Object.getOwnPropertyNames(X)[0] === "5")
        return !1;
      for (var re = {}, H = 0; H < 10; H++)
        re["_" + String.fromCharCode(H)] = H;
      var fe = Object.getOwnPropertyNames(re).map(function(le) {
        return re[le];
      });
      if (fe.join("") !== "0123456789")
        return !1;
      var ce = {};
      return "abcdefghijklmnopqrst".split("").forEach(function(le) {
        ce[le] = le;
      }), Object.keys(Object.assign({}, ce)).join("") === "abcdefghijklmnopqrst";
    } catch {
      return !1;
    }
  }
  return ji = ge() ? Object.assign : function(X, re) {
    for (var H, fe = h(X), ce, le = 1; le < arguments.length; le++) {
      H = Object(arguments[le]);
      for (var Qe in H)
        O.call(H, Qe) && (fe[Qe] = H[Qe]);
      if (S) {
        ce = S(H);
        for (var we = 0; we < ce.length; we++)
          I.call(H, ce[we]) && (fe[ce[we]] = H[ce[we]]);
      }
    }
    return fe;
  }, ji;
}
var xs;
function pf() {
  if (xs)
    return U;
  xs = 1;
  var S = Os(), O = typeof Symbol == "function" && Symbol.for, I = O ? Symbol.for("react.element") : 60103, h = O ? Symbol.for("react.portal") : 60106, ge = O ? Symbol.for("react.fragment") : 60107, X = O ? Symbol.for("react.strict_mode") : 60108, re = O ? Symbol.for("react.profiler") : 60114, H = O ? Symbol.for("react.provider") : 60109, fe = O ? Symbol.for("react.context") : 60110, ce = O ? Symbol.for("react.forward_ref") : 60112, le = O ? Symbol.for("react.suspense") : 60113, Qe = O ? Symbol.for("react.memo") : 60115, we = O ? Symbol.for("react.lazy") : 60116, Ot = typeof Symbol == "function" && Symbol.iterator;
  function He(s) {
    for (var p = "https://reactjs.org/docs/error-decoder.html?invariant=" + s, E = 1; E < arguments.length; E++)
      p += "&args[]=" + encodeURIComponent(arguments[E]);
    return "Minified React error #" + s + "; visit " + p + " for the full message or use the non-minified dev environment for full errors and additional helpful warnings.";
  }
  var Rt = { isMounted: function() {
    return !1;
  }, enqueueForceUpdate: function() {
  }, enqueueReplaceState: function() {
  }, enqueueSetState: function() {
  } }, Re = {};
  function J(s, p, E) {
    this.props = s, this.context = p, this.refs = Re, this.updater = E || Rt;
  }
  J.prototype.isReactComponent = {}, J.prototype.setState = function(s, p) {
    if (typeof s != "object" && typeof s != "function" && s != null)
      throw Error(He(85));
    this.updater.enqueueSetState(this, s, p, "setState");
  }, J.prototype.forceUpdate = function(s) {
    this.updater.enqueueForceUpdate(this, s, "forceUpdate");
  };
  function be() {
  }
  be.prototype = J.prototype;
  function et(s, p, E) {
    this.props = s, this.context = p, this.refs = Re, this.updater = E || Rt;
  }
  var De = et.prototype = new be();
  De.constructor = et, S(De, J.prototype), De.isPureReactComponent = !0;
  var $e = { current: null }, Ee = Object.prototype.hasOwnProperty, Ke = { key: !0, ref: !0, __self: !0, __source: !0 };
  function q(s, p, E) {
    var C, R = {}, A = null, F = null;
    if (p != null)
      for (C in p.ref !== void 0 && (F = p.ref), p.key !== void 0 && (A = "" + p.key), p)
        Ee.call(p, C) && !Ke.hasOwnProperty(C) && (R[C] = p[C]);
    var P = arguments.length - 2;
    if (P === 1)
      R.children = E;
    else if (1 < P) {
      for (var ee = Array(P), xe = 0; xe < P; xe++)
        ee[xe] = arguments[xe + 2];
      R.children = ee;
    }
    if (s && s.defaultProps)
      for (C in P = s.defaultProps, P)
        R[C] === void 0 && (R[C] = P[C]);
    return { $$typeof: I, type: s, key: A, ref: F, props: R, _owner: $e.current };
  }
  function ae(s, p) {
    return { $$typeof: I, type: s.type, key: p, ref: s.ref, props: s.props, _owner: s._owner };
  }
  function Me(s) {
    return typeof s == "object" && s !== null && s.$$typeof === I;
  }
  function G(s) {
    var p = { "=": "=0", ":": "=2" };
    return "$" + ("" + s).replace(/[=:]/g, function(E) {
      return p[E];
    });
  }
  var b = /\/+/g, tt = [];
  function B(s, p, E, C) {
    if (tt.length) {
      var R = tt.pop();
      return R.result = s, R.keyPrefix = p, R.func = E, R.context = C, R.count = 0, R;
    }
    return { result: s, keyPrefix: p, func: E, context: C, count: 0 };
  }
  function $(s) {
    s.result = null, s.keyPrefix = null, s.func = null, s.context = null, s.count = 0, 10 > tt.length && tt.push(s);
  }
  function ze(s, p, E, C) {
    var R = typeof s;
    (R === "undefined" || R === "boolean") && (s = null);
    var A = !1;
    if (s === null)
      A = !0;
    else
      switch (R) {
        case "string":
        case "number":
          A = !0;
          break;
        case "object":
          switch (s.$$typeof) {
            case I:
            case h:
              A = !0;
          }
      }
    if (A)
      return E(C, s, p === "" ? "." + Te(s, 0) : p), 1;
    if (A = 0, p = p === "" ? "." : p + ":", Array.isArray(s))
      for (var F = 0; F < s.length; F++) {
        R = s[F];
        var P = p + Te(R, F);
        A += ze(R, P, E, C);
      }
    else if (s === null || typeof s != "object" ? P = null : (P = Ot && s[Ot] || s["@@iterator"], P = typeof P == "function" ? P : null), typeof P == "function")
      for (s = P.call(s), F = 0; !(R = s.next()).done; )
        R = R.value, P = p + Te(R, F++), A += ze(R, P, E, C);
    else if (R === "object")
      throw E = "" + s, Error(He(31, E === "[object Object]" ? "object with keys {" + Object.keys(s).join(", ") + "}" : E, ""));
    return A;
  }
  function de(s, p, E) {
    return s == null ? 0 : ze(s, "", p, E);
  }
  function Te(s, p) {
    return typeof s == "object" && s !== null && s.key != null ? G(s.key) : p.toString(36);
  }
  function nt(s, p) {
    s.func.call(s.context, p, s.count++);
  }
  function ke(s, p, E) {
    var C = s.result, R = s.keyPrefix;
    s = s.func.call(s.context, p, s.count++), Array.isArray(s) ? Ue(s, C, E, function(A) {
      return A;
    }) : s != null && (Me(s) && (s = ae(s, R + (!s.key || p && p.key === s.key ? "" : ("" + s.key).replace(b, "$&/") + "/") + E)), C.push(s));
  }
  function Ue(s, p, E, C, R) {
    var A = "";
    E != null && (A = ("" + E).replace(b, "$&/") + "/"), p = B(p, A, C, R), de(s, ke, p), $(p);
  }
  var rt = { current: null };
  function pe() {
    var s = rt.current;
    if (s === null)
      throw Error(He(321));
    return s;
  }
  var y = { ReactCurrentDispatcher: rt, ReactCurrentBatchConfig: { suspense: null }, ReactCurrentOwner: $e, IsSomeRendererActing: { current: !1 }, assign: S };
  return U.Children = { map: function(s, p, E) {
    if (s == null)
      return s;
    var C = [];
    return Ue(s, C, null, p, E), C;
  }, forEach: function(s, p, E) {
    if (s == null)
      return s;
    p = B(null, null, p, E), de(s, nt, p), $(p);
  }, count: function(s) {
    return de(s, function() {
      return null;
    }, null);
  }, toArray: function(s) {
    var p = [];
    return Ue(s, p, null, function(E) {
      return E;
    }), p;
  }, only: function(s) {
    if (!Me(s))
      throw Error(He(143));
    return s;
  } }, U.Component = J, U.Fragment = ge, U.Profiler = re, U.PureComponent = et, U.StrictMode = X, U.Suspense = le, U.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED = y, U.cloneElement = function(s, p, E) {
    if (s == null)
      throw Error(He(267, s));
    var C = S({}, s.props), R = s.key, A = s.ref, F = s._owner;
    if (p != null) {
      if (p.ref !== void 0 && (A = p.ref, F = $e.current), p.key !== void 0 && (R = "" + p.key), s.type && s.type.defaultProps)
        var P = s.type.defaultProps;
      for (ee in p)
        Ee.call(p, ee) && !Ke.hasOwnProperty(ee) && (C[ee] = p[ee] === void 0 && P !== void 0 ? P[ee] : p[ee]);
    }
    var ee = arguments.length - 2;
    if (ee === 1)
      C.children = E;
    else if (1 < ee) {
      P = Array(ee);
      for (var xe = 0; xe < ee; xe++)
        P[xe] = arguments[xe + 2];
      C.children = P;
    }
    return {
      $$typeof: I,
      type: s.type,
      key: R,
      ref: A,
      props: C,
      _owner: F
    };
  }, U.createContext = function(s, p) {
    return p === void 0 && (p = null), s = { $$typeof: fe, _calculateChangedBits: p, _currentValue: s, _currentValue2: s, _threadCount: 0, Provider: null, Consumer: null }, s.Provider = { $$typeof: H, _context: s }, s.Consumer = s;
  }, U.createElement = q, U.createFactory = function(s) {
    var p = q.bind(null, s);
    return p.type = s, p;
  }, U.createRef = function() {
    return { current: null };
  }, U.forwardRef = function(s) {
    return { $$typeof: ce, render: s };
  }, U.isValidElement = Me, U.lazy = function(s) {
    return { $$typeof: we, _ctor: s, _status: -1, _result: null };
  }, U.memo = function(s, p) {
    return { $$typeof: Qe, type: s, compare: p === void 0 ? null : p };
  }, U.useCallback = function(s, p) {
    return pe().useCallback(s, p);
  }, U.useContext = function(s, p) {
    return pe().useContext(s, p);
  }, U.useDebugValue = function() {
  }, U.useEffect = function(s, p) {
    return pe().useEffect(s, p);
  }, U.useImperativeHandle = function(s, p, E) {
    return pe().useImperativeHandle(s, p, E);
  }, U.useLayoutEffect = function(s, p) {
    return pe().useLayoutEffect(s, p);
  }, U.useMemo = function(s, p) {
    return pe().useMemo(s, p);
  }, U.useReducer = function(s, p, E) {
    return pe().useReducer(s, p, E);
  }, U.useRef = function(s) {
    return pe().useRef(s);
  }, U.useState = function(s) {
    return pe().useState(s);
  }, U.version = "16.14.0", U;
}
var Ss;
function mf() {
  return Ss || (Ss = 1, Ii.exports = pf()), Ii.exports;
}
var Fi = { exports: {} }, Oe = {}, Li = { exports: {} }, Di = {};
var Cs;
function hf() {
  return Cs || (Cs = 1, function(S) {
    var O, I, h, ge, X;
    if (typeof window > "u" || typeof MessageChannel != "function") {
      var re = null, H = null, fe = function() {
        if (re !== null)
          try {
            var y = S.unstable_now();
            re(!0, y), re = null;
          } catch (s) {
            throw setTimeout(fe, 0), s;
          }
      }, ce = Date.now();
      S.unstable_now = function() {
        return Date.now() - ce;
      }, O = function(y) {
        re !== null ? setTimeout(O, 0, y) : (re = y, setTimeout(fe, 0));
      }, I = function(y, s) {
        H = setTimeout(y, s);
      }, h = function() {
        clearTimeout(H);
      }, ge = function() {
        return !1;
      }, X = S.unstable_forceFrameRate = function() {
      };
    } else {
      var le = window.performance, Qe = window.Date, we = window.setTimeout, Ot = window.clearTimeout;
      if (typeof console < "u") {
        var He = window.cancelAnimationFrame;
        typeof window.requestAnimationFrame != "function" && console.error("This browser doesn't support requestAnimationFrame. Make sure that you load a polyfill in older browsers. https://fb.me/react-polyfills"), typeof He != "function" && console.error("This browser doesn't support cancelAnimationFrame. Make sure that you load a polyfill in older browsers. https://fb.me/react-polyfills");
      }
      if (typeof le == "object" && typeof le.now == "function")
        S.unstable_now = function() {
          return le.now();
        };
      else {
        var Rt = Qe.now();
        S.unstable_now = function() {
          return Qe.now() - Rt;
        };
      }
      var Re = !1, J = null, be = -1, et = 5, De = 0;
      ge = function() {
        return S.unstable_now() >= De;
      }, X = function() {
      }, S.unstable_forceFrameRate = function(y) {
        0 > y || 125 < y ? console.error("forceFrameRate takes a positive int between 0 and 125, forcing framerates higher than 125 fps is not unsupported") : et = 0 < y ? Math.floor(1e3 / y) : 5;
      };
      var $e = new MessageChannel(), Ee = $e.port2;
      $e.port1.onmessage = function() {
        if (J !== null) {
          var y = S.unstable_now();
          De = y + et;
          try {
            J(!0, y) ? Ee.postMessage(null) : (Re = !1, J = null);
          } catch (s) {
            throw Ee.postMessage(null), s;
          }
        } else
          Re = !1;
      }, O = function(y) {
        J = y, Re || (Re = !0, Ee.postMessage(null));
      }, I = function(y, s) {
        be = we(function() {
          y(S.unstable_now());
        }, s);
      }, h = function() {
        Ot(be), be = -1;
      };
    }
    function Ke(y, s) {
      var p = y.length;
      y.push(s);
      e:
        for (; ; ) {
          var E = p - 1 >>> 1, C = y[E];
          if (C !== void 0 && 0 < Me(C, s))
            y[E] = s, y[p] = C, p = E;
          else
            break e;
        }
    }
    function q(y) {
      return y = y[0], y === void 0 ? null : y;
    }
    function ae(y) {
      var s = y[0];
      if (s !== void 0) {
        var p = y.pop();
        if (p !== s) {
          y[0] = p;
          e:
            for (var E = 0, C = y.length; E < C; ) {
              var R = 2 * (E + 1) - 1, A = y[R], F = R + 1, P = y[F];
              if (A !== void 0 && 0 > Me(A, p))
                P !== void 0 && 0 > Me(P, A) ? (y[E] = P, y[F] = p, E = F) : (y[E] = A, y[R] = p, E = R);
              else if (P !== void 0 && 0 > Me(P, p))
                y[E] = P, y[F] = p, E = F;
              else
                break e;
            }
        }
        return s;
      }
      return null;
    }
    function Me(y, s) {
      var p = y.sortIndex - s.sortIndex;
      return p !== 0 ? p : y.id - s.id;
    }
    var G = [], b = [], tt = 1, B = null, $ = 3, ze = !1, de = !1, Te = !1;
    function nt(y) {
      for (var s = q(b); s !== null; ) {
        if (s.callback === null)
          ae(b);
        else if (s.startTime <= y)
          ae(b), s.sortIndex = s.expirationTime, Ke(G, s);
        else
          break;
        s = q(b);
      }
    }
    function ke(y) {
      if (Te = !1, nt(y), !de)
        if (q(G) !== null)
          de = !0, O(Ue);
        else {
          var s = q(b);
          s !== null && I(ke, s.startTime - y);
        }
    }
    function Ue(y, s) {
      de = !1, Te && (Te = !1, h()), ze = !0;
      var p = $;
      try {
        for (nt(s), B = q(G); B !== null && (!(B.expirationTime > s) || y && !ge()); ) {
          var E = B.callback;
          if (E !== null) {
            B.callback = null, $ = B.priorityLevel;
            var C = E(B.expirationTime <= s);
            s = S.unstable_now(), typeof C == "function" ? B.callback = C : B === q(G) && ae(G), nt(s);
          } else
            ae(G);
          B = q(G);
        }
        if (B !== null)
          var R = !0;
        else {
          var A = q(b);
          A !== null && I(ke, A.startTime - s), R = !1;
        }
        return R;
      } finally {
        B = null, $ = p, ze = !1;
      }
    }
    function rt(y) {
      switch (y) {
        case 1:
          return -1;
        case 2:
          return 250;
        case 5:
          return 1073741823;
        case 4:
          return 1e4;
        default:
          return 5e3;
      }
    }
    var pe = X;
    S.unstable_IdlePriority = 5, S.unstable_ImmediatePriority = 1, S.unstable_LowPriority = 4, S.unstable_NormalPriority = 3, S.unstable_Profiling = null, S.unstable_UserBlockingPriority = 2, S.unstable_cancelCallback = function(y) {
      y.callback = null;
    }, S.unstable_continueExecution = function() {
      de || ze || (de = !0, O(Ue));
    }, S.unstable_getCurrentPriorityLevel = function() {
      return $;
    }, S.unstable_getFirstCallbackNode = function() {
      return q(G);
    }, S.unstable_next = function(y) {
      switch ($) {
        case 1:
        case 2:
        case 3:
          var s = 3;
          break;
        default:
          s = $;
      }
      var p = $;
      $ = s;
      try {
        return y();
      } finally {
        $ = p;
      }
    }, S.unstable_pauseExecution = function() {
    }, S.unstable_requestPaint = pe, S.unstable_runWithPriority = function(y, s) {
      switch (y) {
        case 1:
        case 2:
        case 3:
        case 4:
        case 5:
          break;
        default:
          y = 3;
      }
      var p = $;
      $ = y;
      try {
        return s();
      } finally {
        $ = p;
      }
    }, S.unstable_scheduleCallback = function(y, s, p) {
      var E = S.unstable_now();
      if (typeof p == "object" && p !== null) {
        var C = p.delay;
        C = typeof C == "number" && 0 < C ? E + C : E, p = typeof p.timeout == "number" ? p.timeout : rt(y);
      } else
        p = rt(y), C = E;
      return p = C + p, y = { id: tt++, callback: s, priorityLevel: y, startTime: C, expirationTime: p, sortIndex: -1 }, C > E ? (y.sortIndex = C, Ke(b, y), q(G) === null && y === q(b) && (Te ? h() : Te = !0, I(ke, C - E))) : (y.sortIndex = p, Ke(G, y), de || ze || (de = !0, O(Ue))), y;
    }, S.unstable_shouldYield = function() {
      var y = S.unstable_now();
      nt(y);
      var s = q(G);
      return s !== B && B !== null && s !== null && s.callback !== null && s.startTime <= y && s.expirationTime < B.expirationTime || ge();
    }, S.unstable_wrapCallback = function(y) {
      var s = $;
      return function() {
        var p = $;
        $ = s;
        try {
          return y.apply(this, arguments);
        } finally {
          $ = p;
        }
      };
    };
  }(Di)), Di;
}
var _s;
function vf() {
  return _s || (_s = 1, Li.exports = hf()), Li.exports;
}
var Ps;
function yf() {
  if (Ps)
    return Oe;
  Ps = 1;
  var S = mf(), O = Os(), I = vf();
  function h(e) {
    for (var t = "https://reactjs.org/docs/error-decoder.html?invariant=" + e, n = 1; n < arguments.length; n++)
      t += "&args[]=" + encodeURIComponent(arguments[n]);
    return "Minified React error #" + e + "; visit " + t + " for the full message or use the non-minified dev environment for full errors and additional helpful warnings.";
  }
  if (!S)
    throw Error(h(227));
  function ge(e, t, n, r, l, i, o, u, c) {
    var d = Array.prototype.slice.call(arguments, 3);
    try {
      t.apply(n, d);
    } catch (w) {
      this.onError(w);
    }
  }
  var X = !1, re = null, H = !1, fe = null, ce = { onError: function(e) {
    X = !0, re = e;
  } };
  function le(e, t, n, r, l, i, o, u, c) {
    X = !1, re = null, ge.apply(ce, arguments);
  }
  function Qe(e, t, n, r, l, i, o, u, c) {
    if (le.apply(this, arguments), X) {
      if (X) {
        var d = re;
        X = !1, re = null;
      } else
        throw Error(h(198));
      H || (H = !0, fe = d);
    }
  }
  var we = null, Ot = null, He = null;
  function Rt(e, t, n) {
    var r = e.type || "unknown-event";
    e.currentTarget = He(n), Qe(r, t, void 0, e), e.currentTarget = null;
  }
  var Re = null, J = {};
  function be() {
    if (Re)
      for (var e in J) {
        var t = J[e], n = Re.indexOf(e);
        if (!(-1 < n))
          throw Error(h(96, e));
        if (!De[n]) {
          if (!t.extractEvents)
            throw Error(h(97, e));
          De[n] = t, n = t.eventTypes;
          for (var r in n) {
            var l = void 0, i = n[r], o = t, u = r;
            if ($e.hasOwnProperty(u))
              throw Error(h(99, u));
            $e[u] = i;
            var c = i.phasedRegistrationNames;
            if (c) {
              for (l in c)
                c.hasOwnProperty(l) && et(c[l], o, u);
              l = !0;
            } else
              i.registrationName ? (et(i.registrationName, o, u), l = !0) : l = !1;
            if (!l)
              throw Error(h(98, r, e));
          }
        }
      }
  }
  function et(e, t, n) {
    if (Ee[e])
      throw Error(h(100, e));
    Ee[e] = t, Ke[e] = t.eventTypes[n].dependencies;
  }
  var De = [], $e = {}, Ee = {}, Ke = {};
  function q(e) {
    var t = !1, n;
    for (n in e)
      if (e.hasOwnProperty(n)) {
        var r = e[n];
        if (!J.hasOwnProperty(n) || J[n] !== r) {
          if (J[n])
            throw Error(h(102, n));
          J[n] = r, t = !0;
        }
      }
    t && be();
  }
  var ae = !(typeof window > "u" || typeof window.document > "u" || typeof window.document.createElement > "u"), Me = null, G = null, b = null;
  function tt(e) {
    if (e = Ot(e)) {
      if (typeof Me != "function")
        throw Error(h(280));
      var t = e.stateNode;
      t && (t = we(t), Me(e.stateNode, e.type, t));
    }
  }
  function B(e) {
    G ? b ? b.push(e) : b = [e] : G = e;
  }
  function $() {
    if (G) {
      var e = G, t = b;
      if (b = G = null, tt(e), t)
        for (e = 0; e < t.length; e++)
          tt(t[e]);
    }
  }
  function ze(e, t) {
    return e(t);
  }
  function de(e, t, n, r, l) {
    return e(t, n, r, l);
  }
  function Te() {
  }
  var nt = ze, ke = !1, Ue = !1;
  function rt() {
    (G !== null || b !== null) && (Te(), $());
  }
  function pe(e, t, n) {
    if (Ue)
      return e(t, n);
    Ue = !0;
    try {
      return nt(e, t, n);
    } finally {
      Ue = !1, rt();
    }
  }
  var y = /^[:A-Z_a-z\u00C0-\u00D6\u00D8-\u00F6\u00F8-\u02FF\u0370-\u037D\u037F-\u1FFF\u200C-\u200D\u2070-\u218F\u2C00-\u2FEF\u3001-\uD7FF\uF900-\uFDCF\uFDF0-\uFFFD][:A-Z_a-z\u00C0-\u00D6\u00D8-\u00F6\u00F8-\u02FF\u0370-\u037D\u037F-\u1FFF\u200C-\u200D\u2070-\u218F\u2C00-\u2FEF\u3001-\uD7FF\uF900-\uFDCF\uFDF0-\uFFFD\-.0-9\u00B7\u0300-\u036F\u203F-\u2040]*$/, s = Object.prototype.hasOwnProperty, p = {}, E = {};
  function C(e) {
    return s.call(E, e) ? !0 : s.call(p, e) ? !1 : y.test(e) ? E[e] = !0 : (p[e] = !0, !1);
  }
  function R(e, t, n, r) {
    if (n !== null && n.type === 0)
      return !1;
    switch (typeof t) {
      case "function":
      case "symbol":
        return !0;
      case "boolean":
        return r ? !1 : n !== null ? !n.acceptsBooleans : (e = e.toLowerCase().slice(0, 5), e !== "data-" && e !== "aria-");
      default:
        return !1;
    }
  }
  function A(e, t, n, r) {
    if (t === null || typeof t > "u" || R(e, t, n, r))
      return !0;
    if (r)
      return !1;
    if (n !== null)
      switch (n.type) {
        case 3:
          return !t;
        case 4:
          return t === !1;
        case 5:
          return isNaN(t);
        case 6:
          return isNaN(t) || 1 > t;
      }
    return !1;
  }
  function F(e, t, n, r, l, i) {
    this.acceptsBooleans = t === 2 || t === 3 || t === 4, this.attributeName = r, this.attributeNamespace = l, this.mustUseProperty = n, this.propertyName = e, this.type = t, this.sanitizeURL = i;
  }
  var P = {};
  "children dangerouslySetInnerHTML defaultValue defaultChecked innerHTML suppressContentEditableWarning suppressHydrationWarning style".split(" ").forEach(function(e) {
    P[e] = new F(e, 0, !1, e, null, !1);
  }), [["acceptCharset", "accept-charset"], ["className", "class"], ["htmlFor", "for"], ["httpEquiv", "http-equiv"]].forEach(function(e) {
    var t = e[0];
    P[t] = new F(t, 1, !1, e[1], null, !1);
  }), ["contentEditable", "draggable", "spellCheck", "value"].forEach(function(e) {
    P[e] = new F(e, 2, !1, e.toLowerCase(), null, !1);
  }), ["autoReverse", "externalResourcesRequired", "focusable", "preserveAlpha"].forEach(function(e) {
    P[e] = new F(e, 2, !1, e, null, !1);
  }), "allowFullScreen async autoFocus autoPlay controls default defer disabled disablePictureInPicture formNoValidate hidden loop noModule noValidate open playsInline readOnly required reversed scoped seamless itemScope".split(" ").forEach(function(e) {
    P[e] = new F(e, 3, !1, e.toLowerCase(), null, !1);
  }), ["checked", "multiple", "muted", "selected"].forEach(function(e) {
    P[e] = new F(e, 3, !0, e, null, !1);
  }), ["capture", "download"].forEach(function(e) {
    P[e] = new F(e, 4, !1, e, null, !1);
  }), ["cols", "rows", "size", "span"].forEach(function(e) {
    P[e] = new F(e, 6, !1, e, null, !1);
  }), ["rowSpan", "start"].forEach(function(e) {
    P[e] = new F(e, 5, !1, e.toLowerCase(), null, !1);
  });
  var ee = /[\-:]([a-z])/g;
  function xe(e) {
    return e[1].toUpperCase();
  }
  "accent-height alignment-baseline arabic-form baseline-shift cap-height clip-path clip-rule color-interpolation color-interpolation-filters color-profile color-rendering dominant-baseline enable-background fill-opacity fill-rule flood-color flood-opacity font-family font-size font-size-adjust font-stretch font-style font-variant font-weight glyph-name glyph-orientation-horizontal glyph-orientation-vertical horiz-adv-x horiz-origin-x image-rendering letter-spacing lighting-color marker-end marker-mid marker-start overline-position overline-thickness paint-order panose-1 pointer-events rendering-intent shape-rendering stop-color stop-opacity strikethrough-position strikethrough-thickness stroke-dasharray stroke-dashoffset stroke-linecap stroke-linejoin stroke-miterlimit stroke-opacity stroke-width text-anchor text-decoration text-rendering underline-position underline-thickness unicode-bidi unicode-range units-per-em v-alphabetic v-hanging v-ideographic v-mathematical vector-effect vert-adv-y vert-origin-x vert-origin-y word-spacing writing-mode xmlns:xlink x-height".split(" ").forEach(function(e) {
    var t = e.replace(
      ee,
      xe
    );
    P[t] = new F(t, 1, !1, e, null, !1);
  }), "xlink:actuate xlink:arcrole xlink:role xlink:show xlink:title xlink:type".split(" ").forEach(function(e) {
    var t = e.replace(ee, xe);
    P[t] = new F(t, 1, !1, e, "http://www.w3.org/1999/xlink", !1);
  }), ["xml:base", "xml:lang", "xml:space"].forEach(function(e) {
    var t = e.replace(ee, xe);
    P[t] = new F(t, 1, !1, e, "http://www.w3.org/XML/1998/namespace", !1);
  }), ["tabIndex", "crossOrigin"].forEach(function(e) {
    P[e] = new F(e, 1, !1, e.toLowerCase(), null, !1);
  }), P.xlinkHref = new F("xlinkHref", 1, !1, "xlink:href", "http://www.w3.org/1999/xlink", !0), ["src", "href", "action", "formAction"].forEach(function(e) {
    P[e] = new F(e, 1, !1, e.toLowerCase(), null, !0);
  });
  var Ae = S.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED;
  Ae.hasOwnProperty("ReactCurrentDispatcher") || (Ae.ReactCurrentDispatcher = { current: null }), Ae.hasOwnProperty("ReactCurrentBatchConfig") || (Ae.ReactCurrentBatchConfig = { suspense: null });
  function Gr(e, t, n, r) {
    var l = P.hasOwnProperty(t) ? P[t] : null, i = l !== null ? l.type === 0 : r ? !1 : !(!(2 < t.length) || t[0] !== "o" && t[0] !== "O" || t[1] !== "n" && t[1] !== "N");
    i || (A(t, n, l, r) && (n = null), r || l === null ? C(t) && (n === null ? e.removeAttribute(t) : e.setAttribute(t, "" + n)) : l.mustUseProperty ? e[l.propertyName] = n === null ? l.type === 3 ? !1 : "" : n : (t = l.attributeName, r = l.attributeNamespace, n === null ? e.removeAttribute(t) : (l = l.type, n = l === 3 || l === 4 && n === !0 ? "" : "" + n, r ? e.setAttributeNS(r, t, n) : e.setAttribute(t, n))));
  }
  var Rs = /^(.*)[\\\/]/, me = typeof Symbol == "function" && Symbol.for, Kn = me ? Symbol.for("react.element") : 60103, Yt = me ? Symbol.for("react.portal") : 60106, Mt = me ? Symbol.for("react.fragment") : 60107, Ui = me ? Symbol.for("react.strict_mode") : 60108, Bn = me ? Symbol.for("react.profiler") : 60114, Ai = me ? Symbol.for("react.provider") : 60109, Vi = me ? Symbol.for("react.context") : 60110, Ms = me ? Symbol.for("react.concurrent_mode") : 60111, Zr = me ? Symbol.for("react.forward_ref") : 60112, Yn = me ? Symbol.for("react.suspense") : 60113, Jr = me ? Symbol.for("react.suspense_list") : 60120, qr = me ? Symbol.for("react.memo") : 60115, Wi = me ? Symbol.for("react.lazy") : 60116, Qi = me ? Symbol.for("react.block") : 60121, Hi = typeof Symbol == "function" && Symbol.iterator;
  function cn(e) {
    return e === null || typeof e != "object" ? null : (e = Hi && e[Hi] || e["@@iterator"], typeof e == "function" ? e : null);
  }
  function zs(e) {
    if (e._status === -1) {
      e._status = 0;
      var t = e._ctor;
      t = t(), e._result = t, t.then(function(n) {
        e._status === 0 && (n = n.default, e._status = 1, e._result = n);
      }, function(n) {
        e._status === 0 && (e._status = 2, e._result = n);
      });
    }
  }
  function lt(e) {
    if (e == null)
      return null;
    if (typeof e == "function")
      return e.displayName || e.name || null;
    if (typeof e == "string")
      return e;
    switch (e) {
      case Mt:
        return "Fragment";
      case Yt:
        return "Portal";
      case Bn:
        return "Profiler";
      case Ui:
        return "StrictMode";
      case Yn:
        return "Suspense";
      case Jr:
        return "SuspenseList";
    }
    if (typeof e == "object")
      switch (e.$$typeof) {
        case Vi:
          return "Context.Consumer";
        case Ai:
          return "Context.Provider";
        case Zr:
          var t = e.render;
          return t = t.displayName || t.name || "", e.displayName || (t !== "" ? "ForwardRef(" + t + ")" : "ForwardRef");
        case qr:
          return lt(e.type);
        case Qi:
          return lt(e.render);
        case Wi:
          if (e = e._status === 1 ? e._result : null)
            return lt(e);
      }
    return null;
  }
  function br(e) {
    var t = "";
    do {
      e:
        switch (e.tag) {
          case 3:
          case 4:
          case 6:
          case 7:
          case 10:
          case 9:
            var n = "";
            break e;
          default:
            var r = e._debugOwner, l = e._debugSource, i = lt(e.type);
            n = null, r && (n = lt(r.type)), r = i, i = "", l ? i = " (at " + l.fileName.replace(Rs, "") + ":" + l.lineNumber + ")" : n && (i = " (created by " + n + ")"), n = `
    in ` + (r || "Unknown") + i;
        }
      t += n, e = e.return;
    } while (e);
    return t;
  }
  function dt(e) {
    switch (typeof e) {
      case "boolean":
      case "number":
      case "object":
      case "string":
      case "undefined":
        return e;
      default:
        return "";
    }
  }
  function $i(e) {
    var t = e.type;
    return (e = e.nodeName) && e.toLowerCase() === "input" && (t === "checkbox" || t === "radio");
  }
  function Is(e) {
    var t = $i(e) ? "checked" : "value", n = Object.getOwnPropertyDescriptor(e.constructor.prototype, t), r = "" + e[t];
    if (!e.hasOwnProperty(t) && typeof n < "u" && typeof n.get == "function" && typeof n.set == "function") {
      var l = n.get, i = n.set;
      return Object.defineProperty(e, t, { configurable: !0, get: function() {
        return l.call(this);
      }, set: function(o) {
        r = "" + o, i.call(this, o);
      } }), Object.defineProperty(e, t, { enumerable: n.enumerable }), { getValue: function() {
        return r;
      }, setValue: function(o) {
        r = "" + o;
      }, stopTracking: function() {
        e._valueTracker = null, delete e[t];
      } };
    }
  }
  function Xn(e) {
    e._valueTracker || (e._valueTracker = Is(e));
  }
  function Ki(e) {
    if (!e)
      return !1;
    var t = e._valueTracker;
    if (!t)
      return !0;
    var n = t.getValue(), r = "";
    return e && (r = $i(e) ? e.checked ? "true" : "false" : e.value), e = r, e !== n ? (t.setValue(e), !0) : !1;
  }
  function el(e, t) {
    var n = t.checked;
    return O({}, t, { defaultChecked: void 0, defaultValue: void 0, value: void 0, checked: n ?? e._wrapperState.initialChecked });
  }
  function Bi(e, t) {
    var n = t.defaultValue == null ? "" : t.defaultValue, r = t.checked != null ? t.checked : t.defaultChecked;
    n = dt(t.value != null ? t.value : n), e._wrapperState = { initialChecked: r, initialValue: n, controlled: t.type === "checkbox" || t.type === "radio" ? t.checked != null : t.value != null };
  }
  function Yi(e, t) {
    t = t.checked, t != null && Gr(e, "checked", t, !1);
  }
  function tl(e, t) {
    Yi(e, t);
    var n = dt(t.value), r = t.type;
    if (n != null)
      r === "number" ? (n === 0 && e.value === "" || e.value != n) && (e.value = "" + n) : e.value !== "" + n && (e.value = "" + n);
    else if (r === "submit" || r === "reset") {
      e.removeAttribute("value");
      return;
    }
    t.hasOwnProperty("value") ? nl(e, t.type, n) : t.hasOwnProperty("defaultValue") && nl(e, t.type, dt(t.defaultValue)), t.checked == null && t.defaultChecked != null && (e.defaultChecked = !!t.defaultChecked);
  }
  function Xi(e, t, n) {
    if (t.hasOwnProperty("value") || t.hasOwnProperty("defaultValue")) {
      var r = t.type;
      if (!(r !== "submit" && r !== "reset" || t.value !== void 0 && t.value !== null))
        return;
      t = "" + e._wrapperState.initialValue, n || t === e.value || (e.value = t), e.defaultValue = t;
    }
    n = e.name, n !== "" && (e.name = ""), e.defaultChecked = !!e._wrapperState.initialChecked, n !== "" && (e.name = n);
  }
  function nl(e, t, n) {
    (t !== "number" || e.ownerDocument.activeElement !== e) && (n == null ? e.defaultValue = "" + e._wrapperState.initialValue : e.defaultValue !== "" + n && (e.defaultValue = "" + n));
  }
  function js(e) {
    var t = "";
    return S.Children.forEach(e, function(n) {
      n != null && (t += n);
    }), t;
  }
  function rl(e, t) {
    return e = O({ children: void 0 }, t), (t = js(t.children)) && (e.children = t), e;
  }
  function Xt(e, t, n, r) {
    if (e = e.options, t) {
      t = {};
      for (var l = 0; l < n.length; l++)
        t["$" + n[l]] = !0;
      for (n = 0; n < e.length; n++)
        l = t.hasOwnProperty("$" + e[n].value), e[n].selected !== l && (e[n].selected = l), l && r && (e[n].defaultSelected = !0);
    } else {
      for (n = "" + dt(n), t = null, l = 0; l < e.length; l++) {
        if (e[l].value === n) {
          e[l].selected = !0, r && (e[l].defaultSelected = !0);
          return;
        }
        t !== null || e[l].disabled || (t = e[l]);
      }
      t !== null && (t.selected = !0);
    }
  }
  function ll(e, t) {
    if (t.dangerouslySetInnerHTML != null)
      throw Error(h(91));
    return O({}, t, { value: void 0, defaultValue: void 0, children: "" + e._wrapperState.initialValue });
  }
  function Gi(e, t) {
    var n = t.value;
    if (n == null) {
      if (n = t.children, t = t.defaultValue, n != null) {
        if (t != null)
          throw Error(h(92));
        if (Array.isArray(n)) {
          if (!(1 >= n.length))
            throw Error(h(93));
          n = n[0];
        }
        t = n;
      }
      t == null && (t = ""), n = t;
    }
    e._wrapperState = { initialValue: dt(n) };
  }
  function Zi(e, t) {
    var n = dt(t.value), r = dt(t.defaultValue);
    n != null && (n = "" + n, n !== e.value && (e.value = n), t.defaultValue == null && e.defaultValue !== n && (e.defaultValue = n)), r != null && (e.defaultValue = "" + r);
  }
  function Ji(e) {
    var t = e.textContent;
    t === e._wrapperState.initialValue && t !== "" && t !== null && (e.value = t);
  }
  var qi = { html: "http://www.w3.org/1999/xhtml", mathml: "http://www.w3.org/1998/Math/MathML", svg: "http://www.w3.org/2000/svg" };
  function bi(e) {
    switch (e) {
      case "svg":
        return "http://www.w3.org/2000/svg";
      case "math":
        return "http://www.w3.org/1998/Math/MathML";
      default:
        return "http://www.w3.org/1999/xhtml";
    }
  }
  function il(e, t) {
    return e == null || e === "http://www.w3.org/1999/xhtml" ? bi(t) : e === "http://www.w3.org/2000/svg" && t === "foreignObject" ? "http://www.w3.org/1999/xhtml" : e;
  }
  var Gn, eo = function(e) {
    return typeof MSApp < "u" && MSApp.execUnsafeLocalFunction ? function(t, n, r, l) {
      MSApp.execUnsafeLocalFunction(function() {
        return e(t, n, r, l);
      });
    } : e;
  }(function(e, t) {
    if (e.namespaceURI !== qi.svg || "innerHTML" in e)
      e.innerHTML = t;
    else {
      for (Gn = Gn || document.createElement("div"), Gn.innerHTML = "<svg>" + t.valueOf().toString() + "</svg>", t = Gn.firstChild; e.firstChild; )
        e.removeChild(e.firstChild);
      for (; t.firstChild; )
        e.appendChild(t.firstChild);
    }
  });
  function dn(e, t) {
    if (t) {
      var n = e.firstChild;
      if (n && n === e.lastChild && n.nodeType === 3) {
        n.nodeValue = t;
        return;
      }
    }
    e.textContent = t;
  }
  function Zn(e, t) {
    var n = {};
    return n[e.toLowerCase()] = t.toLowerCase(), n["Webkit" + e] = "webkit" + t, n["Moz" + e] = "moz" + t, n;
  }
  var Gt = { animationend: Zn("Animation", "AnimationEnd"), animationiteration: Zn("Animation", "AnimationIteration"), animationstart: Zn("Animation", "AnimationStart"), transitionend: Zn("Transition", "TransitionEnd") }, ol = {}, to = {};
  ae && (to = document.createElement("div").style, "AnimationEvent" in window || (delete Gt.animationend.animation, delete Gt.animationiteration.animation, delete Gt.animationstart.animation), "TransitionEvent" in window || delete Gt.transitionend.transition);
  function Jn(e) {
    if (ol[e])
      return ol[e];
    if (!Gt[e])
      return e;
    var t = Gt[e], n;
    for (n in t)
      if (t.hasOwnProperty(n) && n in to)
        return ol[e] = t[n];
    return e;
  }
  var no = Jn("animationend"), ro = Jn("animationiteration"), lo = Jn("animationstart"), io = Jn("transitionend"), pn = "abort canplay canplaythrough durationchange emptied encrypted ended error loadeddata loadedmetadata loadstart pause play playing progress ratechange seeked seeking stalled suspend timeupdate volumechange waiting".split(" "), oo = new (typeof WeakMap == "function" ? WeakMap : Map)();
  function ul(e) {
    var t = oo.get(e);
    return t === void 0 && (t = /* @__PURE__ */ new Map(), oo.set(e, t)), t;
  }
  function zt(e) {
    var t = e, n = e;
    if (e.alternate)
      for (; t.return; )
        t = t.return;
    else {
      e = t;
      do
        t = e, t.effectTag & 1026 && (n = t.return), e = t.return;
      while (e);
    }
    return t.tag === 3 ? n : null;
  }
  function uo(e) {
    if (e.tag === 13) {
      var t = e.memoizedState;
      if (t === null && (e = e.alternate, e !== null && (t = e.memoizedState)), t !== null)
        return t.dehydrated;
    }
    return null;
  }
  function so(e) {
    if (zt(e) !== e)
      throw Error(h(188));
  }
  function Fs(e) {
    var t = e.alternate;
    if (!t) {
      if (t = zt(e), t === null)
        throw Error(h(188));
      return t !== e ? null : e;
    }
    for (var n = e, r = t; ; ) {
      var l = n.return;
      if (l === null)
        break;
      var i = l.alternate;
      if (i === null) {
        if (r = l.return, r !== null) {
          n = r;
          continue;
        }
        break;
      }
      if (l.child === i.child) {
        for (i = l.child; i; ) {
          if (i === n)
            return so(l), e;
          if (i === r)
            return so(l), t;
          i = i.sibling;
        }
        throw Error(h(188));
      }
      if (n.return !== r.return)
        n = l, r = i;
      else {
        for (var o = !1, u = l.child; u; ) {
          if (u === n) {
            o = !0, n = l, r = i;
            break;
          }
          if (u === r) {
            o = !0, r = l, n = i;
            break;
          }
          u = u.sibling;
        }
        if (!o) {
          for (u = i.child; u; ) {
            if (u === n) {
              o = !0, n = i, r = l;
              break;
            }
            if (u === r) {
              o = !0, r = i, n = l;
              break;
            }
            u = u.sibling;
          }
          if (!o)
            throw Error(h(189));
        }
      }
      if (n.alternate !== r)
        throw Error(h(190));
    }
    if (n.tag !== 3)
      throw Error(h(188));
    return n.stateNode.current === n ? e : t;
  }
  function ao(e) {
    if (e = Fs(e), !e)
      return null;
    for (var t = e; ; ) {
      if (t.tag === 5 || t.tag === 6)
        return t;
      if (t.child)
        t.child.return = t, t = t.child;
      else {
        if (t === e)
          break;
        for (; !t.sibling; ) {
          if (!t.return || t.return === e)
            return null;
          t = t.return;
        }
        t.sibling.return = t.return, t = t.sibling;
      }
    }
    return null;
  }
  function Zt(e, t) {
    if (t == null)
      throw Error(h(30));
    return e == null ? t : Array.isArray(e) ? Array.isArray(t) ? (e.push.apply(e, t), e) : (e.push(t), e) : Array.isArray(t) ? [e].concat(t) : [e, t];
  }
  function sl(e, t, n) {
    Array.isArray(e) ? e.forEach(t, n) : e && t.call(n, e);
  }
  var mn = null;
  function Ls(e) {
    if (e) {
      var t = e._dispatchListeners, n = e._dispatchInstances;
      if (Array.isArray(t))
        for (var r = 0; r < t.length && !e.isPropagationStopped(); r++)
          Rt(e, t[r], n[r]);
      else
        t && Rt(e, t, n);
      e._dispatchListeners = null, e._dispatchInstances = null, e.isPersistent() || e.constructor.release(e);
    }
  }
  function qn(e) {
    if (e !== null && (mn = Zt(mn, e)), e = mn, mn = null, e) {
      if (sl(e, Ls), mn)
        throw Error(h(95));
      if (H)
        throw e = fe, H = !1, fe = null, e;
    }
  }
  function al(e) {
    return e = e.target || e.srcElement || window, e.correspondingUseElement && (e = e.correspondingUseElement), e.nodeType === 3 ? e.parentNode : e;
  }
  function fo(e) {
    if (!ae)
      return !1;
    e = "on" + e;
    var t = e in document;
    return t || (t = document.createElement("div"), t.setAttribute(e, "return;"), t = typeof t[e] == "function"), t;
  }
  var bn = [];
  function co(e) {
    e.topLevelType = null, e.nativeEvent = null, e.targetInst = null, e.ancestors.length = 0, 10 > bn.length && bn.push(e);
  }
  function po(e, t, n, r) {
    if (bn.length) {
      var l = bn.pop();
      return l.topLevelType = e, l.eventSystemFlags = r, l.nativeEvent = t, l.targetInst = n, l;
    }
    return { topLevelType: e, eventSystemFlags: r, nativeEvent: t, targetInst: n, ancestors: [] };
  }
  function mo(e) {
    var t = e.targetInst, n = t;
    do {
      if (!n) {
        e.ancestors.push(n);
        break;
      }
      var r = n;
      if (r.tag === 3)
        r = r.stateNode.containerInfo;
      else {
        for (; r.return; )
          r = r.return;
        r = r.tag !== 3 ? null : r.stateNode.containerInfo;
      }
      if (!r)
        break;
      t = n.tag, t !== 5 && t !== 6 || e.ancestors.push(n), n = xn(r);
    } while (n);
    for (n = 0; n < e.ancestors.length; n++) {
      t = e.ancestors[n];
      var l = al(e.nativeEvent);
      r = e.topLevelType;
      var i = e.nativeEvent, o = e.eventSystemFlags;
      n === 0 && (o |= 64);
      for (var u = null, c = 0; c < De.length; c++) {
        var d = De[c];
        d && (d = d.extractEvents(r, t, i, l, o)) && (u = Zt(u, d));
      }
      qn(u);
    }
  }
  function fl(e, t, n) {
    if (!n.has(e)) {
      switch (e) {
        case "scroll":
          En(t, "scroll", !0);
          break;
        case "focus":
        case "blur":
          En(t, "focus", !0), En(t, "blur", !0), n.set("blur", null), n.set("focus", null);
          break;
        case "cancel":
        case "close":
          fo(e) && En(t, e, !0);
          break;
        case "invalid":
        case "submit":
        case "reset":
          break;
        default:
          pn.indexOf(e) === -1 && W(e, t);
      }
      n.set(e, null);
    }
  }
  var ho, cl, vo, dl = !1, Be = [], pt = null, mt = null, ht = null, hn = /* @__PURE__ */ new Map(), vn = /* @__PURE__ */ new Map(), yn = [], pl = "mousedown mouseup touchcancel touchend touchstart auxclick dblclick pointercancel pointerdown pointerup dragend dragstart drop compositionend compositionstart keydown keypress keyup input textInput close cancel copy cut paste click change contextmenu reset submit".split(" "), Ds = "focus blur dragenter dragleave mouseover mouseout pointerover pointerout gotpointercapture lostpointercapture".split(" ");
  function Us(e, t) {
    var n = ul(t);
    pl.forEach(function(r) {
      fl(r, t, n);
    }), Ds.forEach(function(r) {
      fl(r, t, n);
    });
  }
  function ml(e, t, n, r, l) {
    return { blockedOn: e, topLevelType: t, eventSystemFlags: n | 32, nativeEvent: l, container: r };
  }
  function yo(e, t) {
    switch (e) {
      case "focus":
      case "blur":
        pt = null;
        break;
      case "dragenter":
      case "dragleave":
        mt = null;
        break;
      case "mouseover":
      case "mouseout":
        ht = null;
        break;
      case "pointerover":
      case "pointerout":
        hn.delete(t.pointerId);
        break;
      case "gotpointercapture":
      case "lostpointercapture":
        vn.delete(t.pointerId);
    }
  }
  function gn(e, t, n, r, l, i) {
    return e === null || e.nativeEvent !== i ? (e = ml(t, n, r, l, i), t !== null && (t = Sn(t), t !== null && cl(t)), e) : (e.eventSystemFlags |= r, e);
  }
  function As(e, t, n, r, l) {
    switch (t) {
      case "focus":
        return pt = gn(pt, e, t, n, r, l), !0;
      case "dragenter":
        return mt = gn(mt, e, t, n, r, l), !0;
      case "mouseover":
        return ht = gn(ht, e, t, n, r, l), !0;
      case "pointerover":
        var i = l.pointerId;
        return hn.set(i, gn(hn.get(i) || null, e, t, n, r, l)), !0;
      case "gotpointercapture":
        return i = l.pointerId, vn.set(i, gn(vn.get(i) || null, e, t, n, r, l)), !0;
    }
    return !1;
  }
  function Vs(e) {
    var t = xn(e.target);
    if (t !== null) {
      var n = zt(t);
      if (n !== null) {
        if (t = n.tag, t === 13) {
          if (t = uo(n), t !== null) {
            e.blockedOn = t, I.unstable_runWithPriority(e.priority, function() {
              vo(n);
            });
            return;
          }
        } else if (t === 3 && n.stateNode.hydrate) {
          e.blockedOn = n.tag === 3 ? n.stateNode.containerInfo : null;
          return;
        }
      }
    }
    e.blockedOn = null;
  }
  function er(e) {
    if (e.blockedOn !== null)
      return !1;
    var t = gl(e.topLevelType, e.eventSystemFlags, e.container, e.nativeEvent);
    if (t !== null) {
      var n = Sn(t);
      return n !== null && cl(n), e.blockedOn = t, !1;
    }
    return !0;
  }
  function go(e, t, n) {
    er(e) && n.delete(t);
  }
  function Ws() {
    for (dl = !1; 0 < Be.length; ) {
      var e = Be[0];
      if (e.blockedOn !== null) {
        e = Sn(e.blockedOn), e !== null && ho(e);
        break;
      }
      var t = gl(e.topLevelType, e.eventSystemFlags, e.container, e.nativeEvent);
      t !== null ? e.blockedOn = t : Be.shift();
    }
    pt !== null && er(pt) && (pt = null), mt !== null && er(mt) && (mt = null), ht !== null && er(ht) && (ht = null), hn.forEach(go), vn.forEach(go);
  }
  function wn(e, t) {
    e.blockedOn === t && (e.blockedOn = null, dl || (dl = !0, I.unstable_scheduleCallback(I.unstable_NormalPriority, Ws)));
  }
  function wo(e) {
    function t(l) {
      return wn(l, e);
    }
    if (0 < Be.length) {
      wn(Be[0], e);
      for (var n = 1; n < Be.length; n++) {
        var r = Be[n];
        r.blockedOn === e && (r.blockedOn = null);
      }
    }
    for (pt !== null && wn(pt, e), mt !== null && wn(mt, e), ht !== null && wn(ht, e), hn.forEach(t), vn.forEach(t), n = 0; n < yn.length; n++)
      r = yn[n], r.blockedOn === e && (r.blockedOn = null);
    for (; 0 < yn.length && (n = yn[0], n.blockedOn === null); )
      Vs(n), n.blockedOn === null && yn.shift();
  }
  var Eo = {}, To = /* @__PURE__ */ new Map(), hl = /* @__PURE__ */ new Map(), Qs = [
    "abort",
    "abort",
    no,
    "animationEnd",
    ro,
    "animationIteration",
    lo,
    "animationStart",
    "canplay",
    "canPlay",
    "canplaythrough",
    "canPlayThrough",
    "durationchange",
    "durationChange",
    "emptied",
    "emptied",
    "encrypted",
    "encrypted",
    "ended",
    "ended",
    "error",
    "error",
    "gotpointercapture",
    "gotPointerCapture",
    "load",
    "load",
    "loadeddata",
    "loadedData",
    "loadedmetadata",
    "loadedMetadata",
    "loadstart",
    "loadStart",
    "lostpointercapture",
    "lostPointerCapture",
    "playing",
    "playing",
    "progress",
    "progress",
    "seeking",
    "seeking",
    "stalled",
    "stalled",
    "suspend",
    "suspend",
    "timeupdate",
    "timeUpdate",
    io,
    "transitionEnd",
    "waiting",
    "waiting"
  ];
  function vl(e, t) {
    for (var n = 0; n < e.length; n += 2) {
      var r = e[n], l = e[n + 1], i = "on" + (l[0].toUpperCase() + l.slice(1));
      i = { phasedRegistrationNames: { bubbled: i, captured: i + "Capture" }, dependencies: [r], eventPriority: t }, hl.set(r, t), To.set(r, i), Eo[l] = i;
    }
  }
  vl("blur blur cancel cancel click click close close contextmenu contextMenu copy copy cut cut auxclick auxClick dblclick doubleClick dragend dragEnd dragstart dragStart drop drop focus focus input input invalid invalid keydown keyDown keypress keyPress keyup keyUp mousedown mouseDown mouseup mouseUp paste paste pause pause play play pointercancel pointerCancel pointerdown pointerDown pointerup pointerUp ratechange rateChange reset reset seeked seeked submit submit touchcancel touchCancel touchend touchEnd touchstart touchStart volumechange volumeChange".split(" "), 0), vl("drag drag dragenter dragEnter dragexit dragExit dragleave dragLeave dragover dragOver mousemove mouseMove mouseout mouseOut mouseover mouseOver pointermove pointerMove pointerout pointerOut pointerover pointerOver scroll scroll toggle toggle touchmove touchMove wheel wheel".split(" "), 1), vl(Qs, 2);
  for (var ko = "change selectionchange textInput compositionstart compositionend compositionupdate".split(" "), yl = 0; yl < ko.length; yl++)
    hl.set(ko[yl], 0);
  var Hs = I.unstable_UserBlockingPriority, $s = I.unstable_runWithPriority, tr = !0;
  function W(e, t) {
    En(t, e, !1);
  }
  function En(e, t, n) {
    var r = hl.get(t);
    switch (r === void 0 ? 2 : r) {
      case 0:
        r = Ks.bind(null, t, 1, e);
        break;
      case 1:
        r = Bs.bind(null, t, 1, e);
        break;
      default:
        r = nr.bind(null, t, 1, e);
    }
    n ? e.addEventListener(t, r, !0) : e.addEventListener(t, r, !1);
  }
  function Ks(e, t, n, r) {
    ke || Te();
    var l = nr, i = ke;
    ke = !0;
    try {
      de(l, e, t, n, r);
    } finally {
      (ke = i) || rt();
    }
  }
  function Bs(e, t, n, r) {
    $s(Hs, nr.bind(null, e, t, n, r));
  }
  function nr(e, t, n, r) {
    if (tr)
      if (0 < Be.length && -1 < pl.indexOf(e))
        e = ml(null, e, t, n, r), Be.push(e);
      else {
        var l = gl(e, t, n, r);
        if (l === null)
          yo(e, r);
        else if (-1 < pl.indexOf(e))
          e = ml(l, e, t, n, r), Be.push(e);
        else if (!As(l, e, t, n, r)) {
          yo(e, r), e = po(e, r, null, t);
          try {
            pe(mo, e);
          } finally {
            co(e);
          }
        }
      }
  }
  function gl(e, t, n, r) {
    if (n = al(r), n = xn(n), n !== null) {
      var l = zt(n);
      if (l === null)
        n = null;
      else {
        var i = l.tag;
        if (i === 13) {
          if (n = uo(l), n !== null)
            return n;
          n = null;
        } else if (i === 3) {
          if (l.stateNode.hydrate)
            return l.tag === 3 ? l.stateNode.containerInfo : null;
          n = null;
        } else
          l !== n && (n = null);
      }
    }
    e = po(e, r, n, t);
    try {
      pe(mo, e);
    } finally {
      co(e);
    }
    return null;
  }
  var Tn = {
    animationIterationCount: !0,
    borderImageOutset: !0,
    borderImageSlice: !0,
    borderImageWidth: !0,
    boxFlex: !0,
    boxFlexGroup: !0,
    boxOrdinalGroup: !0,
    columnCount: !0,
    columns: !0,
    flex: !0,
    flexGrow: !0,
    flexPositive: !0,
    flexShrink: !0,
    flexNegative: !0,
    flexOrder: !0,
    gridArea: !0,
    gridRow: !0,
    gridRowEnd: !0,
    gridRowSpan: !0,
    gridRowStart: !0,
    gridColumn: !0,
    gridColumnEnd: !0,
    gridColumnSpan: !0,
    gridColumnStart: !0,
    fontWeight: !0,
    lineClamp: !0,
    lineHeight: !0,
    opacity: !0,
    order: !0,
    orphans: !0,
    tabSize: !0,
    widows: !0,
    zIndex: !0,
    zoom: !0,
    fillOpacity: !0,
    floodOpacity: !0,
    stopOpacity: !0,
    strokeDasharray: !0,
    strokeDashoffset: !0,
    strokeMiterlimit: !0,
    strokeOpacity: !0,
    strokeWidth: !0
  }, Ys = ["Webkit", "ms", "Moz", "O"];
  Object.keys(Tn).forEach(function(e) {
    Ys.forEach(function(t) {
      t = t + e.charAt(0).toUpperCase() + e.substring(1), Tn[t] = Tn[e];
    });
  });
  function xo(e, t, n) {
    return t == null || typeof t == "boolean" || t === "" ? "" : n || typeof t != "number" || t === 0 || Tn.hasOwnProperty(e) && Tn[e] ? ("" + t).trim() : t + "px";
  }
  function So(e, t) {
    e = e.style;
    for (var n in t)
      if (t.hasOwnProperty(n)) {
        var r = n.indexOf("--") === 0, l = xo(n, t[n], r);
        n === "float" && (n = "cssFloat"), r ? e.setProperty(n, l) : e[n] = l;
      }
  }
  var Xs = O({ menuitem: !0 }, { area: !0, base: !0, br: !0, col: !0, embed: !0, hr: !0, img: !0, input: !0, keygen: !0, link: !0, meta: !0, param: !0, source: !0, track: !0, wbr: !0 });
  function wl(e, t) {
    if (t) {
      if (Xs[e] && (t.children != null || t.dangerouslySetInnerHTML != null))
        throw Error(h(137, e, ""));
      if (t.dangerouslySetInnerHTML != null) {
        if (t.children != null)
          throw Error(h(60));
        if (!(typeof t.dangerouslySetInnerHTML == "object" && "__html" in t.dangerouslySetInnerHTML))
          throw Error(h(61));
      }
      if (t.style != null && typeof t.style != "object")
        throw Error(h(62, ""));
    }
  }
  function El(e, t) {
    if (e.indexOf("-") === -1)
      return typeof t.is == "string";
    switch (e) {
      case "annotation-xml":
      case "color-profile":
      case "font-face":
      case "font-face-src":
      case "font-face-uri":
      case "font-face-format":
      case "font-face-name":
      case "missing-glyph":
        return !1;
      default:
        return !0;
    }
  }
  var Co = qi.html;
  function it(e, t) {
    e = e.nodeType === 9 || e.nodeType === 11 ? e : e.ownerDocument;
    var n = ul(e);
    t = Ke[t];
    for (var r = 0; r < t.length; r++)
      fl(t[r], e, n);
  }
  function rr() {
  }
  function Tl(e) {
    if (e = e || (typeof document < "u" ? document : void 0), typeof e > "u")
      return null;
    try {
      return e.activeElement || e.body;
    } catch {
      return e.body;
    }
  }
  function _o(e) {
    for (; e && e.firstChild; )
      e = e.firstChild;
    return e;
  }
  function Po(e, t) {
    var n = _o(e);
    e = 0;
    for (var r; n; ) {
      if (n.nodeType === 3) {
        if (r = e + n.textContent.length, e <= t && r >= t)
          return { node: n, offset: t - e };
        e = r;
      }
      e: {
        for (; n; ) {
          if (n.nextSibling) {
            n = n.nextSibling;
            break e;
          }
          n = n.parentNode;
        }
        n = void 0;
      }
      n = _o(n);
    }
  }
  function No(e, t) {
    return e && t ? e === t ? !0 : e && e.nodeType === 3 ? !1 : t && t.nodeType === 3 ? No(e, t.parentNode) : "contains" in e ? e.contains(t) : e.compareDocumentPosition ? !!(e.compareDocumentPosition(t) & 16) : !1 : !1;
  }
  function Oo() {
    for (var e = window, t = Tl(); t instanceof e.HTMLIFrameElement; ) {
      try {
        var n = typeof t.contentWindow.location.href == "string";
      } catch {
        n = !1;
      }
      if (n)
        e = t.contentWindow;
      else
        break;
      t = Tl(e.document);
    }
    return t;
  }
  function kl(e) {
    var t = e && e.nodeName && e.nodeName.toLowerCase();
    return t && (t === "input" && (e.type === "text" || e.type === "search" || e.type === "tel" || e.type === "url" || e.type === "password") || t === "textarea" || e.contentEditable === "true");
  }
  var Ro = "$", Mo = "/$", xl = "$?", Sl = "$!", Cl = null, _l = null;
  function zo(e, t) {
    switch (e) {
      case "button":
      case "input":
      case "select":
      case "textarea":
        return !!t.autoFocus;
    }
    return !1;
  }
  function Pl(e, t) {
    return e === "textarea" || e === "option" || e === "noscript" || typeof t.children == "string" || typeof t.children == "number" || typeof t.dangerouslySetInnerHTML == "object" && t.dangerouslySetInnerHTML !== null && t.dangerouslySetInnerHTML.__html != null;
  }
  var Nl = typeof setTimeout == "function" ? setTimeout : void 0, Gs = typeof clearTimeout == "function" ? clearTimeout : void 0;
  function Jt(e) {
    for (; e != null; e = e.nextSibling) {
      var t = e.nodeType;
      if (t === 1 || t === 3)
        break;
    }
    return e;
  }
  function Io(e) {
    e = e.previousSibling;
    for (var t = 0; e; ) {
      if (e.nodeType === 8) {
        var n = e.data;
        if (n === Ro || n === Sl || n === xl) {
          if (t === 0)
            return e;
          t--;
        } else
          n === Mo && t++;
      }
      e = e.previousSibling;
    }
    return null;
  }
  var Ol = Math.random().toString(36).slice(2), vt = "__reactInternalInstance$" + Ol, lr = "__reactEventHandlers$" + Ol, kn = "__reactContainere$" + Ol;
  function xn(e) {
    var t = e[vt];
    if (t)
      return t;
    for (var n = e.parentNode; n; ) {
      if (t = n[kn] || n[vt]) {
        if (n = t.alternate, t.child !== null || n !== null && n.child !== null)
          for (e = Io(e); e !== null; ) {
            if (n = e[vt])
              return n;
            e = Io(e);
          }
        return t;
      }
      e = n, n = e.parentNode;
    }
    return null;
  }
  function Sn(e) {
    return e = e[vt] || e[kn], !e || e.tag !== 5 && e.tag !== 6 && e.tag !== 13 && e.tag !== 3 ? null : e;
  }
  function It(e) {
    if (e.tag === 5 || e.tag === 6)
      return e.stateNode;
    throw Error(h(33));
  }
  function Rl(e) {
    return e[lr] || null;
  }
  function ot(e) {
    do
      e = e.return;
    while (e && e.tag !== 5);
    return e || null;
  }
  function jo(e, t) {
    var n = e.stateNode;
    if (!n)
      return null;
    var r = we(n);
    if (!r)
      return null;
    n = r[t];
    e:
      switch (t) {
        case "onClick":
        case "onClickCapture":
        case "onDoubleClick":
        case "onDoubleClickCapture":
        case "onMouseDown":
        case "onMouseDownCapture":
        case "onMouseMove":
        case "onMouseMoveCapture":
        case "onMouseUp":
        case "onMouseUpCapture":
        case "onMouseEnter":
          (r = !r.disabled) || (e = e.type, r = !(e === "button" || e === "input" || e === "select" || e === "textarea")), e = !r;
          break e;
        default:
          e = !1;
      }
    if (e)
      return null;
    if (n && typeof n != "function")
      throw Error(h(
        231,
        t,
        typeof n
      ));
    return n;
  }
  function Fo(e, t, n) {
    (t = jo(e, n.dispatchConfig.phasedRegistrationNames[t])) && (n._dispatchListeners = Zt(n._dispatchListeners, t), n._dispatchInstances = Zt(n._dispatchInstances, e));
  }
  function Zs(e) {
    if (e && e.dispatchConfig.phasedRegistrationNames) {
      for (var t = e._targetInst, n = []; t; )
        n.push(t), t = ot(t);
      for (t = n.length; 0 < t--; )
        Fo(n[t], "captured", e);
      for (t = 0; t < n.length; t++)
        Fo(n[t], "bubbled", e);
    }
  }
  function Ml(e, t, n) {
    e && n && n.dispatchConfig.registrationName && (t = jo(e, n.dispatchConfig.registrationName)) && (n._dispatchListeners = Zt(n._dispatchListeners, t), n._dispatchInstances = Zt(n._dispatchInstances, e));
  }
  function Js(e) {
    e && e.dispatchConfig.registrationName && Ml(e._targetInst, null, e);
  }
  function qt(e) {
    sl(e, Zs);
  }
  var yt = null, zl = null, ir = null;
  function Lo() {
    if (ir)
      return ir;
    var e, t = zl, n = t.length, r, l = "value" in yt ? yt.value : yt.textContent, i = l.length;
    for (e = 0; e < n && t[e] === l[e]; e++)
      ;
    var o = n - e;
    for (r = 1; r <= o && t[n - r] === l[i - r]; r++)
      ;
    return ir = l.slice(e, 1 < r ? 1 - r : void 0);
  }
  function or() {
    return !0;
  }
  function ur() {
    return !1;
  }
  function Se(e, t, n, r) {
    this.dispatchConfig = e, this._targetInst = t, this.nativeEvent = n, e = this.constructor.Interface;
    for (var l in e)
      e.hasOwnProperty(l) && ((t = e[l]) ? this[l] = t(n) : l === "target" ? this.target = r : this[l] = n[l]);
    return this.isDefaultPrevented = (n.defaultPrevented != null ? n.defaultPrevented : n.returnValue === !1) ? or : ur, this.isPropagationStopped = ur, this;
  }
  O(Se.prototype, { preventDefault: function() {
    this.defaultPrevented = !0;
    var e = this.nativeEvent;
    e && (e.preventDefault ? e.preventDefault() : typeof e.returnValue != "unknown" && (e.returnValue = !1), this.isDefaultPrevented = or);
  }, stopPropagation: function() {
    var e = this.nativeEvent;
    e && (e.stopPropagation ? e.stopPropagation() : typeof e.cancelBubble != "unknown" && (e.cancelBubble = !0), this.isPropagationStopped = or);
  }, persist: function() {
    this.isPersistent = or;
  }, isPersistent: ur, destructor: function() {
    var e = this.constructor.Interface, t;
    for (t in e)
      this[t] = null;
    this.nativeEvent = this._targetInst = this.dispatchConfig = null, this.isPropagationStopped = this.isDefaultPrevented = ur, this._dispatchInstances = this._dispatchListeners = null;
  } }), Se.Interface = { type: null, target: null, currentTarget: function() {
    return null;
  }, eventPhase: null, bubbles: null, cancelable: null, timeStamp: function(e) {
    return e.timeStamp || Date.now();
  }, defaultPrevented: null, isTrusted: null }, Se.extend = function(e) {
    function t() {
    }
    function n() {
      return r.apply(this, arguments);
    }
    var r = this;
    t.prototype = r.prototype;
    var l = new t();
    return O(l, n.prototype), n.prototype = l, n.prototype.constructor = n, n.Interface = O({}, r.Interface, e), n.extend = r.extend, Do(n), n;
  }, Do(Se);
  function qs(e, t, n, r) {
    if (this.eventPool.length) {
      var l = this.eventPool.pop();
      return this.call(l, e, t, n, r), l;
    }
    return new this(e, t, n, r);
  }
  function bs(e) {
    if (!(e instanceof this))
      throw Error(h(279));
    e.destructor(), 10 > this.eventPool.length && this.eventPool.push(e);
  }
  function Do(e) {
    e.eventPool = [], e.getPooled = qs, e.release = bs;
  }
  var ea = Se.extend({ data: null }), ta = Se.extend({ data: null }), na = [9, 13, 27, 32], Il = ae && "CompositionEvent" in window, Cn = null;
  ae && "documentMode" in document && (Cn = document.documentMode);
  var ra = ae && "TextEvent" in window && !Cn, Uo = ae && (!Il || Cn && 8 < Cn && 11 >= Cn), Ao = " ", ut = { beforeInput: { phasedRegistrationNames: { bubbled: "onBeforeInput", captured: "onBeforeInputCapture" }, dependencies: ["compositionend", "keypress", "textInput", "paste"] }, compositionEnd: { phasedRegistrationNames: { bubbled: "onCompositionEnd", captured: "onCompositionEndCapture" }, dependencies: "blur compositionend keydown keypress keyup mousedown".split(" ") }, compositionStart: { phasedRegistrationNames: {
    bubbled: "onCompositionStart",
    captured: "onCompositionStartCapture"
  }, dependencies: "blur compositionstart keydown keypress keyup mousedown".split(" ") }, compositionUpdate: { phasedRegistrationNames: { bubbled: "onCompositionUpdate", captured: "onCompositionUpdateCapture" }, dependencies: "blur compositionupdate keydown keypress keyup mousedown".split(" ") } }, Vo = !1;
  function Wo(e, t) {
    switch (e) {
      case "keyup":
        return na.indexOf(t.keyCode) !== -1;
      case "keydown":
        return t.keyCode !== 229;
      case "keypress":
      case "mousedown":
      case "blur":
        return !0;
      default:
        return !1;
    }
  }
  function Qo(e) {
    return e = e.detail, typeof e == "object" && "data" in e ? e.data : null;
  }
  var bt = !1;
  function la(e, t) {
    switch (e) {
      case "compositionend":
        return Qo(t);
      case "keypress":
        return t.which !== 32 ? null : (Vo = !0, Ao);
      case "textInput":
        return e = t.data, e === Ao && Vo ? null : e;
      default:
        return null;
    }
  }
  function ia(e, t) {
    if (bt)
      return e === "compositionend" || !Il && Wo(e, t) ? (e = Lo(), ir = zl = yt = null, bt = !1, e) : null;
    switch (e) {
      case "paste":
        return null;
      case "keypress":
        if (!(t.ctrlKey || t.altKey || t.metaKey) || t.ctrlKey && t.altKey) {
          if (t.char && 1 < t.char.length)
            return t.char;
          if (t.which)
            return String.fromCharCode(t.which);
        }
        return null;
      case "compositionend":
        return Uo && t.locale !== "ko" ? null : t.data;
      default:
        return null;
    }
  }
  var oa = { eventTypes: ut, extractEvents: function(e, t, n, r) {
    var l;
    if (Il)
      e: {
        switch (e) {
          case "compositionstart":
            var i = ut.compositionStart;
            break e;
          case "compositionend":
            i = ut.compositionEnd;
            break e;
          case "compositionupdate":
            i = ut.compositionUpdate;
            break e;
        }
        i = void 0;
      }
    else
      bt ? Wo(e, n) && (i = ut.compositionEnd) : e === "keydown" && n.keyCode === 229 && (i = ut.compositionStart);
    return i ? (Uo && n.locale !== "ko" && (bt || i !== ut.compositionStart ? i === ut.compositionEnd && bt && (l = Lo()) : (yt = r, zl = "value" in yt ? yt.value : yt.textContent, bt = !0)), i = ea.getPooled(
      i,
      t,
      n,
      r
    ), l ? i.data = l : (l = Qo(n), l !== null && (i.data = l)), qt(i), l = i) : l = null, (e = ra ? la(e, n) : ia(e, n)) ? (t = ta.getPooled(ut.beforeInput, t, n, r), t.data = e, qt(t)) : t = null, l === null ? t : t === null ? l : [l, t];
  } }, ua = { color: !0, date: !0, datetime: !0, "datetime-local": !0, email: !0, month: !0, number: !0, password: !0, range: !0, search: !0, tel: !0, text: !0, time: !0, url: !0, week: !0 };
  function Ho(e) {
    var t = e && e.nodeName && e.nodeName.toLowerCase();
    return t === "input" ? !!ua[e.type] : t === "textarea";
  }
  var $o = { change: { phasedRegistrationNames: { bubbled: "onChange", captured: "onChangeCapture" }, dependencies: "blur change click focus input keydown keyup selectionchange".split(" ") } };
  function Ko(e, t, n) {
    return e = Se.getPooled($o.change, e, t, n), e.type = "change", B(n), qt(e), e;
  }
  var _n = null, Pn = null;
  function sa(e) {
    qn(e);
  }
  function sr(e) {
    var t = It(e);
    if (Ki(t))
      return e;
  }
  function aa(e, t) {
    if (e === "change")
      return t;
  }
  var jl = !1;
  ae && (jl = fo("input") && (!document.documentMode || 9 < document.documentMode));
  function Bo() {
    _n && (_n.detachEvent("onpropertychange", Yo), Pn = _n = null);
  }
  function Yo(e) {
    if (e.propertyName === "value" && sr(Pn))
      if (e = Ko(Pn, e, al(e)), ke)
        qn(e);
      else {
        ke = !0;
        try {
          ze(sa, e);
        } finally {
          ke = !1, rt();
        }
      }
  }
  function fa(e, t, n) {
    e === "focus" ? (Bo(), _n = t, Pn = n, _n.attachEvent("onpropertychange", Yo)) : e === "blur" && Bo();
  }
  function ca(e) {
    if (e === "selectionchange" || e === "keyup" || e === "keydown")
      return sr(Pn);
  }
  function da(e, t) {
    if (e === "click")
      return sr(t);
  }
  function pa(e, t) {
    if (e === "input" || e === "change")
      return sr(t);
  }
  var ma = { eventTypes: $o, _isInputEventSupported: jl, extractEvents: function(e, t, n, r) {
    var l = t ? It(t) : window, i = l.nodeName && l.nodeName.toLowerCase();
    if (i === "select" || i === "input" && l.type === "file")
      var o = aa;
    else if (Ho(l))
      if (jl)
        o = pa;
      else {
        o = ca;
        var u = fa;
      }
    else
      (i = l.nodeName) && i.toLowerCase() === "input" && (l.type === "checkbox" || l.type === "radio") && (o = da);
    if (o && (o = o(e, t)))
      return Ko(o, n, r);
    u && u(e, l, t), e === "blur" && (e = l._wrapperState) && e.controlled && l.type === "number" && nl(l, "number", l.value);
  } }, Nn = Se.extend({ view: null, detail: null }), ha = { Alt: "altKey", Control: "ctrlKey", Meta: "metaKey", Shift: "shiftKey" };
  function va(e) {
    var t = this.nativeEvent;
    return t.getModifierState ? t.getModifierState(e) : (e = ha[e]) ? !!t[e] : !1;
  }
  function Fl() {
    return va;
  }
  var Xo = 0, Go = 0, Zo = !1, Jo = !1, On = Nn.extend({ screenX: null, screenY: null, clientX: null, clientY: null, pageX: null, pageY: null, ctrlKey: null, shiftKey: null, altKey: null, metaKey: null, getModifierState: Fl, button: null, buttons: null, relatedTarget: function(e) {
    return e.relatedTarget || (e.fromElement === e.srcElement ? e.toElement : e.fromElement);
  }, movementX: function(e) {
    if ("movementX" in e)
      return e.movementX;
    var t = Xo;
    return Xo = e.screenX, Zo ? e.type === "mousemove" ? e.screenX - t : 0 : (Zo = !0, 0);
  }, movementY: function(e) {
    if ("movementY" in e)
      return e.movementY;
    var t = Go;
    return Go = e.screenY, Jo ? e.type === "mousemove" ? e.screenY - t : 0 : (Jo = !0, 0);
  } }), qo = On.extend({ pointerId: null, width: null, height: null, pressure: null, tangentialPressure: null, tiltX: null, tiltY: null, twist: null, pointerType: null, isPrimary: null }), Rn = { mouseEnter: { registrationName: "onMouseEnter", dependencies: ["mouseout", "mouseover"] }, mouseLeave: { registrationName: "onMouseLeave", dependencies: ["mouseout", "mouseover"] }, pointerEnter: { registrationName: "onPointerEnter", dependencies: ["pointerout", "pointerover"] }, pointerLeave: {
    registrationName: "onPointerLeave",
    dependencies: ["pointerout", "pointerover"]
  } }, ya = { eventTypes: Rn, extractEvents: function(e, t, n, r, l) {
    var i = e === "mouseover" || e === "pointerover", o = e === "mouseout" || e === "pointerout";
    if (i && !(l & 32) && (n.relatedTarget || n.fromElement) || !o && !i)
      return null;
    if (i = r.window === r ? r : (i = r.ownerDocument) ? i.defaultView || i.parentWindow : window, o) {
      if (o = t, t = (t = n.relatedTarget || n.toElement) ? xn(t) : null, t !== null) {
        var u = zt(t);
        (t !== u || t.tag !== 5 && t.tag !== 6) && (t = null);
      }
    } else
      o = null;
    if (o === t)
      return null;
    if (e === "mouseout" || e === "mouseover")
      var c = On, d = Rn.mouseLeave, w = Rn.mouseEnter, T = "mouse";
    else
      (e === "pointerout" || e === "pointerover") && (c = qo, d = Rn.pointerLeave, w = Rn.pointerEnter, T = "pointer");
    if (e = o == null ? i : It(o), i = t == null ? i : It(t), d = c.getPooled(d, o, n, r), d.type = T + "leave", d.target = e, d.relatedTarget = i, n = c.getPooled(w, t, n, r), n.type = T + "enter", n.target = i, n.relatedTarget = e, r = o, T = t, r && T)
      e: {
        for (c = r, w = T, o = 0, e = c; e; e = ot(e))
          o++;
        for (e = 0, t = w; t; t = ot(t))
          e++;
        for (; 0 < o - e; )
          c = ot(c), o--;
        for (; 0 < e - o; )
          w = ot(w), e--;
        for (; o--; ) {
          if (c === w || c === w.alternate)
            break e;
          c = ot(c), w = ot(w);
        }
        c = null;
      }
    else
      c = null;
    for (w = c, c = []; r && r !== w && (o = r.alternate, !(o !== null && o === w)); )
      c.push(r), r = ot(r);
    for (r = []; T && T !== w && (o = T.alternate, !(o !== null && o === w)); )
      r.push(T), T = ot(T);
    for (T = 0; T < c.length; T++)
      Ml(c[T], "bubbled", d);
    for (T = r.length; 0 < T--; )
      Ml(r[T], "captured", n);
    return l & 64 ? [d, n] : [d];
  } };
  function ga(e, t) {
    return e === t && (e !== 0 || 1 / e === 1 / t) || e !== e && t !== t;
  }
  var jt = typeof Object.is == "function" ? Object.is : ga, wa = Object.prototype.hasOwnProperty;
  function Mn(e, t) {
    if (jt(e, t))
      return !0;
    if (typeof e != "object" || e === null || typeof t != "object" || t === null)
      return !1;
    var n = Object.keys(e), r = Object.keys(t);
    if (n.length !== r.length)
      return !1;
    for (r = 0; r < n.length; r++)
      if (!wa.call(t, n[r]) || !jt(e[n[r]], t[n[r]]))
        return !1;
    return !0;
  }
  var Ea = ae && "documentMode" in document && 11 >= document.documentMode, bo = { select: { phasedRegistrationNames: { bubbled: "onSelect", captured: "onSelectCapture" }, dependencies: "blur contextmenu dragend focus keydown keyup mousedown mouseup selectionchange".split(" ") } }, en = null, Ll = null, zn = null, Dl = !1;
  function eu(e, t) {
    var n = t.window === t ? t.document : t.nodeType === 9 ? t : t.ownerDocument;
    return Dl || en == null || en !== Tl(n) ? null : (n = en, "selectionStart" in n && kl(n) ? n = { start: n.selectionStart, end: n.selectionEnd } : (n = (n.ownerDocument && n.ownerDocument.defaultView || window).getSelection(), n = { anchorNode: n.anchorNode, anchorOffset: n.anchorOffset, focusNode: n.focusNode, focusOffset: n.focusOffset }), zn && Mn(zn, n) ? null : (zn = n, e = Se.getPooled(bo.select, Ll, e, t), e.type = "select", e.target = en, qt(e), e));
  }
  var Ta = { eventTypes: bo, extractEvents: function(e, t, n, r, l, i) {
    if (l = i || (r.window === r ? r.document : r.nodeType === 9 ? r : r.ownerDocument), !(i = !l)) {
      e: {
        l = ul(l), i = Ke.onSelect;
        for (var o = 0; o < i.length; o++)
          if (!l.has(i[o])) {
            l = !1;
            break e;
          }
        l = !0;
      }
      i = !l;
    }
    if (i)
      return null;
    switch (l = t ? It(t) : window, e) {
      case "focus":
        (Ho(l) || l.contentEditable === "true") && (en = l, Ll = t, zn = null);
        break;
      case "blur":
        zn = Ll = en = null;
        break;
      case "mousedown":
        Dl = !0;
        break;
      case "contextmenu":
      case "mouseup":
      case "dragend":
        return Dl = !1, eu(n, r);
      case "selectionchange":
        if (Ea)
          break;
      case "keydown":
      case "keyup":
        return eu(n, r);
    }
    return null;
  } }, ka = Se.extend({ animationName: null, elapsedTime: null, pseudoElement: null }), xa = Se.extend({ clipboardData: function(e) {
    return "clipboardData" in e ? e.clipboardData : window.clipboardData;
  } }), Sa = Nn.extend({ relatedTarget: null });
  function ar(e) {
    var t = e.keyCode;
    return "charCode" in e ? (e = e.charCode, e === 0 && t === 13 && (e = 13)) : e = t, e === 10 && (e = 13), 32 <= e || e === 13 ? e : 0;
  }
  var Ca = { Esc: "Escape", Spacebar: " ", Left: "ArrowLeft", Up: "ArrowUp", Right: "ArrowRight", Down: "ArrowDown", Del: "Delete", Win: "OS", Menu: "ContextMenu", Apps: "ContextMenu", Scroll: "ScrollLock", MozPrintableKey: "Unidentified" }, _a = {
    8: "Backspace",
    9: "Tab",
    12: "Clear",
    13: "Enter",
    16: "Shift",
    17: "Control",
    18: "Alt",
    19: "Pause",
    20: "CapsLock",
    27: "Escape",
    32: " ",
    33: "PageUp",
    34: "PageDown",
    35: "End",
    36: "Home",
    37: "ArrowLeft",
    38: "ArrowUp",
    39: "ArrowRight",
    40: "ArrowDown",
    45: "Insert",
    46: "Delete",
    112: "F1",
    113: "F2",
    114: "F3",
    115: "F4",
    116: "F5",
    117: "F6",
    118: "F7",
    119: "F8",
    120: "F9",
    121: "F10",
    122: "F11",
    123: "F12",
    144: "NumLock",
    145: "ScrollLock",
    224: "Meta"
  }, Pa = Nn.extend({ key: function(e) {
    if (e.key) {
      var t = Ca[e.key] || e.key;
      if (t !== "Unidentified")
        return t;
    }
    return e.type === "keypress" ? (e = ar(e), e === 13 ? "Enter" : String.fromCharCode(e)) : e.type === "keydown" || e.type === "keyup" ? _a[e.keyCode] || "Unidentified" : "";
  }, location: null, ctrlKey: null, shiftKey: null, altKey: null, metaKey: null, repeat: null, locale: null, getModifierState: Fl, charCode: function(e) {
    return e.type === "keypress" ? ar(e) : 0;
  }, keyCode: function(e) {
    return e.type === "keydown" || e.type === "keyup" ? e.keyCode : 0;
  }, which: function(e) {
    return e.type === "keypress" ? ar(e) : e.type === "keydown" || e.type === "keyup" ? e.keyCode : 0;
  } }), Na = On.extend({ dataTransfer: null }), Oa = Nn.extend({ touches: null, targetTouches: null, changedTouches: null, altKey: null, metaKey: null, ctrlKey: null, shiftKey: null, getModifierState: Fl }), Ra = Se.extend({ propertyName: null, elapsedTime: null, pseudoElement: null }), Ma = On.extend({ deltaX: function(e) {
    return "deltaX" in e ? e.deltaX : "wheelDeltaX" in e ? -e.wheelDeltaX : 0;
  }, deltaY: function(e) {
    return "deltaY" in e ? e.deltaY : "wheelDeltaY" in e ? -e.wheelDeltaY : "wheelDelta" in e ? -e.wheelDelta : 0;
  }, deltaZ: null, deltaMode: null }), za = { eventTypes: Eo, extractEvents: function(e, t, n, r) {
    var l = To.get(e);
    if (!l)
      return null;
    switch (e) {
      case "keypress":
        if (ar(n) === 0)
          return null;
      case "keydown":
      case "keyup":
        e = Pa;
        break;
      case "blur":
      case "focus":
        e = Sa;
        break;
      case "click":
        if (n.button === 2)
          return null;
      case "auxclick":
      case "dblclick":
      case "mousedown":
      case "mousemove":
      case "mouseup":
      case "mouseout":
      case "mouseover":
      case "contextmenu":
        e = On;
        break;
      case "drag":
      case "dragend":
      case "dragenter":
      case "dragexit":
      case "dragleave":
      case "dragover":
      case "dragstart":
      case "drop":
        e = Na;
        break;
      case "touchcancel":
      case "touchend":
      case "touchmove":
      case "touchstart":
        e = Oa;
        break;
      case no:
      case ro:
      case lo:
        e = ka;
        break;
      case io:
        e = Ra;
        break;
      case "scroll":
        e = Nn;
        break;
      case "wheel":
        e = Ma;
        break;
      case "copy":
      case "cut":
      case "paste":
        e = xa;
        break;
      case "gotpointercapture":
      case "lostpointercapture":
      case "pointercancel":
      case "pointerdown":
      case "pointermove":
      case "pointerout":
      case "pointerover":
      case "pointerup":
        e = qo;
        break;
      default:
        e = Se;
    }
    return t = e.getPooled(l, t, n, r), qt(t), t;
  } };
  if (Re)
    throw Error(h(101));
  Re = Array.prototype.slice.call("ResponderEventPlugin SimpleEventPlugin EnterLeaveEventPlugin ChangeEventPlugin SelectEventPlugin BeforeInputEventPlugin".split(" ")), be();
  var Ia = Sn;
  we = Rl, Ot = Ia, He = It, q({ SimpleEventPlugin: za, EnterLeaveEventPlugin: ya, ChangeEventPlugin: ma, SelectEventPlugin: Ta, BeforeInputEventPlugin: oa });
  var Ul = [], tn = -1;
  function V(e) {
    0 > tn || (e.current = Ul[tn], Ul[tn] = null, tn--);
  }
  function Y(e, t) {
    tn++, Ul[tn] = e.current, e.current = t;
  }
  var gt = {}, oe = { current: gt }, he = { current: !1 }, Ft = gt;
  function nn(e, t) {
    var n = e.type.contextTypes;
    if (!n)
      return gt;
    var r = e.stateNode;
    if (r && r.__reactInternalMemoizedUnmaskedChildContext === t)
      return r.__reactInternalMemoizedMaskedChildContext;
    var l = {}, i;
    for (i in n)
      l[i] = t[i];
    return r && (e = e.stateNode, e.__reactInternalMemoizedUnmaskedChildContext = t, e.__reactInternalMemoizedMaskedChildContext = l), l;
  }
  function ve(e) {
    return e = e.childContextTypes, e != null;
  }
  function fr() {
    V(he), V(oe);
  }
  function tu(e, t, n) {
    if (oe.current !== gt)
      throw Error(h(168));
    Y(oe, t), Y(he, n);
  }
  function nu(e, t, n) {
    var r = e.stateNode;
    if (e = t.childContextTypes, typeof r.getChildContext != "function")
      return n;
    r = r.getChildContext();
    for (var l in r)
      if (!(l in e))
        throw Error(h(108, lt(t) || "Unknown", l));
    return O({}, n, {}, r);
  }
  function cr(e) {
    return e = (e = e.stateNode) && e.__reactInternalMemoizedMergedChildContext || gt, Ft = oe.current, Y(oe, e), Y(he, he.current), !0;
  }
  function ru(e, t, n) {
    var r = e.stateNode;
    if (!r)
      throw Error(h(169));
    n ? (e = nu(e, t, Ft), r.__reactInternalMemoizedMergedChildContext = e, V(he), V(oe), Y(oe, e)) : V(he), Y(he, n);
  }
  var ja = I.unstable_runWithPriority, Al = I.unstable_scheduleCallback, lu = I.unstable_cancelCallback, iu = I.unstable_requestPaint, Vl = I.unstable_now, Fa = I.unstable_getCurrentPriorityLevel, dr = I.unstable_ImmediatePriority, ou = I.unstable_UserBlockingPriority, uu = I.unstable_NormalPriority, su = I.unstable_LowPriority, au = I.unstable_IdlePriority, fu = {}, La = I.unstable_shouldYield, Da = iu !== void 0 ? iu : function() {
  }, st = null, pr = null, Wl = !1, cu = Vl(), Ie = 1e4 > cu ? Vl : function() {
    return Vl() - cu;
  };
  function mr() {
    switch (Fa()) {
      case dr:
        return 99;
      case ou:
        return 98;
      case uu:
        return 97;
      case su:
        return 96;
      case au:
        return 95;
      default:
        throw Error(h(332));
    }
  }
  function du(e) {
    switch (e) {
      case 99:
        return dr;
      case 98:
        return ou;
      case 97:
        return uu;
      case 96:
        return su;
      case 95:
        return au;
      default:
        throw Error(h(332));
    }
  }
  function wt(e, t) {
    return e = du(e), ja(e, t);
  }
  function pu(e, t, n) {
    return e = du(e), Al(e, t, n);
  }
  function mu(e) {
    return st === null ? (st = [e], pr = Al(dr, hu)) : st.push(e), fu;
  }
  function Ye() {
    if (pr !== null) {
      var e = pr;
      pr = null, lu(e);
    }
    hu();
  }
  function hu() {
    if (!Wl && st !== null) {
      Wl = !0;
      var e = 0;
      try {
        var t = st;
        wt(99, function() {
          for (; e < t.length; e++) {
            var n = t[e];
            do
              n = n(!0);
            while (n !== null);
          }
        }), st = null;
      } catch (n) {
        throw st !== null && (st = st.slice(e + 1)), Al(dr, Ye), n;
      } finally {
        Wl = !1;
      }
    }
  }
  function hr(e, t, n) {
    return n /= 10, 1073741821 - (((1073741821 - e + t / 10) / n | 0) + 1) * n;
  }
  function Ve(e, t) {
    if (e && e.defaultProps) {
      t = O({}, t), e = e.defaultProps;
      for (var n in e)
        t[n] === void 0 && (t[n] = e[n]);
    }
    return t;
  }
  var vr = { current: null }, yr = null, rn = null, gr = null;
  function Ql() {
    gr = rn = yr = null;
  }
  function Hl(e) {
    var t = vr.current;
    V(vr), e.type._context._currentValue = t;
  }
  function vu(e, t) {
    for (; e !== null; ) {
      var n = e.alternate;
      if (e.childExpirationTime < t)
        e.childExpirationTime = t, n !== null && n.childExpirationTime < t && (n.childExpirationTime = t);
      else if (n !== null && n.childExpirationTime < t)
        n.childExpirationTime = t;
      else
        break;
      e = e.return;
    }
  }
  function ln(e, t) {
    yr = e, gr = rn = null, e = e.dependencies, e !== null && e.firstContext !== null && (e.expirationTime >= t && (Ge = !0), e.firstContext = null);
  }
  function je(e, t) {
    if (gr !== e && t !== !1 && t !== 0)
      if ((typeof t != "number" || t === 1073741823) && (gr = e, t = 1073741823), t = { context: e, observedBits: t, next: null }, rn === null) {
        if (yr === null)
          throw Error(h(308));
        rn = t, yr.dependencies = { expirationTime: 0, firstContext: t, responders: null };
      } else
        rn = rn.next = t;
    return e._currentValue;
  }
  var Et = !1;
  function $l(e) {
    e.updateQueue = { baseState: e.memoizedState, baseQueue: null, shared: { pending: null }, effects: null };
  }
  function Kl(e, t) {
    e = e.updateQueue, t.updateQueue === e && (t.updateQueue = { baseState: e.baseState, baseQueue: e.baseQueue, shared: e.shared, effects: e.effects });
  }
  function Tt(e, t) {
    return e = { expirationTime: e, suspenseConfig: t, tag: 0, payload: null, callback: null, next: null }, e.next = e;
  }
  function kt(e, t) {
    if (e = e.updateQueue, e !== null) {
      e = e.shared;
      var n = e.pending;
      n === null ? t.next = t : (t.next = n.next, n.next = t), e.pending = t;
    }
  }
  function yu(e, t) {
    var n = e.alternate;
    n !== null && Kl(n, e), e = e.updateQueue, n = e.baseQueue, n === null ? (e.baseQueue = t.next = t, t.next = t) : (t.next = n.next, n.next = t);
  }
  function In(e, t, n, r) {
    var l = e.updateQueue;
    Et = !1;
    var i = l.baseQueue, o = l.shared.pending;
    if (o !== null) {
      if (i !== null) {
        var u = i.next;
        i.next = o.next, o.next = u;
      }
      i = o, l.shared.pending = null, u = e.alternate, u !== null && (u = u.updateQueue, u !== null && (u.baseQueue = o));
    }
    if (i !== null) {
      u = i.next;
      var c = l.baseState, d = 0, w = null, T = null, j = null;
      if (u !== null) {
        var D = u;
        do {
          if (o = D.expirationTime, o < r) {
            var Le = { expirationTime: D.expirationTime, suspenseConfig: D.suspenseConfig, tag: D.tag, payload: D.payload, callback: D.callback, next: null };
            j === null ? (T = j = Le, w = c) : j = j.next = Le, o > d && (d = o);
          } else {
            j !== null && (j = j.next = { expirationTime: 1073741823, suspenseConfig: D.suspenseConfig, tag: D.tag, payload: D.payload, callback: D.callback, next: null }), ps(o, D.suspenseConfig);
            e: {
              var ie = e, f = D;
              switch (o = t, Le = n, f.tag) {
                case 1:
                  if (ie = f.payload, typeof ie == "function") {
                    c = ie.call(Le, c, o);
                    break e;
                  }
                  c = ie;
                  break e;
                case 3:
                  ie.effectTag = ie.effectTag & -4097 | 64;
                case 0:
                  if (ie = f.payload, o = typeof ie == "function" ? ie.call(Le, c, o) : ie, o == null)
                    break e;
                  c = O({}, c, o);
                  break e;
                case 2:
                  Et = !0;
              }
            }
            D.callback !== null && (e.effectTag |= 32, o = l.effects, o === null ? l.effects = [D] : o.push(D));
          }
          if (D = D.next, D === null || D === u) {
            if (o = l.shared.pending, o === null)
              break;
            D = i.next = o.next, o.next = u, l.baseQueue = i = o, l.shared.pending = null;
          }
        } while (!0);
      }
      j === null ? w = c : j.next = T, l.baseState = w, l.baseQueue = j, Kr(d), e.expirationTime = d, e.memoizedState = c;
    }
  }
  function gu(e, t, n) {
    if (e = t.effects, t.effects = null, e !== null)
      for (t = 0; t < e.length; t++) {
        var r = e[t], l = r.callback;
        if (l !== null) {
          if (r.callback = null, r = l, l = n, typeof r != "function")
            throw Error(h(191, r));
          r.call(l);
        }
      }
  }
  var jn = Ae.ReactCurrentBatchConfig, wu = new S.Component().refs;
  function wr(e, t, n, r) {
    t = e.memoizedState, n = n(r, t), n = n == null ? t : O({}, t, n), e.memoizedState = n, e.expirationTime === 0 && (e.updateQueue.baseState = n);
  }
  var Er = { isMounted: function(e) {
    return (e = e._reactInternalFiber) ? zt(e) === e : !1;
  }, enqueueSetState: function(e, t, n) {
    e = e._reactInternalFiber;
    var r = Je(), l = jn.suspense;
    r = Wt(r, e, l), l = Tt(r, l), l.payload = t, n != null && (l.callback = n), kt(e, l), _t(e, r);
  }, enqueueReplaceState: function(e, t, n) {
    e = e._reactInternalFiber;
    var r = Je(), l = jn.suspense;
    r = Wt(r, e, l), l = Tt(r, l), l.tag = 1, l.payload = t, n != null && (l.callback = n), kt(e, l), _t(e, r);
  }, enqueueForceUpdate: function(e, t) {
    e = e._reactInternalFiber;
    var n = Je(), r = jn.suspense;
    n = Wt(n, e, r), r = Tt(n, r), r.tag = 2, t != null && (r.callback = t), kt(e, r), _t(e, n);
  } };
  function Eu(e, t, n, r, l, i, o) {
    return e = e.stateNode, typeof e.shouldComponentUpdate == "function" ? e.shouldComponentUpdate(r, i, o) : t.prototype && t.prototype.isPureReactComponent ? !Mn(n, r) || !Mn(l, i) : !0;
  }
  function Tu(e, t, n) {
    var r = !1, l = gt, i = t.contextType;
    return typeof i == "object" && i !== null ? i = je(i) : (l = ve(t) ? Ft : oe.current, r = t.contextTypes, i = (r = r != null) ? nn(e, l) : gt), t = new t(n, i), e.memoizedState = t.state !== null && t.state !== void 0 ? t.state : null, t.updater = Er, e.stateNode = t, t._reactInternalFiber = e, r && (e = e.stateNode, e.__reactInternalMemoizedUnmaskedChildContext = l, e.__reactInternalMemoizedMaskedChildContext = i), t;
  }
  function ku(e, t, n, r) {
    e = t.state, typeof t.componentWillReceiveProps == "function" && t.componentWillReceiveProps(n, r), typeof t.UNSAFE_componentWillReceiveProps == "function" && t.UNSAFE_componentWillReceiveProps(n, r), t.state !== e && Er.enqueueReplaceState(t, t.state, null);
  }
  function Bl(e, t, n, r) {
    var l = e.stateNode;
    l.props = n, l.state = e.memoizedState, l.refs = wu, $l(e);
    var i = t.contextType;
    typeof i == "object" && i !== null ? l.context = je(i) : (i = ve(t) ? Ft : oe.current, l.context = nn(e, i)), In(e, n, l, r), l.state = e.memoizedState, i = t.getDerivedStateFromProps, typeof i == "function" && (wr(e, t, i, n), l.state = e.memoizedState), typeof t.getDerivedStateFromProps == "function" || typeof l.getSnapshotBeforeUpdate == "function" || typeof l.UNSAFE_componentWillMount != "function" && typeof l.componentWillMount != "function" || (t = l.state, typeof l.componentWillMount == "function" && l.componentWillMount(), typeof l.UNSAFE_componentWillMount == "function" && l.UNSAFE_componentWillMount(), t !== l.state && Er.enqueueReplaceState(l, l.state, null), In(e, n, l, r), l.state = e.memoizedState), typeof l.componentDidMount == "function" && (e.effectTag |= 4);
  }
  var Tr = Array.isArray;
  function Fn(e, t, n) {
    if (e = n.ref, e !== null && typeof e != "function" && typeof e != "object") {
      if (n._owner) {
        if (n = n._owner, n) {
          if (n.tag !== 1)
            throw Error(h(309));
          var r = n.stateNode;
        }
        if (!r)
          throw Error(h(147, e));
        var l = "" + e;
        return t !== null && t.ref !== null && typeof t.ref == "function" && t.ref._stringRef === l ? t.ref : (t = function(i) {
          var o = r.refs;
          o === wu && (o = r.refs = {}), i === null ? delete o[l] : o[l] = i;
        }, t._stringRef = l, t);
      }
      if (typeof e != "string")
        throw Error(h(284));
      if (!n._owner)
        throw Error(h(290, e));
    }
    return e;
  }
  function kr(e, t) {
    if (e.type !== "textarea")
      throw Error(h(31, Object.prototype.toString.call(t) === "[object Object]" ? "object with keys {" + Object.keys(t).join(", ") + "}" : t, ""));
  }
  function xu(e) {
    function t(f, a) {
      if (e) {
        var m = f.lastEffect;
        m !== null ? (m.nextEffect = a, f.lastEffect = a) : f.firstEffect = f.lastEffect = a, a.nextEffect = null, a.effectTag = 8;
      }
    }
    function n(f, a) {
      if (!e)
        return null;
      for (; a !== null; )
        t(f, a), a = a.sibling;
      return null;
    }
    function r(f, a) {
      for (f = /* @__PURE__ */ new Map(); a !== null; )
        a.key !== null ? f.set(a.key, a) : f.set(a.index, a), a = a.sibling;
      return f;
    }
    function l(f, a) {
      return f = Kt(f, a), f.index = 0, f.sibling = null, f;
    }
    function i(f, a, m) {
      return f.index = m, e ? (m = f.alternate, m !== null ? (m = m.index, m < a ? (f.effectTag = 2, a) : m) : (f.effectTag = 2, a)) : a;
    }
    function o(f) {
      return e && f.alternate === null && (f.effectTag = 2), f;
    }
    function u(f, a, m, v) {
      return a === null || a.tag !== 6 ? (a = Pi(m, f.mode, v), a.return = f, a) : (a = l(a, m), a.return = f, a);
    }
    function c(f, a, m, v) {
      return a !== null && a.elementType === m.type ? (v = l(a, m.props), v.ref = Fn(f, a, m), v.return = f, v) : (v = Br(m.type, m.key, m.props, null, f.mode, v), v.ref = Fn(f, a, m), v.return = f, v);
    }
    function d(f, a, m, v) {
      return a === null || a.tag !== 4 || a.stateNode.containerInfo !== m.containerInfo || a.stateNode.implementation !== m.implementation ? (a = Ni(m, f.mode, v), a.return = f, a) : (a = l(a, m.children || []), a.return = f, a);
    }
    function w(f, a, m, v, g) {
      return a === null || a.tag !== 7 ? (a = Pt(m, f.mode, v, g), a.return = f, a) : (a = l(a, m), a.return = f, a);
    }
    function T(f, a, m) {
      if (typeof a == "string" || typeof a == "number")
        return a = Pi("" + a, f.mode, m), a.return = f, a;
      if (typeof a == "object" && a !== null) {
        switch (a.$$typeof) {
          case Kn:
            return m = Br(a.type, a.key, a.props, null, f.mode, m), m.ref = Fn(f, null, a), m.return = f, m;
          case Yt:
            return a = Ni(a, f.mode, m), a.return = f, a;
        }
        if (Tr(a) || cn(a))
          return a = Pt(a, f.mode, m, null), a.return = f, a;
        kr(f, a);
      }
      return null;
    }
    function j(f, a, m, v) {
      var g = a !== null ? a.key : null;
      if (typeof m == "string" || typeof m == "number")
        return g !== null ? null : u(f, a, "" + m, v);
      if (typeof m == "object" && m !== null) {
        switch (m.$$typeof) {
          case Kn:
            return m.key === g ? m.type === Mt ? w(f, a, m.props.children, v, g) : c(f, a, m, v) : null;
          case Yt:
            return m.key === g ? d(f, a, m, v) : null;
        }
        if (Tr(m) || cn(m))
          return g !== null ? null : w(f, a, m, v, null);
        kr(f, m);
      }
      return null;
    }
    function D(f, a, m, v, g) {
      if (typeof v == "string" || typeof v == "number")
        return f = f.get(m) || null, u(a, f, "" + v, g);
      if (typeof v == "object" && v !== null) {
        switch (v.$$typeof) {
          case Kn:
            return f = f.get(v.key === null ? m : v.key) || null, v.type === Mt ? w(a, f, v.props.children, g, v.key) : c(a, f, v, g);
          case Yt:
            return f = f.get(v.key === null ? m : v.key) || null, d(a, f, v, g);
        }
        if (Tr(v) || cn(v))
          return f = f.get(m) || null, w(a, f, v, g, null);
        kr(a, v);
      }
      return null;
    }
    function Le(f, a, m, v) {
      for (var g = null, k = null, _ = a, L = a = 0, Q = null; _ !== null && L < m.length; L++) {
        _.index > L ? (Q = _, _ = null) : Q = _.sibling;
        var z = j(f, _, m[L], v);
        if (z === null) {
          _ === null && (_ = Q);
          break;
        }
        e && _ && z.alternate === null && t(f, _), a = i(z, a, L), k === null ? g = z : k.sibling = z, k = z, _ = Q;
      }
      if (L === m.length)
        return n(f, _), g;
      if (_ === null) {
        for (; L < m.length; L++)
          _ = T(f, m[L], v), _ !== null && (a = i(_, a, L), k === null ? g = _ : k.sibling = _, k = _);
        return g;
      }
      for (_ = r(f, _); L < m.length; L++)
        Q = D(_, f, L, m[L], v), Q !== null && (e && Q.alternate !== null && _.delete(Q.key === null ? L : Q.key), a = i(Q, a, L), k === null ? g = Q : k.sibling = Q, k = Q);
      return e && _.forEach(function(Nt) {
        return t(f, Nt);
      }), g;
    }
    function ie(f, a, m, v) {
      var g = cn(m);
      if (typeof g != "function")
        throw Error(h(150));
      if (m = g.call(m), m == null)
        throw Error(h(151));
      for (var k = g = null, _ = a, L = a = 0, Q = null, z = m.next(); _ !== null && !z.done; L++, z = m.next()) {
        _.index > L ? (Q = _, _ = null) : Q = _.sibling;
        var Nt = j(f, _, z.value, v);
        if (Nt === null) {
          _ === null && (_ = Q);
          break;
        }
        e && _ && Nt.alternate === null && t(f, _), a = i(Nt, a, L), k === null ? g = Nt : k.sibling = Nt, k = Nt, _ = Q;
      }
      if (z.done)
        return n(f, _), g;
      if (_ === null) {
        for (; !z.done; L++, z = m.next())
          z = T(f, z.value, v), z !== null && (a = i(z, a, L), k === null ? g = z : k.sibling = z, k = z);
        return g;
      }
      for (_ = r(f, _); !z.done; L++, z = m.next())
        z = D(_, f, L, z.value, v), z !== null && (e && z.alternate !== null && _.delete(z.key === null ? L : z.key), a = i(z, a, L), k === null ? g = z : k.sibling = z, k = z);
      return e && _.forEach(function(df) {
        return t(f, df);
      }), g;
    }
    return function(f, a, m, v) {
      var g = typeof m == "object" && m !== null && m.type === Mt && m.key === null;
      g && (m = m.props.children);
      var k = typeof m == "object" && m !== null;
      if (k)
        switch (m.$$typeof) {
          case Kn:
            e: {
              for (k = m.key, g = a; g !== null; ) {
                if (g.key === k) {
                  switch (g.tag) {
                    case 7:
                      if (m.type === Mt) {
                        n(f, g.sibling), a = l(g, m.props.children), a.return = f, f = a;
                        break e;
                      }
                      break;
                    default:
                      if (g.elementType === m.type) {
                        n(
                          f,
                          g.sibling
                        ), a = l(g, m.props), a.ref = Fn(f, g, m), a.return = f, f = a;
                        break e;
                      }
                  }
                  n(f, g);
                  break;
                } else
                  t(f, g);
                g = g.sibling;
              }
              m.type === Mt ? (a = Pt(m.props.children, f.mode, v, m.key), a.return = f, f = a) : (v = Br(m.type, m.key, m.props, null, f.mode, v), v.ref = Fn(f, a, m), v.return = f, f = v);
            }
            return o(f);
          case Yt:
            e: {
              for (g = m.key; a !== null; ) {
                if (a.key === g)
                  if (a.tag === 4 && a.stateNode.containerInfo === m.containerInfo && a.stateNode.implementation === m.implementation) {
                    n(f, a.sibling), a = l(a, m.children || []), a.return = f, f = a;
                    break e;
                  } else {
                    n(f, a);
                    break;
                  }
                else
                  t(f, a);
                a = a.sibling;
              }
              a = Ni(m, f.mode, v), a.return = f, f = a;
            }
            return o(f);
        }
      if (typeof m == "string" || typeof m == "number")
        return m = "" + m, a !== null && a.tag === 6 ? (n(f, a.sibling), a = l(a, m), a.return = f, f = a) : (n(f, a), a = Pi(m, f.mode, v), a.return = f, f = a), o(f);
      if (Tr(m))
        return Le(f, a, m, v);
      if (cn(m))
        return ie(f, a, m, v);
      if (k && kr(f, m), typeof m > "u" && !g)
        switch (f.tag) {
          case 1:
          case 0:
            throw f = f.type, Error(h(152, f.displayName || f.name || "Component"));
        }
      return n(f, a);
    };
  }
  var on = xu(!0), Yl = xu(!1), Ln = {}, Xe = { current: Ln }, Dn = { current: Ln }, Un = { current: Ln };
  function Lt(e) {
    if (e === Ln)
      throw Error(h(174));
    return e;
  }
  function Xl(e, t) {
    switch (Y(Un, t), Y(Dn, e), Y(Xe, Ln), e = t.nodeType, e) {
      case 9:
      case 11:
        t = (t = t.documentElement) ? t.namespaceURI : il(null, "");
        break;
      default:
        e = e === 8 ? t.parentNode : t, t = e.namespaceURI || null, e = e.tagName, t = il(t, e);
    }
    V(Xe), Y(Xe, t);
  }
  function un() {
    V(Xe), V(Dn), V(Un);
  }
  function Su(e) {
    Lt(Un.current);
    var t = Lt(Xe.current), n = il(t, e.type);
    t !== n && (Y(Dn, e), Y(Xe, n));
  }
  function Gl(e) {
    Dn.current === e && (V(Xe), V(Dn));
  }
  var K = { current: 0 };
  function xr(e) {
    for (var t = e; t !== null; ) {
      if (t.tag === 13) {
        var n = t.memoizedState;
        if (n !== null && (n = n.dehydrated, n === null || n.data === xl || n.data === Sl))
          return t;
      } else if (t.tag === 19 && t.memoizedProps.revealOrder !== void 0) {
        if (t.effectTag & 64)
          return t;
      } else if (t.child !== null) {
        t.child.return = t, t = t.child;
        continue;
      }
      if (t === e)
        break;
      for (; t.sibling === null; ) {
        if (t.return === null || t.return === e)
          return null;
        t = t.return;
      }
      t.sibling.return = t.return, t = t.sibling;
    }
    return null;
  }
  function Zl(e, t) {
    return { responder: e, props: t };
  }
  var Sr = Ae.ReactCurrentDispatcher, Fe = Ae.ReactCurrentBatchConfig, xt = 0, Z = null, ue = null, se = null, Cr = !1;
  function Ce() {
    throw Error(h(321));
  }
  function Jl(e, t) {
    if (t === null)
      return !1;
    for (var n = 0; n < t.length && n < e.length; n++)
      if (!jt(e[n], t[n]))
        return !1;
    return !0;
  }
  function ql(e, t, n, r, l, i) {
    if (xt = i, Z = t, t.memoizedState = null, t.updateQueue = null, t.expirationTime = 0, Sr.current = e === null || e.memoizedState === null ? Ua : Aa, e = n(r, l), t.expirationTime === xt) {
      i = 0;
      do {
        if (t.expirationTime = 0, !(25 > i))
          throw Error(h(301));
        i += 1, se = ue = null, t.updateQueue = null, Sr.current = Va, e = n(r, l);
      } while (t.expirationTime === xt);
    }
    if (Sr.current = Rr, t = ue !== null && ue.next !== null, xt = 0, se = ue = Z = null, Cr = !1, t)
      throw Error(h(300));
    return e;
  }
  function sn() {
    var e = { memoizedState: null, baseState: null, baseQueue: null, queue: null, next: null };
    return se === null ? Z.memoizedState = se = e : se = se.next = e, se;
  }
  function an() {
    if (ue === null) {
      var e = Z.alternate;
      e = e !== null ? e.memoizedState : null;
    } else
      e = ue.next;
    var t = se === null ? Z.memoizedState : se.next;
    if (t !== null)
      se = t, ue = e;
    else {
      if (e === null)
        throw Error(h(310));
      ue = e, e = { memoizedState: ue.memoizedState, baseState: ue.baseState, baseQueue: ue.baseQueue, queue: ue.queue, next: null }, se === null ? Z.memoizedState = se = e : se = se.next = e;
    }
    return se;
  }
  function Dt(e, t) {
    return typeof t == "function" ? t(e) : t;
  }
  function _r(e) {
    var t = an(), n = t.queue;
    if (n === null)
      throw Error(h(311));
    n.lastRenderedReducer = e;
    var r = ue, l = r.baseQueue, i = n.pending;
    if (i !== null) {
      if (l !== null) {
        var o = l.next;
        l.next = i.next, i.next = o;
      }
      r.baseQueue = l = i, n.pending = null;
    }
    if (l !== null) {
      l = l.next, r = r.baseState;
      var u = o = i = null, c = l;
      do {
        var d = c.expirationTime;
        if (d < xt) {
          var w = { expirationTime: c.expirationTime, suspenseConfig: c.suspenseConfig, action: c.action, eagerReducer: c.eagerReducer, eagerState: c.eagerState, next: null };
          u === null ? (o = u = w, i = r) : u = u.next = w, d > Z.expirationTime && (Z.expirationTime = d, Kr(d));
        } else
          u !== null && (u = u.next = { expirationTime: 1073741823, suspenseConfig: c.suspenseConfig, action: c.action, eagerReducer: c.eagerReducer, eagerState: c.eagerState, next: null }), ps(d, c.suspenseConfig), r = c.eagerReducer === e ? c.eagerState : e(r, c.action);
        c = c.next;
      } while (c !== null && c !== l);
      u === null ? i = r : u.next = o, jt(r, t.memoizedState) || (Ge = !0), t.memoizedState = r, t.baseState = i, t.baseQueue = u, n.lastRenderedState = r;
    }
    return [t.memoizedState, n.dispatch];
  }
  function Pr(e) {
    var t = an(), n = t.queue;
    if (n === null)
      throw Error(h(311));
    n.lastRenderedReducer = e;
    var r = n.dispatch, l = n.pending, i = t.memoizedState;
    if (l !== null) {
      n.pending = null;
      var o = l = l.next;
      do
        i = e(i, o.action), o = o.next;
      while (o !== l);
      jt(i, t.memoizedState) || (Ge = !0), t.memoizedState = i, t.baseQueue === null && (t.baseState = i), n.lastRenderedState = i;
    }
    return [i, r];
  }
  function bl(e) {
    var t = sn();
    return typeof e == "function" && (e = e()), t.memoizedState = t.baseState = e, e = t.queue = { pending: null, dispatch: null, lastRenderedReducer: Dt, lastRenderedState: e }, e = e.dispatch = zu.bind(null, Z, e), [t.memoizedState, e];
  }
  function ei(e, t, n, r) {
    return e = { tag: e, create: t, destroy: n, deps: r, next: null }, t = Z.updateQueue, t === null ? (t = { lastEffect: null }, Z.updateQueue = t, t.lastEffect = e.next = e) : (n = t.lastEffect, n === null ? t.lastEffect = e.next = e : (r = n.next, n.next = e, e.next = r, t.lastEffect = e)), e;
  }
  function Cu() {
    return an().memoizedState;
  }
  function ti(e, t, n, r) {
    var l = sn();
    Z.effectTag |= e, l.memoizedState = ei(1 | t, n, void 0, r === void 0 ? null : r);
  }
  function ni(e, t, n, r) {
    var l = an();
    r = r === void 0 ? null : r;
    var i = void 0;
    if (ue !== null) {
      var o = ue.memoizedState;
      if (i = o.destroy, r !== null && Jl(r, o.deps)) {
        ei(t, n, i, r);
        return;
      }
    }
    Z.effectTag |= e, l.memoizedState = ei(1 | t, n, i, r);
  }
  function _u(e, t) {
    return ti(516, 4, e, t);
  }
  function Nr(e, t) {
    return ni(516, 4, e, t);
  }
  function Pu(e, t) {
    return ni(4, 2, e, t);
  }
  function Nu(e, t) {
    if (typeof t == "function")
      return e = e(), t(e), function() {
        t(null);
      };
    if (t != null)
      return e = e(), t.current = e, function() {
        t.current = null;
      };
  }
  function Ou(e, t, n) {
    return n = n != null ? n.concat([e]) : null, ni(4, 2, Nu.bind(null, t, e), n);
  }
  function ri() {
  }
  function Ru(e, t) {
    return sn().memoizedState = [e, t === void 0 ? null : t], e;
  }
  function Or(e, t) {
    var n = an();
    t = t === void 0 ? null : t;
    var r = n.memoizedState;
    return r !== null && t !== null && Jl(t, r[1]) ? r[0] : (n.memoizedState = [e, t], e);
  }
  function Mu(e, t) {
    var n = an();
    t = t === void 0 ? null : t;
    var r = n.memoizedState;
    return r !== null && t !== null && Jl(t, r[1]) ? r[0] : (e = e(), n.memoizedState = [e, t], e);
  }
  function li(e, t, n) {
    var r = mr();
    wt(98 > r ? 98 : r, function() {
      e(!0);
    }), wt(97 < r ? 97 : r, function() {
      var l = Fe.suspense;
      Fe.suspense = t === void 0 ? null : t;
      try {
        e(!1), n();
      } finally {
        Fe.suspense = l;
      }
    });
  }
  function zu(e, t, n) {
    var r = Je(), l = jn.suspense;
    r = Wt(r, e, l), l = { expirationTime: r, suspenseConfig: l, action: n, eagerReducer: null, eagerState: null, next: null };
    var i = t.pending;
    if (i === null ? l.next = l : (l.next = i.next, i.next = l), t.pending = l, i = e.alternate, e === Z || i !== null && i === Z)
      Cr = !0, l.expirationTime = xt, Z.expirationTime = xt;
    else {
      if (e.expirationTime === 0 && (i === null || i.expirationTime === 0) && (i = t.lastRenderedReducer, i !== null))
        try {
          var o = t.lastRenderedState, u = i(o, n);
          if (l.eagerReducer = i, l.eagerState = u, jt(u, o))
            return;
        } catch {
        } finally {
        }
      _t(
        e,
        r
      );
    }
  }
  var Rr = { readContext: je, useCallback: Ce, useContext: Ce, useEffect: Ce, useImperativeHandle: Ce, useLayoutEffect: Ce, useMemo: Ce, useReducer: Ce, useRef: Ce, useState: Ce, useDebugValue: Ce, useResponder: Ce, useDeferredValue: Ce, useTransition: Ce }, Ua = { readContext: je, useCallback: Ru, useContext: je, useEffect: _u, useImperativeHandle: function(e, t, n) {
    return n = n != null ? n.concat([e]) : null, ti(4, 2, Nu.bind(null, t, e), n);
  }, useLayoutEffect: function(e, t) {
    return ti(4, 2, e, t);
  }, useMemo: function(e, t) {
    var n = sn();
    return t = t === void 0 ? null : t, e = e(), n.memoizedState = [
      e,
      t
    ], e;
  }, useReducer: function(e, t, n) {
    var r = sn();
    return t = n !== void 0 ? n(t) : t, r.memoizedState = r.baseState = t, e = r.queue = { pending: null, dispatch: null, lastRenderedReducer: e, lastRenderedState: t }, e = e.dispatch = zu.bind(null, Z, e), [r.memoizedState, e];
  }, useRef: function(e) {
    var t = sn();
    return e = { current: e }, t.memoizedState = e;
  }, useState: bl, useDebugValue: ri, useResponder: Zl, useDeferredValue: function(e, t) {
    var n = bl(e), r = n[0], l = n[1];
    return _u(function() {
      var i = Fe.suspense;
      Fe.suspense = t === void 0 ? null : t;
      try {
        l(e);
      } finally {
        Fe.suspense = i;
      }
    }, [e, t]), r;
  }, useTransition: function(e) {
    var t = bl(!1), n = t[0];
    return t = t[1], [Ru(li.bind(null, t, e), [t, e]), n];
  } }, Aa = { readContext: je, useCallback: Or, useContext: je, useEffect: Nr, useImperativeHandle: Ou, useLayoutEffect: Pu, useMemo: Mu, useReducer: _r, useRef: Cu, useState: function() {
    return _r(Dt);
  }, useDebugValue: ri, useResponder: Zl, useDeferredValue: function(e, t) {
    var n = _r(Dt), r = n[0], l = n[1];
    return Nr(function() {
      var i = Fe.suspense;
      Fe.suspense = t === void 0 ? null : t;
      try {
        l(e);
      } finally {
        Fe.suspense = i;
      }
    }, [e, t]), r;
  }, useTransition: function(e) {
    var t = _r(Dt), n = t[0];
    return t = t[1], [Or(li.bind(null, t, e), [t, e]), n];
  } }, Va = { readContext: je, useCallback: Or, useContext: je, useEffect: Nr, useImperativeHandle: Ou, useLayoutEffect: Pu, useMemo: Mu, useReducer: Pr, useRef: Cu, useState: function() {
    return Pr(Dt);
  }, useDebugValue: ri, useResponder: Zl, useDeferredValue: function(e, t) {
    var n = Pr(Dt), r = n[0], l = n[1];
    return Nr(function() {
      var i = Fe.suspense;
      Fe.suspense = t === void 0 ? null : t;
      try {
        l(e);
      } finally {
        Fe.suspense = i;
      }
    }, [e, t]), r;
  }, useTransition: function(e) {
    var t = Pr(Dt), n = t[0];
    return t = t[1], [Or(li.bind(
      null,
      t,
      e
    ), [t, e]), n];
  } }, at = null, St = null, Ut = !1;
  function Iu(e, t) {
    var n = qe(5, null, null, 0);
    n.elementType = "DELETED", n.type = "DELETED", n.stateNode = t, n.return = e, n.effectTag = 8, e.lastEffect !== null ? (e.lastEffect.nextEffect = n, e.lastEffect = n) : e.firstEffect = e.lastEffect = n;
  }
  function ju(e, t) {
    switch (e.tag) {
      case 5:
        var n = e.type;
        return t = t.nodeType !== 1 || n.toLowerCase() !== t.nodeName.toLowerCase() ? null : t, t !== null ? (e.stateNode = t, !0) : !1;
      case 6:
        return t = e.pendingProps === "" || t.nodeType !== 3 ? null : t, t !== null ? (e.stateNode = t, !0) : !1;
      case 13:
        return !1;
      default:
        return !1;
    }
  }
  function ii(e) {
    if (Ut) {
      var t = St;
      if (t) {
        var n = t;
        if (!ju(e, t)) {
          if (t = Jt(n.nextSibling), !t || !ju(e, t)) {
            e.effectTag = e.effectTag & -1025 | 2, Ut = !1, at = e;
            return;
          }
          Iu(at, n);
        }
        at = e, St = Jt(t.firstChild);
      } else
        e.effectTag = e.effectTag & -1025 | 2, Ut = !1, at = e;
    }
  }
  function Fu(e) {
    for (e = e.return; e !== null && e.tag !== 5 && e.tag !== 3 && e.tag !== 13; )
      e = e.return;
    at = e;
  }
  function Mr(e) {
    if (e !== at)
      return !1;
    if (!Ut)
      return Fu(e), Ut = !0, !1;
    var t = e.type;
    if (e.tag !== 5 || t !== "head" && t !== "body" && !Pl(t, e.memoizedProps))
      for (t = St; t; )
        Iu(e, t), t = Jt(t.nextSibling);
    if (Fu(e), e.tag === 13) {
      if (e = e.memoizedState, e = e !== null ? e.dehydrated : null, !e)
        throw Error(h(317));
      e: {
        for (e = e.nextSibling, t = 0; e; ) {
          if (e.nodeType === 8) {
            var n = e.data;
            if (n === Mo) {
              if (t === 0) {
                St = Jt(e.nextSibling);
                break e;
              }
              t--;
            } else
              n !== Ro && n !== Sl && n !== xl || t++;
          }
          e = e.nextSibling;
        }
        St = null;
      }
    } else
      St = at ? Jt(e.stateNode.nextSibling) : null;
    return !0;
  }
  function oi() {
    St = at = null, Ut = !1;
  }
  var Wa = Ae.ReactCurrentOwner, Ge = !1;
  function _e(e, t, n, r) {
    t.child = e === null ? Yl(t, null, n, r) : on(t, e.child, n, r);
  }
  function Lu(e, t, n, r, l) {
    n = n.render;
    var i = t.ref;
    return ln(t, l), r = ql(e, t, n, r, i, l), e !== null && !Ge ? (t.updateQueue = e.updateQueue, t.effectTag &= -517, e.expirationTime <= l && (e.expirationTime = 0), ft(e, t, l)) : (t.effectTag |= 1, _e(e, t, r, l), t.child);
  }
  function Du(e, t, n, r, l, i) {
    if (e === null) {
      var o = n.type;
      return typeof o == "function" && !_i(o) && o.defaultProps === void 0 && n.compare === null && n.defaultProps === void 0 ? (t.tag = 15, t.type = o, Uu(e, t, o, r, l, i)) : (e = Br(n.type, null, r, null, t.mode, i), e.ref = t.ref, e.return = t, t.child = e);
    }
    return o = e.child, l < i && (l = o.memoizedProps, n = n.compare, n = n !== null ? n : Mn, n(l, r) && e.ref === t.ref) ? ft(e, t, i) : (t.effectTag |= 1, e = Kt(o, r), e.ref = t.ref, e.return = t, t.child = e);
  }
  function Uu(e, t, n, r, l, i) {
    return e !== null && Mn(e.memoizedProps, r) && e.ref === t.ref && (Ge = !1, l < i) ? (t.expirationTime = e.expirationTime, ft(e, t, i)) : ui(e, t, n, r, i);
  }
  function Au(e, t) {
    var n = t.ref;
    (e === null && n !== null || e !== null && e.ref !== n) && (t.effectTag |= 128);
  }
  function ui(e, t, n, r, l) {
    var i = ve(n) ? Ft : oe.current;
    return i = nn(t, i), ln(t, l), n = ql(e, t, n, r, i, l), e !== null && !Ge ? (t.updateQueue = e.updateQueue, t.effectTag &= -517, e.expirationTime <= l && (e.expirationTime = 0), ft(e, t, l)) : (t.effectTag |= 1, _e(e, t, n, l), t.child);
  }
  function Vu(e, t, n, r, l) {
    if (ve(n)) {
      var i = !0;
      cr(t);
    } else
      i = !1;
    if (ln(t, l), t.stateNode === null)
      e !== null && (e.alternate = null, t.alternate = null, t.effectTag |= 2), Tu(t, n, r), Bl(t, n, r, l), r = !0;
    else if (e === null) {
      var o = t.stateNode, u = t.memoizedProps;
      o.props = u;
      var c = o.context, d = n.contextType;
      typeof d == "object" && d !== null ? d = je(d) : (d = ve(n) ? Ft : oe.current, d = nn(t, d));
      var w = n.getDerivedStateFromProps, T = typeof w == "function" || typeof o.getSnapshotBeforeUpdate == "function";
      T || typeof o.UNSAFE_componentWillReceiveProps != "function" && typeof o.componentWillReceiveProps != "function" || (u !== r || c !== d) && ku(t, o, r, d), Et = !1;
      var j = t.memoizedState;
      o.state = j, In(t, r, o, l), c = t.memoizedState, u !== r || j !== c || he.current || Et ? (typeof w == "function" && (wr(t, n, w, r), c = t.memoizedState), (u = Et || Eu(t, n, u, r, j, c, d)) ? (T || typeof o.UNSAFE_componentWillMount != "function" && typeof o.componentWillMount != "function" || (typeof o.componentWillMount == "function" && o.componentWillMount(), typeof o.UNSAFE_componentWillMount == "function" && o.UNSAFE_componentWillMount()), typeof o.componentDidMount == "function" && (t.effectTag |= 4)) : (typeof o.componentDidMount == "function" && (t.effectTag |= 4), t.memoizedProps = r, t.memoizedState = c), o.props = r, o.state = c, o.context = d, r = u) : (typeof o.componentDidMount == "function" && (t.effectTag |= 4), r = !1);
    } else
      o = t.stateNode, Kl(e, t), u = t.memoizedProps, o.props = t.type === t.elementType ? u : Ve(t.type, u), c = o.context, d = n.contextType, typeof d == "object" && d !== null ? d = je(d) : (d = ve(n) ? Ft : oe.current, d = nn(t, d)), w = n.getDerivedStateFromProps, (T = typeof w == "function" || typeof o.getSnapshotBeforeUpdate == "function") || typeof o.UNSAFE_componentWillReceiveProps != "function" && typeof o.componentWillReceiveProps != "function" || (u !== r || c !== d) && ku(t, o, r, d), Et = !1, c = t.memoizedState, o.state = c, In(t, r, o, l), j = t.memoizedState, u !== r || c !== j || he.current || Et ? (typeof w == "function" && (wr(t, n, w, r), j = t.memoizedState), (w = Et || Eu(t, n, u, r, c, j, d)) ? (T || typeof o.UNSAFE_componentWillUpdate != "function" && typeof o.componentWillUpdate != "function" || (typeof o.componentWillUpdate == "function" && o.componentWillUpdate(
        r,
        j,
        d
      ), typeof o.UNSAFE_componentWillUpdate == "function" && o.UNSAFE_componentWillUpdate(r, j, d)), typeof o.componentDidUpdate == "function" && (t.effectTag |= 4), typeof o.getSnapshotBeforeUpdate == "function" && (t.effectTag |= 256)) : (typeof o.componentDidUpdate != "function" || u === e.memoizedProps && c === e.memoizedState || (t.effectTag |= 4), typeof o.getSnapshotBeforeUpdate != "function" || u === e.memoizedProps && c === e.memoizedState || (t.effectTag |= 256), t.memoizedProps = r, t.memoizedState = j), o.props = r, o.state = j, o.context = d, r = w) : (typeof o.componentDidUpdate != "function" || u === e.memoizedProps && c === e.memoizedState || (t.effectTag |= 4), typeof o.getSnapshotBeforeUpdate != "function" || u === e.memoizedProps && c === e.memoizedState || (t.effectTag |= 256), r = !1);
    return si(e, t, n, r, i, l);
  }
  function si(e, t, n, r, l, i) {
    Au(e, t);
    var o = (t.effectTag & 64) !== 0;
    if (!r && !o)
      return l && ru(t, n, !1), ft(e, t, i);
    r = t.stateNode, Wa.current = t;
    var u = o && typeof n.getDerivedStateFromError != "function" ? null : r.render();
    return t.effectTag |= 1, e !== null && o ? (t.child = on(t, e.child, null, i), t.child = on(t, null, u, i)) : _e(e, t, u, i), t.memoizedState = r.state, l && ru(t, n, !0), t.child;
  }
  function Wu(e) {
    var t = e.stateNode;
    t.pendingContext ? tu(e, t.pendingContext, t.pendingContext !== t.context) : t.context && tu(e, t.context, !1), Xl(e, t.containerInfo);
  }
  var ai = { dehydrated: null, retryTime: 0 };
  function Qu(e, t, n) {
    var r = t.mode, l = t.pendingProps, i = K.current, o = !1, u;
    if ((u = (t.effectTag & 64) !== 0) || (u = (i & 2) !== 0 && (e === null || e.memoizedState !== null)), u ? (o = !0, t.effectTag &= -65) : e !== null && e.memoizedState === null || l.fallback === void 0 || l.unstable_avoidThisFallback === !0 || (i |= 1), Y(K, i & 1), e === null) {
      if (l.fallback !== void 0 && ii(t), o) {
        if (o = l.fallback, l = Pt(null, r, 0, null), l.return = t, !(t.mode & 2))
          for (e = t.memoizedState !== null ? t.child.child : t.child, l.child = e; e !== null; )
            e.return = l, e = e.sibling;
        return n = Pt(o, r, n, null), n.return = t, l.sibling = n, t.memoizedState = ai, t.child = l, n;
      }
      return r = l.children, t.memoizedState = null, t.child = Yl(t, null, r, n);
    }
    if (e.memoizedState !== null) {
      if (e = e.child, r = e.sibling, o) {
        if (l = l.fallback, n = Kt(e, e.pendingProps), n.return = t, !(t.mode & 2) && (o = t.memoizedState !== null ? t.child.child : t.child, o !== e.child))
          for (n.child = o; o !== null; )
            o.return = n, o = o.sibling;
        return r = Kt(r, l), r.return = t, n.sibling = r, n.childExpirationTime = 0, t.memoizedState = ai, t.child = n, r;
      }
      return n = on(t, e.child, l.children, n), t.memoizedState = null, t.child = n;
    }
    if (e = e.child, o) {
      if (o = l.fallback, l = Pt(null, r, 0, null), l.return = t, l.child = e, e !== null && (e.return = l), !(t.mode & 2))
        for (e = t.memoizedState !== null ? t.child.child : t.child, l.child = e; e !== null; )
          e.return = l, e = e.sibling;
      return n = Pt(o, r, n, null), n.return = t, l.sibling = n, n.effectTag |= 2, l.childExpirationTime = 0, t.memoizedState = ai, t.child = l, n;
    }
    return t.memoizedState = null, t.child = on(t, e, l.children, n);
  }
  function Hu(e, t) {
    e.expirationTime < t && (e.expirationTime = t);
    var n = e.alternate;
    n !== null && n.expirationTime < t && (n.expirationTime = t), vu(e.return, t);
  }
  function fi(e, t, n, r, l, i) {
    var o = e.memoizedState;
    o === null ? e.memoizedState = { isBackwards: t, rendering: null, renderingStartTime: 0, last: r, tail: n, tailExpiration: 0, tailMode: l, lastEffect: i } : (o.isBackwards = t, o.rendering = null, o.renderingStartTime = 0, o.last = r, o.tail = n, o.tailExpiration = 0, o.tailMode = l, o.lastEffect = i);
  }
  function $u(e, t, n) {
    var r = t.pendingProps, l = r.revealOrder, i = r.tail;
    if (_e(e, t, r.children, n), r = K.current, r & 2)
      r = r & 1 | 2, t.effectTag |= 64;
    else {
      if (e !== null && e.effectTag & 64)
        e:
          for (e = t.child; e !== null; ) {
            if (e.tag === 13)
              e.memoizedState !== null && Hu(e, n);
            else if (e.tag === 19)
              Hu(e, n);
            else if (e.child !== null) {
              e.child.return = e, e = e.child;
              continue;
            }
            if (e === t)
              break e;
            for (; e.sibling === null; ) {
              if (e.return === null || e.return === t)
                break e;
              e = e.return;
            }
            e.sibling.return = e.return, e = e.sibling;
          }
      r &= 1;
    }
    if (Y(K, r), !(t.mode & 2))
      t.memoizedState = null;
    else
      switch (l) {
        case "forwards":
          for (n = t.child, l = null; n !== null; )
            e = n.alternate, e !== null && xr(e) === null && (l = n), n = n.sibling;
          n = l, n === null ? (l = t.child, t.child = null) : (l = n.sibling, n.sibling = null), fi(t, !1, l, n, i, t.lastEffect);
          break;
        case "backwards":
          for (n = null, l = t.child, t.child = null; l !== null; ) {
            if (e = l.alternate, e !== null && xr(e) === null) {
              t.child = l;
              break;
            }
            e = l.sibling, l.sibling = n, n = l, l = e;
          }
          fi(t, !0, n, null, i, t.lastEffect);
          break;
        case "together":
          fi(t, !1, null, null, void 0, t.lastEffect);
          break;
        default:
          t.memoizedState = null;
      }
    return t.child;
  }
  function ft(e, t, n) {
    e !== null && (t.dependencies = e.dependencies);
    var r = t.expirationTime;
    if (r !== 0 && Kr(r), t.childExpirationTime < n)
      return null;
    if (e !== null && t.child !== e.child)
      throw Error(h(153));
    if (t.child !== null) {
      for (e = t.child, n = Kt(e, e.pendingProps), t.child = n, n.return = t; e.sibling !== null; )
        e = e.sibling, n = n.sibling = Kt(e, e.pendingProps), n.return = t;
      n.sibling = null;
    }
    return t.child;
  }
  var Ku, ci, Bu, Yu;
  Ku = function(e, t) {
    for (var n = t.child; n !== null; ) {
      if (n.tag === 5 || n.tag === 6)
        e.appendChild(n.stateNode);
      else if (n.tag !== 4 && n.child !== null) {
        n.child.return = n, n = n.child;
        continue;
      }
      if (n === t)
        break;
      for (; n.sibling === null; ) {
        if (n.return === null || n.return === t)
          return;
        n = n.return;
      }
      n.sibling.return = n.return, n = n.sibling;
    }
  }, ci = function() {
  }, Bu = function(e, t, n, r, l) {
    var i = e.memoizedProps;
    if (i !== r) {
      var o = t.stateNode;
      switch (Lt(Xe.current), e = null, n) {
        case "input":
          i = el(o, i), r = el(o, r), e = [];
          break;
        case "option":
          i = rl(o, i), r = rl(o, r), e = [];
          break;
        case "select":
          i = O({}, i, { value: void 0 }), r = O({}, r, { value: void 0 }), e = [];
          break;
        case "textarea":
          i = ll(o, i), r = ll(o, r), e = [];
          break;
        default:
          typeof i.onClick != "function" && typeof r.onClick == "function" && (o.onclick = rr);
      }
      wl(n, r);
      var u, c;
      n = null;
      for (u in i)
        if (!r.hasOwnProperty(u) && i.hasOwnProperty(u) && i[u] != null)
          if (u === "style")
            for (c in o = i[u], o)
              o.hasOwnProperty(c) && (n || (n = {}), n[c] = "");
          else
            u !== "dangerouslySetInnerHTML" && u !== "children" && u !== "suppressContentEditableWarning" && u !== "suppressHydrationWarning" && u !== "autoFocus" && (Ee.hasOwnProperty(u) ? e || (e = []) : (e = e || []).push(u, null));
      for (u in r) {
        var d = r[u];
        if (o = i != null ? i[u] : void 0, r.hasOwnProperty(u) && d !== o && (d != null || o != null))
          if (u === "style")
            if (o) {
              for (c in o)
                !o.hasOwnProperty(c) || d && d.hasOwnProperty(c) || (n || (n = {}), n[c] = "");
              for (c in d)
                d.hasOwnProperty(c) && o[c] !== d[c] && (n || (n = {}), n[c] = d[c]);
            } else
              n || (e || (e = []), e.push(u, n)), n = d;
          else
            u === "dangerouslySetInnerHTML" ? (d = d ? d.__html : void 0, o = o ? o.__html : void 0, d != null && o !== d && (e = e || []).push(u, d)) : u === "children" ? o === d || typeof d != "string" && typeof d != "number" || (e = e || []).push(u, "" + d) : u !== "suppressContentEditableWarning" && u !== "suppressHydrationWarning" && (Ee.hasOwnProperty(u) ? (d != null && it(l, u), e || o === d || (e = [])) : (e = e || []).push(u, d));
      }
      n && (e = e || []).push("style", n), l = e, (t.updateQueue = l) && (t.effectTag |= 4);
    }
  }, Yu = function(e, t, n, r) {
    n !== r && (t.effectTag |= 4);
  };
  function zr(e, t) {
    switch (e.tailMode) {
      case "hidden":
        t = e.tail;
        for (var n = null; t !== null; )
          t.alternate !== null && (n = t), t = t.sibling;
        n === null ? e.tail = null : n.sibling = null;
        break;
      case "collapsed":
        n = e.tail;
        for (var r = null; n !== null; )
          n.alternate !== null && (r = n), n = n.sibling;
        r === null ? t || e.tail === null ? e.tail = null : e.tail.sibling = null : r.sibling = null;
    }
  }
  function Qa(e, t, n) {
    var r = t.pendingProps;
    switch (t.tag) {
      case 2:
      case 16:
      case 15:
      case 0:
      case 11:
      case 7:
      case 8:
      case 12:
      case 9:
      case 14:
        return null;
      case 1:
        return ve(t.type) && fr(), null;
      case 3:
        return un(), V(he), V(oe), n = t.stateNode, n.pendingContext && (n.context = n.pendingContext, n.pendingContext = null), e !== null && e.child !== null || !Mr(t) || (t.effectTag |= 4), ci(t), null;
      case 5:
        Gl(t), n = Lt(Un.current);
        var l = t.type;
        if (e !== null && t.stateNode != null)
          Bu(e, t, l, r, n), e.ref !== t.ref && (t.effectTag |= 128);
        else {
          if (!r) {
            if (t.stateNode === null)
              throw Error(h(166));
            return null;
          }
          if (e = Lt(Xe.current), Mr(t)) {
            r = t.stateNode, l = t.type;
            var i = t.memoizedProps;
            switch (r[vt] = t, r[lr] = i, l) {
              case "iframe":
              case "object":
              case "embed":
                W("load", r);
                break;
              case "video":
              case "audio":
                for (e = 0; e < pn.length; e++)
                  W(pn[e], r);
                break;
              case "source":
                W("error", r);
                break;
              case "img":
              case "image":
              case "link":
                W("error", r), W("load", r);
                break;
              case "form":
                W("reset", r), W("submit", r);
                break;
              case "details":
                W("toggle", r);
                break;
              case "input":
                Bi(r, i), W("invalid", r), it(n, "onChange");
                break;
              case "select":
                r._wrapperState = { wasMultiple: !!i.multiple }, W("invalid", r), it(n, "onChange");
                break;
              case "textarea":
                Gi(r, i), W("invalid", r), it(n, "onChange");
            }
            wl(l, i), e = null;
            for (var o in i)
              if (i.hasOwnProperty(o)) {
                var u = i[o];
                o === "children" ? typeof u == "string" ? r.textContent !== u && (e = ["children", u]) : typeof u == "number" && r.textContent !== "" + u && (e = ["children", "" + u]) : Ee.hasOwnProperty(o) && u != null && it(n, o);
              }
            switch (l) {
              case "input":
                Xn(r), Xi(r, i, !0);
                break;
              case "textarea":
                Xn(r), Ji(r);
                break;
              case "select":
              case "option":
                break;
              default:
                typeof i.onClick == "function" && (r.onclick = rr);
            }
            n = e, t.updateQueue = n, n !== null && (t.effectTag |= 4);
          } else {
            switch (o = n.nodeType === 9 ? n : n.ownerDocument, e === Co && (e = bi(l)), e === Co ? l === "script" ? (e = o.createElement("div"), e.innerHTML = "<script><\/script>", e = e.removeChild(e.firstChild)) : typeof r.is == "string" ? e = o.createElement(l, { is: r.is }) : (e = o.createElement(l), l === "select" && (o = e, r.multiple ? o.multiple = !0 : r.size && (o.size = r.size))) : e = o.createElementNS(e, l), e[vt] = t, e[lr] = r, Ku(e, t, !1, !1), t.stateNode = e, o = El(l, r), l) {
              case "iframe":
              case "object":
              case "embed":
                W(
                  "load",
                  e
                ), u = r;
                break;
              case "video":
              case "audio":
                for (u = 0; u < pn.length; u++)
                  W(pn[u], e);
                u = r;
                break;
              case "source":
                W("error", e), u = r;
                break;
              case "img":
              case "image":
              case "link":
                W("error", e), W("load", e), u = r;
                break;
              case "form":
                W("reset", e), W("submit", e), u = r;
                break;
              case "details":
                W("toggle", e), u = r;
                break;
              case "input":
                Bi(e, r), u = el(e, r), W("invalid", e), it(n, "onChange");
                break;
              case "option":
                u = rl(e, r);
                break;
              case "select":
                e._wrapperState = { wasMultiple: !!r.multiple }, u = O({}, r, { value: void 0 }), W("invalid", e), it(n, "onChange");
                break;
              case "textarea":
                Gi(
                  e,
                  r
                ), u = ll(e, r), W("invalid", e), it(n, "onChange");
                break;
              default:
                u = r;
            }
            wl(l, u);
            var c = u;
            for (i in c)
              if (c.hasOwnProperty(i)) {
                var d = c[i];
                i === "style" ? So(e, d) : i === "dangerouslySetInnerHTML" ? (d = d ? d.__html : void 0, d != null && eo(e, d)) : i === "children" ? typeof d == "string" ? (l !== "textarea" || d !== "") && dn(e, d) : typeof d == "number" && dn(e, "" + d) : i !== "suppressContentEditableWarning" && i !== "suppressHydrationWarning" && i !== "autoFocus" && (Ee.hasOwnProperty(i) ? d != null && it(n, i) : d != null && Gr(e, i, d, o));
              }
            switch (l) {
              case "input":
                Xn(e), Xi(e, r, !1);
                break;
              case "textarea":
                Xn(e), Ji(e);
                break;
              case "option":
                r.value != null && e.setAttribute("value", "" + dt(r.value));
                break;
              case "select":
                e.multiple = !!r.multiple, n = r.value, n != null ? Xt(e, !!r.multiple, n, !1) : r.defaultValue != null && Xt(e, !!r.multiple, r.defaultValue, !0);
                break;
              default:
                typeof u.onClick == "function" && (e.onclick = rr);
            }
            zo(l, r) && (t.effectTag |= 4);
          }
          t.ref !== null && (t.effectTag |= 128);
        }
        return null;
      case 6:
        if (e && t.stateNode != null)
          Yu(e, t, e.memoizedProps, r);
        else {
          if (typeof r != "string" && t.stateNode === null)
            throw Error(h(166));
          n = Lt(Un.current), Lt(Xe.current), Mr(t) ? (n = t.stateNode, r = t.memoizedProps, n[vt] = t, n.nodeValue !== r && (t.effectTag |= 4)) : (n = (n.nodeType === 9 ? n : n.ownerDocument).createTextNode(r), n[vt] = t, t.stateNode = n);
        }
        return null;
      case 13:
        return V(K), r = t.memoizedState, t.effectTag & 64 ? (t.expirationTime = n, t) : (n = r !== null, r = !1, e === null ? t.memoizedProps.fallback !== void 0 && Mr(t) : (l = e.memoizedState, r = l !== null, n || l === null || (l = e.child.sibling, l !== null && (i = t.firstEffect, i !== null ? (t.firstEffect = l, l.nextEffect = i) : (t.firstEffect = t.lastEffect = l, l.nextEffect = null), l.effectTag = 8))), n && !r && t.mode & 2 && (e === null && t.memoizedProps.unstable_avoidThisFallback !== !0 || K.current & 1 ? ne === At && (ne = Fr) : ((ne === At || ne === Fr) && (ne = Lr), Vn !== 0 && Pe !== null && (Bt(Pe, ye), ws(Pe, Vn)))), (n || r) && (t.effectTag |= 4), null);
      case 4:
        return un(), ci(t), null;
      case 10:
        return Hl(t), null;
      case 17:
        return ve(t.type) && fr(), null;
      case 19:
        if (V(K), r = t.memoizedState, r === null)
          return null;
        if (l = (t.effectTag & 64) !== 0, i = r.rendering, i === null) {
          if (l)
            zr(r, !1);
          else if (ne !== At || e !== null && e.effectTag & 64)
            for (i = t.child; i !== null; ) {
              if (e = xr(i), e !== null) {
                for (t.effectTag |= 64, zr(r, !1), l = e.updateQueue, l !== null && (t.updateQueue = l, t.effectTag |= 4), r.lastEffect === null && (t.firstEffect = null), t.lastEffect = r.lastEffect, r = t.child; r !== null; )
                  l = r, i = n, l.effectTag &= 2, l.nextEffect = null, l.firstEffect = null, l.lastEffect = null, e = l.alternate, e === null ? (l.childExpirationTime = 0, l.expirationTime = i, l.child = null, l.memoizedProps = null, l.memoizedState = null, l.updateQueue = null, l.dependencies = null) : (l.childExpirationTime = e.childExpirationTime, l.expirationTime = e.expirationTime, l.child = e.child, l.memoizedProps = e.memoizedProps, l.memoizedState = e.memoizedState, l.updateQueue = e.updateQueue, i = e.dependencies, l.dependencies = i === null ? null : { expirationTime: i.expirationTime, firstContext: i.firstContext, responders: i.responders }), r = r.sibling;
                return Y(K, K.current & 1 | 2), t.child;
              }
              i = i.sibling;
            }
        } else {
          if (!l)
            if (e = xr(i), e !== null) {
              if (t.effectTag |= 64, l = !0, n = e.updateQueue, n !== null && (t.updateQueue = n, t.effectTag |= 4), zr(r, !0), r.tail === null && r.tailMode === "hidden" && !i.alternate)
                return t = t.lastEffect = r.lastEffect, t !== null && (t.nextEffect = null), null;
            } else
              2 * Ie() - r.renderingStartTime > r.tailExpiration && 1 < n && (t.effectTag |= 64, l = !0, zr(r, !1), t.expirationTime = t.childExpirationTime = n - 1);
          r.isBackwards ? (i.sibling = t.child, t.child = i) : (n = r.last, n !== null ? n.sibling = i : t.child = i, r.last = i);
        }
        return r.tail !== null ? (r.tailExpiration === 0 && (r.tailExpiration = Ie() + 500), n = r.tail, r.rendering = n, r.tail = n.sibling, r.lastEffect = t.lastEffect, r.renderingStartTime = Ie(), n.sibling = null, t = K.current, Y(K, l ? t & 1 | 2 : t & 1), n) : null;
    }
    throw Error(h(
      156,
      t.tag
    ));
  }
  function Ha(e) {
    switch (e.tag) {
      case 1:
        ve(e.type) && fr();
        var t = e.effectTag;
        return t & 4096 ? (e.effectTag = t & -4097 | 64, e) : null;
      case 3:
        if (un(), V(he), V(oe), t = e.effectTag, t & 64)
          throw Error(h(285));
        return e.effectTag = t & -4097 | 64, e;
      case 5:
        return Gl(e), null;
      case 13:
        return V(K), t = e.effectTag, t & 4096 ? (e.effectTag = t & -4097 | 64, e) : null;
      case 19:
        return V(K), null;
      case 4:
        return un(), null;
      case 10:
        return Hl(e), null;
      default:
        return null;
    }
  }
  function di(e, t) {
    return { value: e, source: t, stack: br(t) };
  }
  var $a = typeof WeakSet == "function" ? WeakSet : Set;
  function pi(e, t) {
    var n = t.source, r = t.stack;
    r === null && n !== null && (r = br(n)), n !== null && lt(n.type), t = t.value, e !== null && e.tag === 1 && lt(e.type);
    try {
      console.error(t);
    } catch (l) {
      setTimeout(function() {
        throw l;
      });
    }
  }
  function Ka(e, t) {
    try {
      t.props = e.memoizedProps, t.state = e.memoizedState, t.componentWillUnmount();
    } catch (n) {
      $t(e, n);
    }
  }
  function Xu(e) {
    var t = e.ref;
    if (t !== null)
      if (typeof t == "function")
        try {
          t(null);
        } catch (n) {
          $t(e, n);
        }
      else
        t.current = null;
  }
  function Ba(e, t) {
    switch (t.tag) {
      case 0:
      case 11:
      case 15:
      case 22:
        return;
      case 1:
        if (t.effectTag & 256 && e !== null) {
          var n = e.memoizedProps, r = e.memoizedState;
          e = t.stateNode, t = e.getSnapshotBeforeUpdate(t.elementType === t.type ? n : Ve(t.type, n), r), e.__reactInternalSnapshotBeforeUpdate = t;
        }
        return;
      case 3:
      case 5:
      case 6:
      case 4:
      case 17:
        return;
    }
    throw Error(h(163));
  }
  function Gu(e, t) {
    if (t = t.updateQueue, t = t !== null ? t.lastEffect : null, t !== null) {
      var n = t = t.next;
      do {
        if ((n.tag & e) === e) {
          var r = n.destroy;
          n.destroy = void 0, r !== void 0 && r();
        }
        n = n.next;
      } while (n !== t);
    }
  }
  function Zu(e, t) {
    if (t = t.updateQueue, t = t !== null ? t.lastEffect : null, t !== null) {
      var n = t = t.next;
      do {
        if ((n.tag & e) === e) {
          var r = n.create;
          n.destroy = r();
        }
        n = n.next;
      } while (n !== t);
    }
  }
  function Ya(e, t, n) {
    switch (n.tag) {
      case 0:
      case 11:
      case 15:
      case 22:
        Zu(3, n);
        return;
      case 1:
        if (e = n.stateNode, n.effectTag & 4)
          if (t === null)
            e.componentDidMount();
          else {
            var r = n.elementType === n.type ? t.memoizedProps : Ve(n.type, t.memoizedProps);
            e.componentDidUpdate(r, t.memoizedState, e.__reactInternalSnapshotBeforeUpdate);
          }
        t = n.updateQueue, t !== null && gu(n, t, e);
        return;
      case 3:
        if (t = n.updateQueue, t !== null) {
          if (e = null, n.child !== null)
            switch (n.child.tag) {
              case 5:
                e = n.child.stateNode;
                break;
              case 1:
                e = n.child.stateNode;
            }
          gu(n, t, e);
        }
        return;
      case 5:
        e = n.stateNode, t === null && n.effectTag & 4 && zo(n.type, n.memoizedProps) && e.focus();
        return;
      case 6:
        return;
      case 4:
        return;
      case 12:
        return;
      case 13:
        n.memoizedState === null && (n = n.alternate, n !== null && (n = n.memoizedState, n !== null && (n = n.dehydrated, n !== null && wo(n))));
        return;
      case 19:
      case 17:
      case 20:
      case 21:
        return;
    }
    throw Error(h(163));
  }
  function Ju(e, t, n) {
    switch (typeof Ci == "function" && Ci(t), t.tag) {
      case 0:
      case 11:
      case 14:
      case 15:
      case 22:
        if (e = t.updateQueue, e !== null && (e = e.lastEffect, e !== null)) {
          var r = e.next;
          wt(97 < n ? 97 : n, function() {
            var l = r;
            do {
              var i = l.destroy;
              if (i !== void 0) {
                var o = t;
                try {
                  i();
                } catch (u) {
                  $t(o, u);
                }
              }
              l = l.next;
            } while (l !== r);
          });
        }
        break;
      case 1:
        Xu(t), n = t.stateNode, typeof n.componentWillUnmount == "function" && Ka(t, n);
        break;
      case 5:
        Xu(t);
        break;
      case 4:
        ts(e, t, n);
    }
  }
  function qu(e) {
    var t = e.alternate;
    e.return = null, e.child = null, e.memoizedState = null, e.updateQueue = null, e.dependencies = null, e.alternate = null, e.firstEffect = null, e.lastEffect = null, e.pendingProps = null, e.memoizedProps = null, e.stateNode = null, t !== null && qu(t);
  }
  function bu(e) {
    return e.tag === 5 || e.tag === 3 || e.tag === 4;
  }
  function es(e) {
    e: {
      for (var t = e.return; t !== null; ) {
        if (bu(t)) {
          var n = t;
          break e;
        }
        t = t.return;
      }
      throw Error(h(160));
    }
    switch (t = n.stateNode, n.tag) {
      case 5:
        var r = !1;
        break;
      case 3:
        t = t.containerInfo, r = !0;
        break;
      case 4:
        t = t.containerInfo, r = !0;
        break;
      default:
        throw Error(h(161));
    }
    n.effectTag & 16 && (dn(t, ""), n.effectTag &= -17);
    e:
      t:
        for (n = e; ; ) {
          for (; n.sibling === null; ) {
            if (n.return === null || bu(n.return)) {
              n = null;
              break e;
            }
            n = n.return;
          }
          for (n.sibling.return = n.return, n = n.sibling; n.tag !== 5 && n.tag !== 6 && n.tag !== 18; ) {
            if (n.effectTag & 2 || n.child === null || n.tag === 4)
              continue t;
            n.child.return = n, n = n.child;
          }
          if (!(n.effectTag & 2)) {
            n = n.stateNode;
            break e;
          }
        }
    r ? mi(e, n, t) : hi(e, n, t);
  }
  function mi(e, t, n) {
    var r = e.tag, l = r === 5 || r === 6;
    if (l)
      e = l ? e.stateNode : e.stateNode.instance, t ? n.nodeType === 8 ? n.parentNode.insertBefore(e, t) : n.insertBefore(e, t) : (n.nodeType === 8 ? (t = n.parentNode, t.insertBefore(e, n)) : (t = n, t.appendChild(e)), n = n._reactRootContainer, n != null || t.onclick !== null || (t.onclick = rr));
    else if (r !== 4 && (e = e.child, e !== null))
      for (mi(e, t, n), e = e.sibling; e !== null; )
        mi(e, t, n), e = e.sibling;
  }
  function hi(e, t, n) {
    var r = e.tag, l = r === 5 || r === 6;
    if (l)
      e = l ? e.stateNode : e.stateNode.instance, t ? n.insertBefore(e, t) : n.appendChild(e);
    else if (r !== 4 && (e = e.child, e !== null))
      for (hi(e, t, n), e = e.sibling; e !== null; )
        hi(e, t, n), e = e.sibling;
  }
  function ts(e, t, n) {
    for (var r = t, l = !1, i, o; ; ) {
      if (!l) {
        l = r.return;
        e:
          for (; ; ) {
            if (l === null)
              throw Error(h(160));
            switch (i = l.stateNode, l.tag) {
              case 5:
                o = !1;
                break e;
              case 3:
                i = i.containerInfo, o = !0;
                break e;
              case 4:
                i = i.containerInfo, o = !0;
                break e;
            }
            l = l.return;
          }
        l = !0;
      }
      if (r.tag === 5 || r.tag === 6) {
        e:
          for (var u = e, c = r, d = n, w = c; ; )
            if (Ju(u, w, d), w.child !== null && w.tag !== 4)
              w.child.return = w, w = w.child;
            else {
              if (w === c)
                break e;
              for (; w.sibling === null; ) {
                if (w.return === null || w.return === c)
                  break e;
                w = w.return;
              }
              w.sibling.return = w.return, w = w.sibling;
            }
        o ? (u = i, c = r.stateNode, u.nodeType === 8 ? u.parentNode.removeChild(c) : u.removeChild(c)) : i.removeChild(r.stateNode);
      } else if (r.tag === 4) {
        if (r.child !== null) {
          i = r.stateNode.containerInfo, o = !0, r.child.return = r, r = r.child;
          continue;
        }
      } else if (Ju(e, r, n), r.child !== null) {
        r.child.return = r, r = r.child;
        continue;
      }
      if (r === t)
        break;
      for (; r.sibling === null; ) {
        if (r.return === null || r.return === t)
          return;
        r = r.return, r.tag === 4 && (l = !1);
      }
      r.sibling.return = r.return, r = r.sibling;
    }
  }
  function vi(e, t) {
    switch (t.tag) {
      case 0:
      case 11:
      case 14:
      case 15:
      case 22:
        Gu(3, t);
        return;
      case 1:
        return;
      case 5:
        var n = t.stateNode;
        if (n != null) {
          var r = t.memoizedProps, l = e !== null ? e.memoizedProps : r;
          e = t.type;
          var i = t.updateQueue;
          if (t.updateQueue = null, i !== null) {
            for (n[lr] = r, e === "input" && r.type === "radio" && r.name != null && Yi(n, r), El(e, l), t = El(e, r), l = 0; l < i.length; l += 2) {
              var o = i[l], u = i[l + 1];
              o === "style" ? So(n, u) : o === "dangerouslySetInnerHTML" ? eo(n, u) : o === "children" ? dn(n, u) : Gr(n, o, u, t);
            }
            switch (e) {
              case "input":
                tl(n, r);
                break;
              case "textarea":
                Zi(n, r);
                break;
              case "select":
                t = n._wrapperState.wasMultiple, n._wrapperState.wasMultiple = !!r.multiple, e = r.value, e != null ? Xt(n, !!r.multiple, e, !1) : t !== !!r.multiple && (r.defaultValue != null ? Xt(n, !!r.multiple, r.defaultValue, !0) : Xt(n, !!r.multiple, r.multiple ? [] : "", !1));
            }
          }
        }
        return;
      case 6:
        if (t.stateNode === null)
          throw Error(h(162));
        t.stateNode.nodeValue = t.memoizedProps;
        return;
      case 3:
        t = t.stateNode, t.hydrate && (t.hydrate = !1, wo(t.containerInfo));
        return;
      case 12:
        return;
      case 13:
        if (n = t, t.memoizedState === null ? r = !1 : (r = !0, n = t.child, wi = Ie()), n !== null)
          e:
            for (e = n; ; ) {
              if (e.tag === 5)
                i = e.stateNode, r ? (i = i.style, typeof i.setProperty == "function" ? i.setProperty("display", "none", "important") : i.display = "none") : (i = e.stateNode, l = e.memoizedProps.style, l = l != null && l.hasOwnProperty("display") ? l.display : null, i.style.display = xo("display", l));
              else if (e.tag === 6)
                e.stateNode.nodeValue = r ? "" : e.memoizedProps;
              else if (e.tag === 13 && e.memoizedState !== null && e.memoizedState.dehydrated === null) {
                i = e.child.sibling, i.return = e, e = i;
                continue;
              } else if (e.child !== null) {
                e.child.return = e, e = e.child;
                continue;
              }
              if (e === n)
                break;
              for (; e.sibling === null; ) {
                if (e.return === null || e.return === n)
                  break e;
                e = e.return;
              }
              e.sibling.return = e.return, e = e.sibling;
            }
        ns(t);
        return;
      case 19:
        ns(t);
        return;
      case 17:
        return;
    }
    throw Error(h(163));
  }
  function ns(e) {
    var t = e.updateQueue;
    if (t !== null) {
      e.updateQueue = null;
      var n = e.stateNode;
      n === null && (n = e.stateNode = new $a()), t.forEach(function(r) {
        var l = rf.bind(null, e, r);
        n.has(r) || (n.add(r), r.then(l, l));
      });
    }
  }
  var Xa = typeof WeakMap == "function" ? WeakMap : Map;
  function rs(e, t, n) {
    n = Tt(n, null), n.tag = 3, n.payload = { element: null };
    var r = t.value;
    return n.callback = function() {
      Vr || (Vr = !0, Ei = r), pi(e, t);
    }, n;
  }
  function ls(e, t, n) {
    n = Tt(n, null), n.tag = 3;
    var r = e.type.getDerivedStateFromError;
    if (typeof r == "function") {
      var l = t.value;
      n.payload = function() {
        return pi(e, t), r(l);
      };
    }
    var i = e.stateNode;
    return i !== null && typeof i.componentDidCatch == "function" && (n.callback = function() {
      typeof r != "function" && (Ct === null ? Ct = /* @__PURE__ */ new Set([this]) : Ct.add(this), pi(e, t));
      var o = t.stack;
      this.componentDidCatch(t.value, { componentStack: o !== null ? o : "" });
    }), n;
  }
  var Ga = Math.ceil, Ir = Ae.ReactCurrentDispatcher, is = Ae.ReactCurrentOwner, te = 0, yi = 8, We = 16, Ze = 32, At = 0, jr = 1, os = 2, Fr = 3, Lr = 4, gi = 5, N = te, Pe = null, M = null, ye = 0, ne = At, Dr = null, ct = 1073741823, An = 1073741823, Ur = null, Vn = 0, Ar = !1, wi = 0, us = 500, x = null, Vr = !1, Ei = null, Ct = null, Wr = !1, Wn = null, Qn = 90, Vt = null, Hn = 0, Ti = null, Qr = 0;
  function Je() {
    return (N & (We | Ze)) !== te ? 1073741821 - (Ie() / 10 | 0) : Qr !== 0 ? Qr : Qr = 1073741821 - (Ie() / 10 | 0);
  }
  function Wt(e, t, n) {
    if (t = t.mode, !(t & 2))
      return 1073741823;
    var r = mr();
    if (!(t & 4))
      return r === 99 ? 1073741823 : 1073741822;
    if ((N & We) !== te)
      return ye;
    if (n !== null)
      e = hr(e, n.timeoutMs | 0 || 5e3, 250);
    else
      switch (r) {
        case 99:
          e = 1073741823;
          break;
        case 98:
          e = hr(e, 150, 100);
          break;
        case 97:
        case 96:
          e = hr(e, 5e3, 250);
          break;
        case 95:
          e = 2;
          break;
        default:
          throw Error(h(326));
      }
    return Pe !== null && e === ye && --e, e;
  }
  function _t(e, t) {
    if (50 < Hn)
      throw Hn = 0, Ti = null, Error(h(185));
    if (e = Hr(e, t), e !== null) {
      var n = mr();
      t === 1073741823 ? (N & yi) !== te && (N & (We | Ze)) === te ? ki(e) : (Ne(e), N === te && Ye()) : Ne(e), (N & 4) === te || n !== 98 && n !== 99 || (Vt === null ? Vt = /* @__PURE__ */ new Map([[e, t]]) : (n = Vt.get(e), (n === void 0 || n > t) && Vt.set(e, t)));
    }
  }
  function Hr(e, t) {
    e.expirationTime < t && (e.expirationTime = t);
    var n = e.alternate;
    n !== null && n.expirationTime < t && (n.expirationTime = t);
    var r = e.return, l = null;
    if (r === null && e.tag === 3)
      l = e.stateNode;
    else
      for (; r !== null; ) {
        if (n = r.alternate, r.childExpirationTime < t && (r.childExpirationTime = t), n !== null && n.childExpirationTime < t && (n.childExpirationTime = t), r.return === null && r.tag === 3) {
          l = r.stateNode;
          break;
        }
        r = r.return;
      }
    return l !== null && (Pe === l && (Kr(t), ne === Lr && Bt(l, ye)), ws(l, t)), l;
  }
  function $r(e) {
    var t = e.lastExpiredTime;
    if (t !== 0 || (t = e.firstPendingTime, !gs(e, t)))
      return t;
    var n = e.lastPingedTime;
    return e = e.nextKnownPendingLevel, e = n > e ? n : e, 2 >= e && t !== e ? 0 : e;
  }
  function Ne(e) {
    if (e.lastExpiredTime !== 0)
      e.callbackExpirationTime = 1073741823, e.callbackPriority = 99, e.callbackNode = mu(ki.bind(null, e));
    else {
      var t = $r(e), n = e.callbackNode;
      if (t === 0)
        n !== null && (e.callbackNode = null, e.callbackExpirationTime = 0, e.callbackPriority = 90);
      else {
        var r = Je();
        if (t === 1073741823 ? r = 99 : t === 1 || t === 2 ? r = 95 : (r = 10 * (1073741821 - t) - 10 * (1073741821 - r), r = 0 >= r ? 99 : 250 >= r ? 98 : 5250 >= r ? 97 : 95), n !== null) {
          var l = e.callbackPriority;
          if (e.callbackExpirationTime === t && l >= r)
            return;
          n !== fu && lu(n);
        }
        e.callbackExpirationTime = t, e.callbackPriority = r, t = t === 1073741823 ? mu(ki.bind(null, e)) : pu(r, ss.bind(null, e), { timeout: 10 * (1073741821 - t) - Ie() }), e.callbackNode = t;
      }
    }
  }
  function ss(e, t) {
    if (Qr = 0, t)
      return t = Je(), Oi(e, t), Ne(e), null;
    var n = $r(e);
    if (n !== 0) {
      if (t = e.callbackNode, (N & (We | Ze)) !== te)
        throw Error(h(327));
      if (fn(), e === Pe && n === ye || Qt(e, n), M !== null) {
        var r = N;
        N |= We;
        var l = ds();
        do
          try {
            qa();
            break;
          } catch (u) {
            cs(e, u);
          }
        while (!0);
        if (Ql(), N = r, Ir.current = l, ne === jr)
          throw t = Dr, Qt(e, n), Bt(e, n), Ne(e), t;
        if (M === null)
          switch (l = e.finishedWork = e.current.alternate, e.finishedExpirationTime = n, r = ne, Pe = null, r) {
            case At:
            case jr:
              throw Error(h(345));
            case os:
              Oi(e, 2 < n ? 2 : n);
              break;
            case Fr:
              if (Bt(e, n), r = e.lastSuspendedTime, n === r && (e.nextKnownPendingLevel = xi(l)), ct === 1073741823 && (l = wi + us - Ie(), 10 < l)) {
                if (Ar) {
                  var i = e.lastPingedTime;
                  if (i === 0 || i >= n) {
                    e.lastPingedTime = n, Qt(e, n);
                    break;
                  }
                }
                if (i = $r(e), i !== 0 && i !== n)
                  break;
                if (r !== 0 && r !== n) {
                  e.lastPingedTime = r;
                  break;
                }
                e.timeoutHandle = Nl(Ht.bind(null, e), l);
                break;
              }
              Ht(e);
              break;
            case Lr:
              if (Bt(e, n), r = e.lastSuspendedTime, n === r && (e.nextKnownPendingLevel = xi(l)), Ar && (l = e.lastPingedTime, l === 0 || l >= n)) {
                e.lastPingedTime = n, Qt(e, n);
                break;
              }
              if (l = $r(e), l !== 0 && l !== n)
                break;
              if (r !== 0 && r !== n) {
                e.lastPingedTime = r;
                break;
              }
              if (An !== 1073741823 ? r = 10 * (1073741821 - An) - Ie() : ct === 1073741823 ? r = 0 : (r = 10 * (1073741821 - ct) - 5e3, l = Ie(), n = 10 * (1073741821 - n) - l, r = l - r, 0 > r && (r = 0), r = (120 > r ? 120 : 480 > r ? 480 : 1080 > r ? 1080 : 1920 > r ? 1920 : 3e3 > r ? 3e3 : 4320 > r ? 4320 : 1960 * Ga(r / 1960)) - r, n < r && (r = n)), 10 < r) {
                e.timeoutHandle = Nl(Ht.bind(null, e), r);
                break;
              }
              Ht(e);
              break;
            case gi:
              if (ct !== 1073741823 && Ur !== null) {
                i = ct;
                var o = Ur;
                if (r = o.busyMinDurationMs | 0, 0 >= r ? r = 0 : (l = o.busyDelayMs | 0, i = Ie() - (10 * (1073741821 - i) - (o.timeoutMs | 0 || 5e3)), r = i <= l ? 0 : l + r - i), 10 < r) {
                  Bt(e, n), e.timeoutHandle = Nl(Ht.bind(null, e), r);
                  break;
                }
              }
              Ht(e);
              break;
            default:
              throw Error(h(329));
          }
        if (Ne(e), e.callbackNode === t)
          return ss.bind(null, e);
      }
    }
    return null;
  }
  function ki(e) {
    var t = e.lastExpiredTime;
    if (t = t !== 0 ? t : 1073741823, (N & (We | Ze)) !== te)
      throw Error(h(327));
    if (fn(), e === Pe && t === ye || Qt(e, t), M !== null) {
      var n = N;
      N |= We;
      var r = ds();
      do
        try {
          Ja();
          break;
        } catch (l) {
          cs(e, l);
        }
      while (!0);
      if (Ql(), N = n, Ir.current = r, ne === jr)
        throw n = Dr, Qt(e, t), Bt(e, t), Ne(e), n;
      if (M !== null)
        throw Error(h(261));
      e.finishedWork = e.current.alternate, e.finishedExpirationTime = t, Pe = null, Ht(e), Ne(e);
    }
    return null;
  }
  function Za() {
    if (Vt !== null) {
      var e = Vt;
      Vt = null, e.forEach(function(t, n) {
        Oi(n, t), Ne(n);
      }), Ye();
    }
  }
  function as(e, t) {
    var n = N;
    N |= 1;
    try {
      return e(t);
    } finally {
      N = n, N === te && Ye();
    }
  }
  function fs(e, t) {
    var n = N;
    N &= -2, N |= yi;
    try {
      return e(t);
    } finally {
      N = n, N === te && Ye();
    }
  }
  function Qt(e, t) {
    e.finishedWork = null, e.finishedExpirationTime = 0;
    var n = e.timeoutHandle;
    if (n !== -1 && (e.timeoutHandle = -1, Gs(n)), M !== null)
      for (n = M.return; n !== null; ) {
        var r = n;
        switch (r.tag) {
          case 1:
            r = r.type.childContextTypes, r != null && fr();
            break;
          case 3:
            un(), V(he), V(oe);
            break;
          case 5:
            Gl(r);
            break;
          case 4:
            un();
            break;
          case 13:
            V(K);
            break;
          case 19:
            V(K);
            break;
          case 10:
            Hl(r);
        }
        n = n.return;
      }
    Pe = e, M = Kt(e.current, null), ye = t, ne = At, Dr = null, An = ct = 1073741823, Ur = null, Vn = 0, Ar = !1;
  }
  function cs(e, t) {
    do {
      try {
        if (Ql(), Sr.current = Rr, Cr)
          for (var n = Z.memoizedState; n !== null; ) {
            var r = n.queue;
            r !== null && (r.pending = null), n = n.next;
          }
        if (xt = 0, se = ue = Z = null, Cr = !1, M === null || M.return === null)
          return ne = jr, Dr = t, M = null;
        e: {
          var l = e, i = M.return, o = M, u = t;
          if (t = ye, o.effectTag |= 2048, o.firstEffect = o.lastEffect = null, u !== null && typeof u == "object" && typeof u.then == "function") {
            var c = u;
            if (!(o.mode & 2)) {
              var d = o.alternate;
              d ? (o.updateQueue = d.updateQueue, o.memoizedState = d.memoizedState, o.expirationTime = d.expirationTime) : (o.updateQueue = null, o.memoizedState = null);
            }
            var w = (K.current & 1) !== 0, T = i;
            do {
              var j;
              if (j = T.tag === 13) {
                var D = T.memoizedState;
                if (D !== null)
                  j = D.dehydrated !== null;
                else {
                  var Le = T.memoizedProps;
                  j = Le.fallback === void 0 ? !1 : Le.unstable_avoidThisFallback !== !0 ? !0 : !w;
                }
              }
              if (j) {
                var ie = T.updateQueue;
                if (ie === null) {
                  var f = /* @__PURE__ */ new Set();
                  f.add(c), T.updateQueue = f;
                } else
                  ie.add(c);
                if (!(T.mode & 2)) {
                  if (T.effectTag |= 64, o.effectTag &= -2981, o.tag === 1)
                    if (o.alternate === null)
                      o.tag = 17;
                    else {
                      var a = Tt(1073741823, null);
                      a.tag = 2, kt(o, a);
                    }
                  o.expirationTime = 1073741823;
                  break e;
                }
                u = void 0, o = t;
                var m = l.pingCache;
                if (m === null ? (m = l.pingCache = new Xa(), u = /* @__PURE__ */ new Set(), m.set(c, u)) : (u = m.get(c), u === void 0 && (u = /* @__PURE__ */ new Set(), m.set(c, u))), !u.has(o)) {
                  u.add(o);
                  var v = nf.bind(null, l, c, o);
                  c.then(v, v);
                }
                T.effectTag |= 4096, T.expirationTime = t;
                break e;
              }
              T = T.return;
            } while (T !== null);
            u = Error((lt(o.type) || "A React component") + ` suspended while rendering, but no fallback UI was specified.

Add a <Suspense fallback=...> component higher in the tree to provide a loading indicator or placeholder to display.` + br(o));
          }
          ne !== gi && (ne = os), u = di(u, o), T = i;
          do {
            switch (T.tag) {
              case 3:
                c = u, T.effectTag |= 4096, T.expirationTime = t;
                var g = rs(T, c, t);
                yu(T, g);
                break e;
              case 1:
                c = u;
                var k = T.type, _ = T.stateNode;
                if (!(T.effectTag & 64) && (typeof k.getDerivedStateFromError == "function" || _ !== null && typeof _.componentDidCatch == "function" && (Ct === null || !Ct.has(_)))) {
                  T.effectTag |= 4096, T.expirationTime = t;
                  var L = ls(T, c, t);
                  yu(T, L);
                  break e;
                }
            }
            T = T.return;
          } while (T !== null);
        }
        M = hs(M);
      } catch (Q) {
        t = Q;
        continue;
      }
      break;
    } while (!0);
  }
  function ds() {
    var e = Ir.current;
    return Ir.current = Rr, e === null ? Rr : e;
  }
  function ps(e, t) {
    e < ct && 2 < e && (ct = e), t !== null && e < An && 2 < e && (An = e, Ur = t);
  }
  function Kr(e) {
    e > Vn && (Vn = e);
  }
  function Ja() {
    for (; M !== null; )
      M = ms(M);
  }
  function qa() {
    for (; M !== null && !La(); )
      M = ms(M);
  }
  function ms(e) {
    var t = ys(e.alternate, e, ye);
    return e.memoizedProps = e.pendingProps, t === null && (t = hs(e)), is.current = null, t;
  }
  function hs(e) {
    M = e;
    do {
      var t = M.alternate;
      if (e = M.return, M.effectTag & 2048) {
        if (t = Ha(M), t !== null)
          return t.effectTag &= 2047, t;
        e !== null && (e.firstEffect = e.lastEffect = null, e.effectTag |= 2048);
      } else {
        if (t = Qa(t, M, ye), ye === 1 || M.childExpirationTime !== 1) {
          for (var n = 0, r = M.child; r !== null; ) {
            var l = r.expirationTime, i = r.childExpirationTime;
            l > n && (n = l), i > n && (n = i), r = r.sibling;
          }
          M.childExpirationTime = n;
        }
        if (t !== null)
          return t;
        e !== null && !(e.effectTag & 2048) && (e.firstEffect === null && (e.firstEffect = M.firstEffect), M.lastEffect !== null && (e.lastEffect !== null && (e.lastEffect.nextEffect = M.firstEffect), e.lastEffect = M.lastEffect), 1 < M.effectTag && (e.lastEffect !== null ? e.lastEffect.nextEffect = M : e.firstEffect = M, e.lastEffect = M));
      }
      if (t = M.sibling, t !== null)
        return t;
      M = e;
    } while (M !== null);
    return ne === At && (ne = gi), null;
  }
  function xi(e) {
    var t = e.expirationTime;
    return e = e.childExpirationTime, t > e ? t : e;
  }
  function Ht(e) {
    var t = mr();
    return wt(99, ba.bind(null, e, t)), null;
  }
  function ba(e, t) {
    do
      fn();
    while (Wn !== null);
    if ((N & (We | Ze)) !== te)
      throw Error(h(327));
    var n = e.finishedWork, r = e.finishedExpirationTime;
    if (n === null)
      return null;
    if (e.finishedWork = null, e.finishedExpirationTime = 0, n === e.current)
      throw Error(h(177));
    e.callbackNode = null, e.callbackExpirationTime = 0, e.callbackPriority = 90, e.nextKnownPendingLevel = 0;
    var l = xi(n);
    if (e.firstPendingTime = l, r <= e.lastSuspendedTime ? e.firstSuspendedTime = e.lastSuspendedTime = e.nextKnownPendingLevel = 0 : r <= e.firstSuspendedTime && (e.firstSuspendedTime = r - 1), r <= e.lastPingedTime && (e.lastPingedTime = 0), r <= e.lastExpiredTime && (e.lastExpiredTime = 0), e === Pe && (M = Pe = null, ye = 0), 1 < n.effectTag ? n.lastEffect !== null ? (n.lastEffect.nextEffect = n, l = n.firstEffect) : l = n : l = n.firstEffect, l !== null) {
      var i = N;
      N |= Ze, is.current = null, Cl = tr;
      var o = Oo();
      if (kl(o)) {
        if ("selectionStart" in o)
          var u = { start: o.selectionStart, end: o.selectionEnd };
        else
          e: {
            u = (u = o.ownerDocument) && u.defaultView || window;
            var c = u.getSelection && u.getSelection();
            if (c && c.rangeCount !== 0) {
              u = c.anchorNode;
              var d = c.anchorOffset, w = c.focusNode;
              c = c.focusOffset;
              try {
                u.nodeType, w.nodeType;
              } catch {
                u = null;
                break e;
              }
              var T = 0, j = -1, D = -1, Le = 0, ie = 0, f = o, a = null;
              t:
                for (; ; ) {
                  for (var m; f !== u || d !== 0 && f.nodeType !== 3 || (j = T + d), f !== w || c !== 0 && f.nodeType !== 3 || (D = T + c), f.nodeType === 3 && (T += f.nodeValue.length), (m = f.firstChild) !== null; )
                    a = f, f = m;
                  for (; ; ) {
                    if (f === o)
                      break t;
                    if (a === u && ++Le === d && (j = T), a === w && ++ie === c && (D = T), (m = f.nextSibling) !== null)
                      break;
                    f = a, a = f.parentNode;
                  }
                  f = m;
                }
              u = j === -1 || D === -1 ? null : { start: j, end: D };
            } else
              u = null;
          }
        u = u || { start: 0, end: 0 };
      } else
        u = null;
      _l = { activeElementDetached: null, focusedElem: o, selectionRange: u }, tr = !1, x = l;
      do
        try {
          ef();
        } catch (z) {
          if (x === null)
            throw Error(h(330));
          $t(x, z), x = x.nextEffect;
        }
      while (x !== null);
      x = l;
      do
        try {
          for (o = e, u = t; x !== null; ) {
            var v = x.effectTag;
            if (v & 16 && dn(x.stateNode, ""), v & 128) {
              var g = x.alternate;
              if (g !== null) {
                var k = g.ref;
                k !== null && (typeof k == "function" ? k(null) : k.current = null);
              }
            }
            switch (v & 1038) {
              case 2:
                es(x), x.effectTag &= -3;
                break;
              case 6:
                es(x), x.effectTag &= -3, vi(x.alternate, x);
                break;
              case 1024:
                x.effectTag &= -1025;
                break;
              case 1028:
                x.effectTag &= -1025, vi(x.alternate, x);
                break;
              case 4:
                vi(x.alternate, x);
                break;
              case 8:
                d = x, ts(o, d, u), qu(d);
            }
            x = x.nextEffect;
          }
        } catch (z) {
          if (x === null)
            throw Error(h(330));
          $t(x, z), x = x.nextEffect;
        }
      while (x !== null);
      if (k = _l, g = Oo(), v = k.focusedElem, u = k.selectionRange, g !== v && v && v.ownerDocument && No(v.ownerDocument.documentElement, v)) {
        for (u !== null && kl(v) && (g = u.start, k = u.end, k === void 0 && (k = g), "selectionStart" in v ? (v.selectionStart = g, v.selectionEnd = Math.min(k, v.value.length)) : (k = (g = v.ownerDocument || document) && g.defaultView || window, k.getSelection && (k = k.getSelection(), d = v.textContent.length, o = Math.min(u.start, d), u = u.end === void 0 ? o : Math.min(u.end, d), !k.extend && o > u && (d = u, u = o, o = d), d = Po(v, o), w = Po(v, u), d && w && (k.rangeCount !== 1 || k.anchorNode !== d.node || k.anchorOffset !== d.offset || k.focusNode !== w.node || k.focusOffset !== w.offset) && (g = g.createRange(), g.setStart(d.node, d.offset), k.removeAllRanges(), o > u ? (k.addRange(g), k.extend(w.node, w.offset)) : (g.setEnd(w.node, w.offset), k.addRange(g)))))), g = [], k = v; k = k.parentNode; )
          k.nodeType === 1 && g.push({
            element: k,
            left: k.scrollLeft,
            top: k.scrollTop
          });
        for (typeof v.focus == "function" && v.focus(), v = 0; v < g.length; v++)
          k = g[v], k.element.scrollLeft = k.left, k.element.scrollTop = k.top;
      }
      tr = !!Cl, _l = Cl = null, e.current = n, x = l;
      do
        try {
          for (v = e; x !== null; ) {
            var _ = x.effectTag;
            if (_ & 36 && Ya(v, x.alternate, x), _ & 128) {
              g = void 0;
              var L = x.ref;
              if (L !== null) {
                var Q = x.stateNode;
                switch (x.tag) {
                  case 5:
                    g = Q;
                    break;
                  default:
                    g = Q;
                }
                typeof L == "function" ? L(g) : L.current = g;
              }
            }
            x = x.nextEffect;
          }
        } catch (z) {
          if (x === null)
            throw Error(h(330));
          $t(x, z), x = x.nextEffect;
        }
      while (x !== null);
      x = null, Da(), N = i;
    } else
      e.current = n;
    if (Wr)
      Wr = !1, Wn = e, Qn = t;
    else
      for (x = l; x !== null; )
        t = x.nextEffect, x.nextEffect = null, x = t;
    if (t = e.firstPendingTime, t === 0 && (Ct = null), t === 1073741823 ? e === Ti ? Hn++ : (Hn = 0, Ti = e) : Hn = 0, typeof Si == "function" && Si(n.stateNode, r), Ne(e), Vr)
      throw Vr = !1, e = Ei, Ei = null, e;
    return (N & yi) !== te || Ye(), null;
  }
  function ef() {
    for (; x !== null; ) {
      var e = x.effectTag;
      e & 256 && Ba(x.alternate, x), !(e & 512) || Wr || (Wr = !0, pu(97, function() {
        return fn(), null;
      })), x = x.nextEffect;
    }
  }
  function fn() {
    if (Qn !== 90) {
      var e = 97 < Qn ? 97 : Qn;
      return Qn = 90, wt(e, tf);
    }
  }
  function tf() {
    if (Wn === null)
      return !1;
    var e = Wn;
    if (Wn = null, (N & (We | Ze)) !== te)
      throw Error(h(331));
    var t = N;
    for (N |= Ze, e = e.current.firstEffect; e !== null; ) {
      try {
        var n = e;
        if (n.effectTag & 512)
          switch (n.tag) {
            case 0:
            case 11:
            case 15:
            case 22:
              Gu(5, n), Zu(5, n);
          }
      } catch (r) {
        if (e === null)
          throw Error(h(330));
        $t(e, r);
      }
      n = e.nextEffect, e.nextEffect = null, e = n;
    }
    return N = t, Ye(), !0;
  }
  function vs(e, t, n) {
    t = di(n, t), t = rs(e, t, 1073741823), kt(e, t), e = Hr(e, 1073741823), e !== null && Ne(e);
  }
  function $t(e, t) {
    if (e.tag === 3)
      vs(e, e, t);
    else
      for (var n = e.return; n !== null; ) {
        if (n.tag === 3) {
          vs(n, e, t);
          break;
        } else if (n.tag === 1) {
          var r = n.stateNode;
          if (typeof n.type.getDerivedStateFromError == "function" || typeof r.componentDidCatch == "function" && (Ct === null || !Ct.has(r))) {
            e = di(t, e), e = ls(n, e, 1073741823), kt(n, e), n = Hr(n, 1073741823), n !== null && Ne(n);
            break;
          }
        }
        n = n.return;
      }
  }
  function nf(e, t, n) {
    var r = e.pingCache;
    r !== null && r.delete(t), Pe === e && ye === n ? ne === Lr || ne === Fr && ct === 1073741823 && Ie() - wi < us ? Qt(e, ye) : Ar = !0 : gs(e, n) && (t = e.lastPingedTime, t !== 0 && t < n || (e.lastPingedTime = n, Ne(e)));
  }
  function rf(e, t) {
    var n = e.stateNode;
    n !== null && n.delete(t), t = 0, t === 0 && (t = Je(), t = Wt(t, e, null)), e = Hr(e, t), e !== null && Ne(e);
  }
  var ys;
  ys = function(e, t, n) {
    var r = t.expirationTime;
    if (e !== null) {
      var l = t.pendingProps;
      if (e.memoizedProps !== l || he.current)
        Ge = !0;
      else {
        if (r < n) {
          switch (Ge = !1, t.tag) {
            case 3:
              Wu(t), oi();
              break;
            case 5:
              if (Su(t), t.mode & 4 && n !== 1 && l.hidden)
                return t.expirationTime = t.childExpirationTime = 1, null;
              break;
            case 1:
              ve(t.type) && cr(t);
              break;
            case 4:
              Xl(t, t.stateNode.containerInfo);
              break;
            case 10:
              r = t.memoizedProps.value, l = t.type._context, Y(vr, l._currentValue), l._currentValue = r;
              break;
            case 13:
              if (t.memoizedState !== null)
                return r = t.child.childExpirationTime, r !== 0 && r >= n ? Qu(e, t, n) : (Y(K, K.current & 1), t = ft(e, t, n), t !== null ? t.sibling : null);
              Y(K, K.current & 1);
              break;
            case 19:
              if (r = t.childExpirationTime >= n, e.effectTag & 64) {
                if (r)
                  return $u(e, t, n);
                t.effectTag |= 64;
              }
              if (l = t.memoizedState, l !== null && (l.rendering = null, l.tail = null), Y(K, K.current), !r)
                return null;
          }
          return ft(e, t, n);
        }
        Ge = !1;
      }
    } else
      Ge = !1;
    switch (t.expirationTime = 0, t.tag) {
      case 2:
        if (r = t.type, e !== null && (e.alternate = null, t.alternate = null, t.effectTag |= 2), e = t.pendingProps, l = nn(t, oe.current), ln(t, n), l = ql(
          null,
          t,
          r,
          e,
          l,
          n
        ), t.effectTag |= 1, typeof l == "object" && l !== null && typeof l.render == "function" && l.$$typeof === void 0) {
          if (t.tag = 1, t.memoizedState = null, t.updateQueue = null, ve(r)) {
            var i = !0;
            cr(t);
          } else
            i = !1;
          t.memoizedState = l.state !== null && l.state !== void 0 ? l.state : null, $l(t);
          var o = r.getDerivedStateFromProps;
          typeof o == "function" && wr(t, r, o, e), l.updater = Er, t.stateNode = l, l._reactInternalFiber = t, Bl(t, r, e, n), t = si(null, t, r, !0, i, n);
        } else
          t.tag = 0, _e(null, t, l, n), t = t.child;
        return t;
      case 16:
        e: {
          if (l = t.elementType, e !== null && (e.alternate = null, t.alternate = null, t.effectTag |= 2), e = t.pendingProps, zs(l), l._status !== 1)
            throw l._result;
          switch (l = l._result, t.type = l, i = t.tag = uf(l), e = Ve(l, e), i) {
            case 0:
              t = ui(null, t, l, e, n);
              break e;
            case 1:
              t = Vu(null, t, l, e, n);
              break e;
            case 11:
              t = Lu(null, t, l, e, n);
              break e;
            case 14:
              t = Du(null, t, l, Ve(l.type, e), r, n);
              break e;
          }
          throw Error(h(306, l, ""));
        }
        return t;
      case 0:
        return r = t.type, l = t.pendingProps, l = t.elementType === r ? l : Ve(r, l), ui(e, t, r, l, n);
      case 1:
        return r = t.type, l = t.pendingProps, l = t.elementType === r ? l : Ve(r, l), Vu(e, t, r, l, n);
      case 3:
        if (Wu(t), r = t.updateQueue, e === null || r === null)
          throw Error(h(282));
        if (r = t.pendingProps, l = t.memoizedState, l = l !== null ? l.element : null, Kl(e, t), In(t, r, null, n), r = t.memoizedState.element, r === l)
          oi(), t = ft(e, t, n);
        else {
          if ((l = t.stateNode.hydrate) && (St = Jt(t.stateNode.containerInfo.firstChild), at = t, l = Ut = !0), l)
            for (n = Yl(t, null, r, n), t.child = n; n; )
              n.effectTag = n.effectTag & -3 | 1024, n = n.sibling;
          else
            _e(e, t, r, n), oi();
          t = t.child;
        }
        return t;
      case 5:
        return Su(t), e === null && ii(t), r = t.type, l = t.pendingProps, i = e !== null ? e.memoizedProps : null, o = l.children, Pl(r, l) ? o = null : i !== null && Pl(r, i) && (t.effectTag |= 16), Au(e, t), t.mode & 4 && n !== 1 && l.hidden ? (t.expirationTime = t.childExpirationTime = 1, t = null) : (_e(e, t, o, n), t = t.child), t;
      case 6:
        return e === null && ii(t), null;
      case 13:
        return Qu(e, t, n);
      case 4:
        return Xl(t, t.stateNode.containerInfo), r = t.pendingProps, e === null ? t.child = on(t, null, r, n) : _e(e, t, r, n), t.child;
      case 11:
        return r = t.type, l = t.pendingProps, l = t.elementType === r ? l : Ve(r, l), Lu(e, t, r, l, n);
      case 7:
        return _e(e, t, t.pendingProps, n), t.child;
      case 8:
        return _e(
          e,
          t,
          t.pendingProps.children,
          n
        ), t.child;
      case 12:
        return _e(e, t, t.pendingProps.children, n), t.child;
      case 10:
        e: {
          r = t.type._context, l = t.pendingProps, o = t.memoizedProps, i = l.value;
          var u = t.type._context;
          if (Y(vr, u._currentValue), u._currentValue = i, o !== null)
            if (u = o.value, i = jt(u, i) ? 0 : (typeof r._calculateChangedBits == "function" ? r._calculateChangedBits(u, i) : 1073741823) | 0, i === 0) {
              if (o.children === l.children && !he.current) {
                t = ft(e, t, n);
                break e;
              }
            } else
              for (u = t.child, u !== null && (u.return = t); u !== null; ) {
                var c = u.dependencies;
                if (c !== null) {
                  o = u.child;
                  for (var d = c.firstContext; d !== null; ) {
                    if (d.context === r && d.observedBits & i) {
                      u.tag === 1 && (d = Tt(n, null), d.tag = 2, kt(u, d)), u.expirationTime < n && (u.expirationTime = n), d = u.alternate, d !== null && d.expirationTime < n && (d.expirationTime = n), vu(u.return, n), c.expirationTime < n && (c.expirationTime = n);
                      break;
                    }
                    d = d.next;
                  }
                } else
                  o = u.tag === 10 && u.type === t.type ? null : u.child;
                if (o !== null)
                  o.return = u;
                else
                  for (o = u; o !== null; ) {
                    if (o === t) {
                      o = null;
                      break;
                    }
                    if (u = o.sibling, u !== null) {
                      u.return = o.return, o = u;
                      break;
                    }
                    o = o.return;
                  }
                u = o;
              }
          _e(e, t, l.children, n), t = t.child;
        }
        return t;
      case 9:
        return l = t.type, i = t.pendingProps, r = i.children, ln(t, n), l = je(l, i.unstable_observedBits), r = r(l), t.effectTag |= 1, _e(e, t, r, n), t.child;
      case 14:
        return l = t.type, i = Ve(l, t.pendingProps), i = Ve(l.type, i), Du(e, t, l, i, r, n);
      case 15:
        return Uu(e, t, t.type, t.pendingProps, r, n);
      case 17:
        return r = t.type, l = t.pendingProps, l = t.elementType === r ? l : Ve(r, l), e !== null && (e.alternate = null, t.alternate = null, t.effectTag |= 2), t.tag = 1, ve(r) ? (e = !0, cr(t)) : e = !1, ln(t, n), Tu(t, r, l), Bl(t, r, l, n), si(
          null,
          t,
          r,
          !0,
          e,
          n
        );
      case 19:
        return $u(e, t, n);
    }
    throw Error(h(156, t.tag));
  };
  var Si = null, Ci = null;
  function lf(e) {
    if (typeof __REACT_DEVTOOLS_GLOBAL_HOOK__ > "u")
      return !1;
    var t = __REACT_DEVTOOLS_GLOBAL_HOOK__;
    if (t.isDisabled || !t.supportsFiber)
      return !0;
    try {
      var n = t.inject(e);
      Si = function(r) {
        try {
          t.onCommitFiberRoot(n, r, void 0, (r.current.effectTag & 64) === 64);
        } catch {
        }
      }, Ci = function(r) {
        try {
          t.onCommitFiberUnmount(n, r);
        } catch {
        }
      };
    } catch {
    }
    return !0;
  }
  function of(e, t, n, r) {
    this.tag = e, this.key = n, this.sibling = this.child = this.return = this.stateNode = this.type = this.elementType = null, this.index = 0, this.ref = null, this.pendingProps = t, this.dependencies = this.memoizedState = this.updateQueue = this.memoizedProps = null, this.mode = r, this.effectTag = 0, this.lastEffect = this.firstEffect = this.nextEffect = null, this.childExpirationTime = this.expirationTime = 0, this.alternate = null;
  }
  function qe(e, t, n, r) {
    return new of(e, t, n, r);
  }
  function _i(e) {
    return e = e.prototype, !(!e || !e.isReactComponent);
  }
  function uf(e) {
    if (typeof e == "function")
      return _i(e) ? 1 : 0;
    if (e != null) {
      if (e = e.$$typeof, e === Zr)
        return 11;
      if (e === qr)
        return 14;
    }
    return 2;
  }
  function Kt(e, t) {
    var n = e.alternate;
    return n === null ? (n = qe(e.tag, t, e.key, e.mode), n.elementType = e.elementType, n.type = e.type, n.stateNode = e.stateNode, n.alternate = e, e.alternate = n) : (n.pendingProps = t, n.effectTag = 0, n.nextEffect = null, n.firstEffect = null, n.lastEffect = null), n.childExpirationTime = e.childExpirationTime, n.expirationTime = e.expirationTime, n.child = e.child, n.memoizedProps = e.memoizedProps, n.memoizedState = e.memoizedState, n.updateQueue = e.updateQueue, t = e.dependencies, n.dependencies = t === null ? null : {
      expirationTime: t.expirationTime,
      firstContext: t.firstContext,
      responders: t.responders
    }, n.sibling = e.sibling, n.index = e.index, n.ref = e.ref, n;
  }
  function Br(e, t, n, r, l, i) {
    var o = 2;
    if (r = e, typeof e == "function")
      _i(e) && (o = 1);
    else if (typeof e == "string")
      o = 5;
    else
      e:
        switch (e) {
          case Mt:
            return Pt(n.children, l, i, t);
          case Ms:
            o = 8, l |= 7;
            break;
          case Ui:
            o = 8, l |= 1;
            break;
          case Bn:
            return e = qe(12, n, t, l | 8), e.elementType = Bn, e.type = Bn, e.expirationTime = i, e;
          case Yn:
            return e = qe(13, n, t, l), e.type = Yn, e.elementType = Yn, e.expirationTime = i, e;
          case Jr:
            return e = qe(19, n, t, l), e.elementType = Jr, e.expirationTime = i, e;
          default:
            if (typeof e == "object" && e !== null)
              switch (e.$$typeof) {
                case Ai:
                  o = 10;
                  break e;
                case Vi:
                  o = 9;
                  break e;
                case Zr:
                  o = 11;
                  break e;
                case qr:
                  o = 14;
                  break e;
                case Wi:
                  o = 16, r = null;
                  break e;
                case Qi:
                  o = 22;
                  break e;
              }
            throw Error(h(130, e == null ? e : typeof e, ""));
        }
    return t = qe(o, n, t, l), t.elementType = e, t.type = r, t.expirationTime = i, t;
  }
  function Pt(e, t, n, r) {
    return e = qe(7, e, r, t), e.expirationTime = n, e;
  }
  function Pi(e, t, n) {
    return e = qe(6, e, null, t), e.expirationTime = n, e;
  }
  function Ni(e, t, n) {
    return t = qe(4, e.children !== null ? e.children : [], e.key, t), t.expirationTime = n, t.stateNode = { containerInfo: e.containerInfo, pendingChildren: null, implementation: e.implementation }, t;
  }
  function sf(e, t, n) {
    this.tag = t, this.current = null, this.containerInfo = e, this.pingCache = this.pendingChildren = null, this.finishedExpirationTime = 0, this.finishedWork = null, this.timeoutHandle = -1, this.pendingContext = this.context = null, this.hydrate = n, this.callbackNode = null, this.callbackPriority = 90, this.lastExpiredTime = this.lastPingedTime = this.nextKnownPendingLevel = this.lastSuspendedTime = this.firstSuspendedTime = this.firstPendingTime = 0;
  }
  function gs(e, t) {
    var n = e.firstSuspendedTime;
    return e = e.lastSuspendedTime, n !== 0 && n >= t && e <= t;
  }
  function Bt(e, t) {
    var n = e.firstSuspendedTime, r = e.lastSuspendedTime;
    n < t && (e.firstSuspendedTime = t), (r > t || n === 0) && (e.lastSuspendedTime = t), t <= e.lastPingedTime && (e.lastPingedTime = 0), t <= e.lastExpiredTime && (e.lastExpiredTime = 0);
  }
  function ws(e, t) {
    t > e.firstPendingTime && (e.firstPendingTime = t);
    var n = e.firstSuspendedTime;
    n !== 0 && (t >= n ? e.firstSuspendedTime = e.lastSuspendedTime = e.nextKnownPendingLevel = 0 : t >= e.lastSuspendedTime && (e.lastSuspendedTime = t + 1), t > e.nextKnownPendingLevel && (e.nextKnownPendingLevel = t));
  }
  function Oi(e, t) {
    var n = e.lastExpiredTime;
    (n === 0 || n > t) && (e.lastExpiredTime = t);
  }
  function Yr(e, t, n, r) {
    var l = t.current, i = Je(), o = jn.suspense;
    i = Wt(i, l, o);
    e:
      if (n) {
        n = n._reactInternalFiber;
        t: {
          if (zt(n) !== n || n.tag !== 1)
            throw Error(h(170));
          var u = n;
          do {
            switch (u.tag) {
              case 3:
                u = u.stateNode.context;
                break t;
              case 1:
                if (ve(u.type)) {
                  u = u.stateNode.__reactInternalMemoizedMergedChildContext;
                  break t;
                }
            }
            u = u.return;
          } while (u !== null);
          throw Error(h(171));
        }
        if (n.tag === 1) {
          var c = n.type;
          if (ve(c)) {
            n = nu(n, c, u);
            break e;
          }
        }
        n = u;
      } else
        n = gt;
    return t.context === null ? t.context = n : t.pendingContext = n, t = Tt(i, o), t.payload = { element: e }, r = r === void 0 ? null : r, r !== null && (t.callback = r), kt(l, t), _t(l, i), i;
  }
  function Ri(e) {
    if (e = e.current, !e.child)
      return null;
    switch (e.child.tag) {
      case 5:
        return e.child.stateNode;
      default:
        return e.child.stateNode;
    }
  }
  function Es(e, t) {
    e = e.memoizedState, e !== null && e.dehydrated !== null && e.retryTime < t && (e.retryTime = t);
  }
  function Mi(e, t) {
    Es(e, t), (e = e.alternate) && Es(e, t);
  }
  function zi(e, t, n) {
    n = n != null && n.hydrate === !0;
    var r = new sf(e, t, n), l = qe(3, null, null, t === 2 ? 7 : t === 1 ? 3 : 0);
    r.current = l, l.stateNode = r, $l(l), e[kn] = r.current, n && t !== 0 && Us(e, e.nodeType === 9 ? e : e.ownerDocument), this._internalRoot = r;
  }
  zi.prototype.render = function(e) {
    Yr(e, this._internalRoot, null, null);
  }, zi.prototype.unmount = function() {
    var e = this._internalRoot, t = e.containerInfo;
    Yr(null, e, null, function() {
      t[kn] = null;
    });
  };
  function $n(e) {
    return !(!e || e.nodeType !== 1 && e.nodeType !== 9 && e.nodeType !== 11 && (e.nodeType !== 8 || e.nodeValue !== " react-mount-point-unstable "));
  }
  function af(e, t) {
    if (t || (t = e ? e.nodeType === 9 ? e.documentElement : e.firstChild : null, t = !(!t || t.nodeType !== 1 || !t.hasAttribute("data-reactroot"))), !t)
      for (var n; n = e.lastChild; )
        e.removeChild(n);
    return new zi(e, 0, t ? { hydrate: !0 } : void 0);
  }
  function Xr(e, t, n, r, l) {
    var i = n._reactRootContainer;
    if (i) {
      var o = i._internalRoot;
      if (typeof l == "function") {
        var u = l;
        l = function() {
          var d = Ri(o);
          u.call(d);
        };
      }
      Yr(t, o, e, l);
    } else {
      if (i = n._reactRootContainer = af(n, r), o = i._internalRoot, typeof l == "function") {
        var c = l;
        l = function() {
          var d = Ri(o);
          c.call(d);
        };
      }
      fs(function() {
        Yr(t, o, e, l);
      });
    }
    return Ri(o);
  }
  function ff(e, t, n) {
    var r = 3 < arguments.length && arguments[3] !== void 0 ? arguments[3] : null;
    return { $$typeof: Yt, key: r == null ? null : "" + r, children: e, containerInfo: t, implementation: n };
  }
  ho = function(e) {
    if (e.tag === 13) {
      var t = hr(Je(), 150, 100);
      _t(e, t), Mi(e, t);
    }
  }, cl = function(e) {
    e.tag === 13 && (_t(e, 3), Mi(e, 3));
  }, vo = function(e) {
    if (e.tag === 13) {
      var t = Je();
      t = Wt(t, e, null), _t(e, t), Mi(e, t);
    }
  }, Me = function(e, t, n) {
    switch (t) {
      case "input":
        if (tl(e, n), t = n.name, n.type === "radio" && t != null) {
          for (n = e; n.parentNode; )
            n = n.parentNode;
          for (n = n.querySelectorAll("input[name=" + JSON.stringify("" + t) + '][type="radio"]'), t = 0; t < n.length; t++) {
            var r = n[t];
            if (r !== e && r.form === e.form) {
              var l = Rl(r);
              if (!l)
                throw Error(h(90));
              Ki(r), tl(r, l);
            }
          }
        }
        break;
      case "textarea":
        Zi(e, n);
        break;
      case "select":
        t = n.value, t != null && Xt(e, !!n.multiple, t, !1);
    }
  }, ze = as, de = function(e, t, n, r, l) {
    var i = N;
    N |= 4;
    try {
      return wt(98, e.bind(null, t, n, r, l));
    } finally {
      N = i, N === te && Ye();
    }
  }, Te = function() {
    (N & (1 | We | Ze)) === te && (Za(), fn());
  }, nt = function(e, t) {
    var n = N;
    N |= 2;
    try {
      return e(t);
    } finally {
      N = n, N === te && Ye();
    }
  };
  function Ts(e, t) {
    var n = 2 < arguments.length && arguments[2] !== void 0 ? arguments[2] : null;
    if (!$n(t))
      throw Error(h(200));
    return ff(e, t, null, n);
  }
  var cf = { Events: [Sn, It, Rl, q, $e, qt, function(e) {
    sl(e, Js);
  }, B, $, nr, qn, fn, { current: !1 }] };
  return function(e) {
    var t = e.findFiberByHostInstance;
    return lf(O({}, e, { overrideHookState: null, overrideProps: null, setSuspenseHandler: null, scheduleUpdate: null, currentDispatcherRef: Ae.ReactCurrentDispatcher, findHostInstanceByFiber: function(n) {
      return n = ao(n), n === null ? null : n.stateNode;
    }, findFiberByHostInstance: function(n) {
      return t ? t(n) : null;
    }, findHostInstancesForRefresh: null, scheduleRefresh: null, scheduleRoot: null, setRefreshHandler: null, getCurrentFiber: null }));
  }({
    findFiberByHostInstance: xn,
    bundleType: 0,
    version: "16.14.0",
    rendererPackageName: "react-dom"
  }), Oe.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED = cf, Oe.createPortal = Ts, Oe.findDOMNode = function(e) {
    if (e == null)
      return null;
    if (e.nodeType === 1)
      return e;
    var t = e._reactInternalFiber;
    if (t === void 0)
      throw typeof e.render == "function" ? Error(h(188)) : Error(h(268, Object.keys(e)));
    return e = ao(t), e = e === null ? null : e.stateNode, e;
  }, Oe.flushSync = function(e, t) {
    if ((N & (We | Ze)) !== te)
      throw Error(h(187));
    var n = N;
    N |= 1;
    try {
      return wt(99, e.bind(null, t));
    } finally {
      N = n, Ye();
    }
  }, Oe.hydrate = function(e, t, n) {
    if (!$n(t))
      throw Error(h(200));
    return Xr(null, e, t, !0, n);
  }, Oe.render = function(e, t, n) {
    if (!$n(t))
      throw Error(h(200));
    return Xr(null, e, t, !1, n);
  }, Oe.unmountComponentAtNode = function(e) {
    if (!$n(e))
      throw Error(h(40));
    return e._reactRootContainer ? (fs(function() {
      Xr(null, null, e, !1, function() {
        e._reactRootContainer = null, e[kn] = null;
      });
    }), !0) : !1;
  }, Oe.unstable_batchedUpdates = as, Oe.unstable_createPortal = function(e, t) {
    return Ts(e, t, 2 < arguments.length && arguments[2] !== void 0 ? arguments[2] : null);
  }, Oe.unstable_renderSubtreeIntoContainer = function(e, t, n, r) {
    if (!$n(n))
      throw Error(h(200));
    if (e == null || e._reactInternalFiber === void 0)
      throw Error(h(38));
    return Xr(e, t, n, !1, r);
  }, Oe.version = "16.14.0", Oe;
}
var Ns;
function Tf() {
  if (Ns)
    return Fi.exports;
  Ns = 1;
  function S() {
    if (!(typeof __REACT_DEVTOOLS_GLOBAL_HOOK__ > "u" || typeof __REACT_DEVTOOLS_GLOBAL_HOOK__.checkDCE != "function"))
      try {
        __REACT_DEVTOOLS_GLOBAL_HOOK__.checkDCE(S);
      } catch (O) {
        console.error(O);
      }
  }
  return S(), Fi.exports = yf(), Fi.exports;
}
export {
  gf as commonjsGlobal,
  Ef as getAugmentedNamespace,
  wf as getDefaultExportFromCjs,
  mf as requireReact,
  Tf as requireReactDom
};
